{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### civic feature engineering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Part 1: fasta, hhm & AF2-pdb rASA/SS, smile coding, fingerprint"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# split fasta\n",
    "civic_gene_table = pd.read_csv('../datasets/middlefile/civic_gene_table_fixed.csv')\n",
    "for i in range(civic_gene_table.shape[0]):\n",
    "    uniprotac = civic_gene_table['uniprotac'][i]\n",
    "    fasta = civic_gene_table['fasta'][i]\n",
    "    with open('../datasets/middlefile/fasta/' + uniprotac + '.fasta', 'w+') as out:\n",
    "        out.write('>' + uniprotac + '\\n')\n",
    "        out.write(fasta)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "pharmgkb_gene_table = pd.read_csv('../datasets/middlefile/pharmgkb_gene_table_fixed.csv')\n",
    "for i in range(pharmgkb_gene_table.shape[0]):\n",
    "    uniprotac = pharmgkb_gene_table['uniprotac'][i]\n",
    "    fasta = pharmgkb_gene_table['fasta'][i]\n",
    "    with open('../datasets/middlefile/fasta/' + uniprotac + '.fasta', 'w+') as out:\n",
    "        out.write('>' + uniprotac + '\\n')\n",
    "        out.write(fasta)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# alphafold database fetch\n",
    "import requests\n",
    "for i in range(civic_gene_table.shape[0]):\n",
    "    uniprotac = civic_gene_table['uniprotac'][i]\n",
    "    url = 'https://alphafold.ebi.ac.uk/files/AF-' + uniprotac + '-F1-model_v4.pdb'\n",
    "    f = requests.get(url)\n",
    "    with open('../datasets/middlefile/AF2pdb/' + uniprotac + '.pdb', 'wb') as out:\n",
    "        out.write(f.content)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "for i in range(pharmgkb_gene_table.shape[0]):\n",
    "    uniprotac = pharmgkb_gene_table['uniprotac'][i]\n",
    "    url = 'https://alphafold.ebi.ac.uk/files/AF-' + uniprotac + '-F1-model_v4.pdb'\n",
    "    f = requests.get(url)\n",
    "    with open('../datasets/middlefile/AF2pdb/' + uniprotac + '.pdb', 'wb') as out:\n",
    "        out.write(f.content)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# generage fasta with mutation\n",
    "civic_evidence_table = pd.read_csv('../datasets/middlefile/civic_evidence_table.csv')\n",
    "for i in range(civic_evidence_table.shape[0]):\n",
    "    gene = civic_evidence_table['gene'][i]\n",
    "    variant = civic_evidence_table['variant'][i]\n",
    "    fasta = civic_gene_table[civic_gene_table['gene'] == gene]['fasta'].values[0]\n",
    "    pos = int(variant[1:-1])\n",
    "    pos_after = variant[-1]\n",
    "    fasta = fasta[:pos-1] + pos_after + fasta[pos:]\n",
    "    with open('../datasets/middlefile/fasta/' + gene + '_' + variant + '.fasta', 'w+') as out:\n",
    "        out.write('>' + gene + '_' + variant + '\\n')\n",
    "        out.write(fasta)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "pharmgkb_evidence_table = pd.read_csv('../datasets/middlefile/pharmgkb_evidence_table.csv')\n",
    "for i in range(pharmgkb_evidence_table.shape[0]):\n",
    "    gene = pharmgkb_evidence_table['gene'][i]\n",
    "    variant = pharmgkb_evidence_table['variant'][i]\n",
    "    fasta = pharmgkb_gene_table[pharmgkb_gene_table['gene'] == gene]['fasta'].values[0]\n",
    "    pos = int(variant[1:-1])\n",
    "    pos_after = variant[-1]\n",
    "    fasta = fasta[:pos-1] + pos_after + fasta[pos:]\n",
    "    with open('../datasets/middlefile/fasta/' + gene + '_' + variant + '.fasta', 'w+') as out:\n",
    "        out.write('>' + gene + '_' + variant + '\\n')\n",
    "        out.write(fasta)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "maxasa_dict = {\n",
    "'C':167, 'D':193, 'S':155, 'Q':225, 'K':236,\n",
    "'I':197, 'P':159, 'T':172, 'F':240, 'N':195,\n",
    "'G':104, 'H':224, 'L':201, 'R':274, 'W':285,\n",
    "'A':129, 'V':174, 'E':223, 'Y':263, 'M':224\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "eyes = np.eye(3)\n",
    "ss_dict = {\n",
    "'H':eyes[0],'G':eyes[0],'I':eyes[0],\n",
    "'B':eyes[1],'E':eyes[1],\n",
    "'T':eyes[2]\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# calculate rASA and padding\n",
    "import os\n",
    "import math\n",
    "zero_list = ['P51587', 'Q13315', 'P04114', 'O60673', 'P21817', 'P42858', 'P98164', 'Q8WXI7', 'Q9H251']\n",
    "pdb_name_list = os.listdir('../datasets/middlefile/AF2pdb/')\n",
    "pdb_name_list = [x.split('.')[0] for x in pdb_name_list]\n",
    "for pdb in pdb_name_list:\n",
    "    try:\n",
    "        with open('../datasets/middlefile/fasta/' + pdb +'.fasta') as file:\n",
    "            fasta_file = file.readlines()\n",
    "            fasta_len = len(fasta_file[1])\n",
    "        if(pdb in zero_list): # large protein\n",
    "            ss_matrix = np.zeros([fasta_len, 3], int)\n",
    "            with open('../datasets/middlefile/SS/' + pdb + '.ss','w+') as out_file:\n",
    "                np.savetxt(out_file, ss_matrix, fmt='%.1f')\n",
    "            rasa_matrix = np.zeros([fasta_len, 1], float)\n",
    "            with open('../datasets/middlefile/rASA/' + pdb + '.rasa','w+') as out_file:\n",
    "                np.savetxt(out_file, rasa_matrix, fmt='%.3f')\n",
    "        else:\n",
    "            ss_matrix = np.zeros([fasta_len, 3], int)\n",
    "            rasa_matrix = np.zeros([fasta_len, 1], float)\n",
    "            with open('../datasets/middlefile/dssp/' + pdb +'.dssp') as dssp_file:\n",
    "                line = dssp_file.readline()\n",
    "                while line:\n",
    "                    if(line.split()[0] == '#'):\n",
    "                        break\n",
    "                    line = dssp_file.readline()\n",
    "                line = dssp_file.readline()\n",
    "                index = 0\n",
    "                while line:\n",
    "                    if(len(line.split()) > 0):\n",
    "                        SS = line[16]\n",
    "                        ACC = int(line[35:38].strip())\n",
    "                        AA = line[13]\n",
    "                        # check ss\n",
    "                        if(SS != ' '):\n",
    "                            if(SS not in ss_dict.keys()):\n",
    "                                ss_matrix[index][0:3] = eyes[2]\n",
    "                            else:\n",
    "                                ss_matrix[index][0:3] = ss_dict[SS]\n",
    "                        # check rasa\n",
    "                        rasa_matrix[index][0] = float(ACC)/maxasa_dict[AA]\n",
    "                        index += 1                    \n",
    "                    line = dssp_file.readline()\n",
    "            with open('../datasets/middlefile/SS/' + pdb + '.ss','w+') as out_file:\n",
    "                np.savetxt(out_file, ss_matrix, fmt='%.1f')\n",
    "            with open('../datasets/middlefile/rASA/' + pdb + '.rasa','w+') as out_file:\n",
    "                np.savetxt(out_file, rasa_matrix, fmt='%.3f')\n",
    "    except IndexError:\n",
    "        print(pdb)\n",
    "        continue\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 97/97 [02:11<00:00,  1.36s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "            drugname                                              smile  \\\n",
      "0        Selumetinib  CN1C=NC2=C1C=C(C(=C2F)NC3=C(C=C(C=C3)Br)Cl)C(=...   \n",
      "1           Imatinib  CC1=C(C=C(C=C1)NC(=O)C2=CC=C(C=C2)CN3CCN(CC3)C...   \n",
      "2          Sorafenib  CNC(=O)C1=NC=CC(=C1)OC2=CC=C(C=C2)NC(=O)NC3=CC...   \n",
      "3         Fedratinib  CC1=CN=C(N=C1NC2=CC(=CC=C2)S(=O)(=O)NC(C)(C)C)...   \n",
      "4       Tanespimycin  C[C@H]1C[C@@H]([C@@H]([C@H](/C=C(/[C@@H]([C@H]...   \n",
      "..               ...                                                ...   \n",
      "92  Arsenic Trioxide                    [O-2].[O-2].[O-2].[As+3].[As+3]   \n",
      "93        Venetoclax  CC1(CCC(=C(C1)C2=CC=C(C=C2)Cl)CN3CCN(CC3)C4=CC...   \n",
      "94      Gilteritinib  CCC1=C(N=C(C(=N1)C(=O)N)NC2=CC(=C(C=C2)N3CCC(C...   \n",
      "95      Tazemetostat  CCN(C1CCOCC1)C2=CC(=CC(=C2C)C(=O)NCC3=C(C=C(NC...   \n",
      "96       Pemigatinib  CCN1C2=C3C=C(NC3=NC=C2CN(C1=O)C4=C(C(=CC(=C4F)...   \n",
      "\n",
      "   molecular_weight molecular_formula  \\\n",
      "0             457.7   C17H15BrClFN4O3   \n",
      "1             493.6         C29H31N7O   \n",
      "2             464.8    C21H16ClF3N4O3   \n",
      "3             524.7       C27H36N6O3S   \n",
      "4             585.7        C31H43N3O8   \n",
      "..              ...               ...   \n",
      "92          197.841             As2O3   \n",
      "93            868.4     C45H50ClN7O7S   \n",
      "94            552.7        C29H44N8O3   \n",
      "95            572.7        C34H44N4O4   \n",
      "96            487.5      C24H27F2N5O4   \n",
      "\n",
      "                                                 atom  \\\n",
      "0   [Atom(1, Br), Atom(2, Cl), Atom(3, F), Atom(4,...   \n",
      "1   [Atom(1, O), Atom(2, N), Atom(3, N), Atom(4, N...   \n",
      "2   [Atom(1, Cl), Atom(2, F), Atom(3, F), Atom(4, ...   \n",
      "3   [Atom(1, S), Atom(2, O), Atom(3, O), Atom(4, O...   \n",
      "4   [Atom(1, O), Atom(2, O), Atom(3, O), Atom(4, O...   \n",
      "..                                                ...   \n",
      "92  [Atom(1, As), Atom(2, As), Atom(3, O), Atom(4,...   \n",
      "93  [Atom(1, Cl), Atom(2, S), Atom(3, O), Atom(4, ...   \n",
      "94  [Atom(1, O), Atom(2, O), Atom(3, O), Atom(4, N...   \n",
      "95  [Atom(1, O), Atom(2, O), Atom(3, O), Atom(4, O...   \n",
      "96  [Atom(1, F), Atom(2, F), Atom(3, O), Atom(4, O...   \n",
      "\n",
      "                                          fingerprint  \\\n",
      "0   00000371C07BB100041000000000000000000000000160...   \n",
      "1   00000371E07BA000000000000000000000000000000000...   \n",
      "2   00000371E07BB180040000000000000000000000000000...   \n",
      "3   00000371F07BB000400000000000000000000000000160...   \n",
      "4   00000371F07B3C00000000000000000000000000000000...   \n",
      "..                                                ...   \n",
      "92  0000037100003000000000004000000000000000000000...   \n",
      "93  00000371F07FB800440000000000000000000000000160...   \n",
      "94  00000371F07BF000000000000000000000000000000000...   \n",
      "95  00000371F07FB800000000000000000000000000000000...   \n",
      "96  00000371E07BB980000000000000000000000000000160...   \n",
      "\n",
      "                                   cactvs_fingerprint  \n",
      "0   1100000001111011101100010000000000000100000100...  \n",
      "1   1110000001111011101000000000000000000000000000...  \n",
      "2   1110000001111011101100011000000000000100000000...  \n",
      "3   1111000001111011101100000000000001000000000000...  \n",
      "4   1111000001111011001111000000000000000000000000...  \n",
      "..                                                ...  \n",
      "92  0000000000000000001100000000000000000000000000...  \n",
      "93  1111000001111111101110000000000001000100000000...  \n",
      "94  1111000001111011111100000000000000000000000000...  \n",
      "95  1111000001111111101110000000000000000000000000...  \n",
      "96  1110000001111011101110011000000000000000000000...  \n",
      "\n",
      "[97 rows x 7 columns]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "# mapping pubchem fingerprint\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "import pubchempy as pcp\n",
    "from tqdm import tqdm\n",
    "\n",
    "drug_table = pd.read_csv('../datasets/middlefile/civic_drug_table.csv')\n",
    "new_drug_table = pd.DataFrame(columns=['drugname', 'smile', 'molecular_weight', 'molecular_formula', 'atom', 'fingerprint', 'cactvs_fingerprint'])\n",
    "for i in tqdm(range(drug_table.shape[0])):\n",
    "    drugname = drug_table['drugname'][i]\n",
    "    compound = pcp.get_compounds(drugname,'name')[0]\n",
    "    try:\n",
    "        smile = compound.isomeric_smiles\n",
    "    except AttributeError:\n",
    "        smile = np.nan\n",
    "    try:\n",
    "        molecular_weight = compound.molecular_weight\n",
    "    except AttributeError:\n",
    "        molecular_weight = np.nan   \n",
    "    try:\n",
    "        molecular_formula = compound.molecular_formula\n",
    "    except AttributeError:\n",
    "        molecular_formula = np.nan\n",
    "    try: \n",
    "        atom = compound.atoms\n",
    "    except AttributeError:\n",
    "        atom = np.nan\n",
    "    try:\n",
    "        fingerprint = compound.fingerprint\n",
    "    except AttributeError:\n",
    "        fingerprint = np.nan\n",
    "    try:\n",
    "        cactvs_fingerprint = compound.cactvs_fingerprint\n",
    "    except AttributeError:\n",
    "        cactvs_fingerprint = np.nan\n",
    "    new_drug_table = new_drug_table.append([{'drugname':drugname, 'smile':smile, 'molecular_weight':molecular_weight, 'molecular_formula':molecular_formula, \n",
    "                                    'atom':atom, 'fingerprint':fingerprint, 'cactvs_fingerprint':cactvs_fingerprint}], ignore_index=True)\n",
    "print(new_drug_table)\n",
    "new_drug_table.to_csv('../datasets/middlefile/civic_drug_table_fpfixed.csv', index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "# smile dict from DeepDTA (source: https://github.com/hkmztrk/DeepDTA/blob/master/source/datahelper.py)\n",
    "CHARISOSMISET = {\"#\": 29, \"%\": 30, \")\": 31, \"(\": 1, \"+\": 32, \"-\": 33, \"/\": 34, \".\": 2, \n",
    "\t\t\t\t\"1\": 35, \"0\": 3, \"3\": 36, \"2\": 4, \"5\": 37, \"4\": 5, \"7\": 38, \"6\": 6, \n",
    "\t\t\t\t\"9\": 39, \"8\": 7, \"=\": 40, \"A\": 41, \"@\": 8, \"C\": 42, \"B\": 9, \"E\": 43, \n",
    "\t\t\t\t\"D\": 10, \"G\": 44, \"F\": 11, \"I\": 45, \"H\": 12, \"K\": 46, \"M\": 47, \"L\": 13, \n",
    "\t\t\t\t\"O\": 48, \"N\": 14, \"P\": 15, \"S\": 49, \"R\": 16, \"U\": 50, \"T\": 17, \"W\": 51, \n",
    "\t\t\t\t\"V\": 18, \"Y\": 52, \"[\": 53, \"Z\": 19, \"]\": 54, \"\\\\\": 20, \"a\": 55, \"c\": 56, \n",
    "\t\t\t\t\"b\": 21, \"e\": 57, \"d\": 22, \"g\": 58, \"f\": 23, \"i\": 59, \"h\": 24, \"m\": 60, \n",
    "\t\t\t\t\"l\": 25, \"o\": 61, \"n\": 26, \"s\": 62, \"r\": 27, \"u\": 63, \"t\": 28, \"y\": 64}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 97/97 [00:00<00:00, 3888.85it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "            drugname                                              smile  \\\n",
      "0        Selumetinib  CN1C=NC2=C1C=C(C(=C2F)NC3=C(C=C(C=C3)Br)Cl)C(=...   \n",
      "1           Imatinib  CC1=C(C=C(C=C1)NC(=O)C2=CC=C(C=C2)CN3CCN(CC3)C...   \n",
      "2          Sorafenib  CNC(=O)C1=NC=CC(=C1)OC2=CC=C(C=C2)NC(=O)NC3=CC...   \n",
      "3         Fedratinib  CC1=CN=C(N=C1NC2=CC(=CC=C2)S(=O)(=O)NC(C)(C)C)...   \n",
      "4       Tanespimycin  C[C@H]1C[C@@H]([C@@H]([C@H](/C=C(/[C@@H]([C@H]...   \n",
      "..               ...                                                ...   \n",
      "92  Arsenic Trioxide                    [O-2].[O-2].[O-2].[As+3].[As+3]   \n",
      "93        Venetoclax  CC1(CCC(=C(C1)C2=CC=C(C=C2)Cl)CN3CCN(CC3)C4=CC...   \n",
      "94      Gilteritinib  CCC1=C(N=C(C(=N1)C(=O)N)NC2=CC(=C(C=C2)N3CCC(C...   \n",
      "95      Tazemetostat  CCN(C1CCOCC1)C2=CC(=CC(=C2C)C(=O)NCC3=C(C=C(NC...   \n",
      "96       Pemigatinib  CCN1C2=C3C=C(NC3=NC=C2CN(C1=O)C4=C(C(=CC(=C4F)...   \n",
      "\n",
      "    molecular_weight molecular_formula  \\\n",
      "0            457.700   C17H15BrClFN4O3   \n",
      "1            493.600         C29H31N7O   \n",
      "2            464.800    C21H16ClF3N4O3   \n",
      "3            524.700       C27H36N6O3S   \n",
      "4            585.700        C31H43N3O8   \n",
      "..               ...               ...   \n",
      "92           197.841             As2O3   \n",
      "93           868.400     C45H50ClN7O7S   \n",
      "94           552.700        C29H44N8O3   \n",
      "95           572.700        C34H44N4O4   \n",
      "96           487.500      C24H27F2N5O4   \n",
      "\n",
      "                                                 atom  \\\n",
      "0   [Atom(1, Br), Atom(2, Cl), Atom(3, F), Atom(4,...   \n",
      "1   [Atom(1, O), Atom(2, N), Atom(3, N), Atom(4, N...   \n",
      "2   [Atom(1, Cl), Atom(2, F), Atom(3, F), Atom(4, ...   \n",
      "3   [Atom(1, S), Atom(2, O), Atom(3, O), Atom(4, O...   \n",
      "4   [Atom(1, O), Atom(2, O), Atom(3, O), Atom(4, O...   \n",
      "..                                                ...   \n",
      "92  [Atom(1, As), Atom(2, As), Atom(3, O), Atom(4,...   \n",
      "93  [Atom(1, Cl), Atom(2, S), Atom(3, O), Atom(4, ...   \n",
      "94  [Atom(1, O), Atom(2, O), Atom(3, O), Atom(4, N...   \n",
      "95  [Atom(1, O), Atom(2, O), Atom(3, O), Atom(4, O...   \n",
      "96  [Atom(1, F), Atom(2, F), Atom(3, O), Atom(4, O...   \n",
      "\n",
      "                                          fingerprint  \\\n",
      "0   00000371C07BB100041000000000000000000000000160...   \n",
      "1   00000371E07BA000000000000000000000000000000000...   \n",
      "2   00000371E07BB180040000000000000000000000000000...   \n",
      "3   00000371F07BB000400000000000000000000000000160...   \n",
      "4   00000371F07B3C00000000000000000000000000000000...   \n",
      "..                                                ...   \n",
      "92  0000037100003000000000004000000000000000000000...   \n",
      "93  00000371F07FB800440000000000000000000000000160...   \n",
      "94  00000371F07BF000000000000000000000000000000000...   \n",
      "95  00000371F07FB800000000000000000000000000000000...   \n",
      "96  00000371E07BB980000000000000000000000000000160...   \n",
      "\n",
      "                                   cactvs_fingerprint  \\\n",
      "0   1100000001111011101100010000000000000100000100...   \n",
      "1   1110000001111011101000000000000000000000000000...   \n",
      "2   1110000001111011101100011000000000000100000000...   \n",
      "3   1111000001111011101100000000000001000000000000...   \n",
      "4   1111000001111011001111000000000000000000000000...   \n",
      "..                                                ...   \n",
      "92  0000000000000000001100000000000000000000000000...   \n",
      "93  1111000001111111101110000000000001000100000000...   \n",
      "94  1111000001111011111100000000000000000000000000...   \n",
      "95  1111000001111111101110000000000000000000000000...   \n",
      "96  1110000001111011101110011000000000000000000000...   \n",
      "\n",
      "                                          smile_array  \n",
      "0   [42, 14, 35, 42, 40, 14, 42, 4, 40, 42, 35, 42...  \n",
      "1   [42, 42, 35, 40, 42, 1, 42, 40, 42, 1, 42, 40,...  \n",
      "2   [42, 14, 42, 1, 40, 48, 31, 42, 35, 40, 14, 42...  \n",
      "3   [42, 42, 35, 40, 42, 14, 40, 42, 1, 14, 40, 42...  \n",
      "4   [42, 53, 42, 8, 12, 54, 35, 42, 53, 42, 8, 8, ...  \n",
      "..                                                ...  \n",
      "92  [53, 48, 33, 4, 54, 2, 53, 48, 33, 4, 54, 2, 5...  \n",
      "93  [42, 42, 35, 1, 42, 42, 42, 1, 40, 42, 1, 42, ...  \n",
      "94  [42, 42, 42, 35, 40, 42, 1, 14, 40, 42, 1, 42,...  \n",
      "95  [42, 42, 14, 1, 42, 35, 42, 42, 48, 42, 42, 35...  \n",
      "96  [42, 42, 14, 35, 42, 4, 40, 42, 36, 42, 40, 42...  \n",
      "\n",
      "[97 rows x 8 columns]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "# encode smile\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "from tqdm import tqdm\n",
    "new_drug_table = pd.read_csv('../datasets/middlefile/civic_drug_table_fpfixed.csv')\n",
    "new_drug_table['smile_array'] = 0\n",
    "new_drug_table['smile_array'] = new_drug_table['smile_array'].astype(object)\n",
    "for i in tqdm(range(new_drug_table.shape[0])):\n",
    "    drugname = new_drug_table['drugname'][i]\n",
    "    smile = new_drug_table['smile'][i]\n",
    "    smile_num_list = []\n",
    "    for strr in smile:\n",
    "        smile_num_list.append(CHARISOSMISET[strr])\n",
    "    #smile_num = np.array(smile_num_list)\n",
    "    #new_drug_table['smile_array'][i] = new_drug_table['smile_array'][i].apply(lambda x: smile_num_list)\n",
    "    new_drug_table.loc[:,'smile_array'].loc[i] = np.array(smile_num_list)\n",
    "print(new_drug_table)\n",
    "new_drug_table.to_pickle('../datasets/middlefile/civic_drug_table_fpfixed_smilenum.pkl')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 223/223 [00:00<00:00, 4067.68it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         drugname                                              smile  \\\n",
      "0    capecitabine  CCCCCOC(=O)NC1=NC(=O)N(C=C1F)[C@H]2[C@@H]([C@@...   \n",
      "1    fluorouracil                               C1=C(C(=O)NC(=O)N1)F   \n",
      "2        warfarin       CC(=O)CC(C1=CC=CC=C1)C2=C(C3=CC=CC=C3OC2=O)O   \n",
      "3       gefitinib  COC1=C(C=C2C(=C1)N=CN=C2NC3=CC(=C(C=C3)F)Cl)OC...   \n",
      "4       efavirenz   C1CC1C#C[C@]2(C3=C(C=CC(=C3)Cl)NC(=O)O2)C(F)(F)F   \n",
      "..            ...                                                ...   \n",
      "218      thiotepa                             C1CN1P(=S)(N2CC2)N3CC3   \n",
      "219      levodopa                  C1=CC(=C(C=C1C[C@@H](C(=O)O)N)O)O   \n",
      "220    naltrexone  C1CC1CN2CC[C@]34[C@@H]5C(=O)CC[C@]3([C@H]2CC6=...   \n",
      "221    folic acid  C1=CC(=CC=C1C(=O)N[C@@H](CCC(=O)O)C(=O)O)NCC2=...   \n",
      "222  Glucarpidase                        [Zn+2].[Zn+2].[Zn+2].[Zn+2]   \n",
      "\n",
      "     molecular_weight molecular_formula  \\\n",
      "0              359.35       C15H22FN3O6   \n",
      "1              130.08         C4H3FN2O2   \n",
      "2              308.30          C19H16O4   \n",
      "3              446.90     C22H24ClFN4O3   \n",
      "4              315.67      C14H9ClF3NO2   \n",
      "..                ...               ...   \n",
      "218            189.22         C6H12N3PS   \n",
      "219            197.19          C9H11NO4   \n",
      "220            341.40         C20H23NO4   \n",
      "221            441.40        C19H19N7O6   \n",
      "222            261.50             Zn4+8   \n",
      "\n",
      "                                                  atom  \\\n",
      "0    [Atom(1, F), Atom(2, O), Atom(3, O), Atom(4, O...   \n",
      "1    [Atom(1, F), Atom(2, O), Atom(3, O), Atom(4, N...   \n",
      "2    [Atom(1, O), Atom(2, O), Atom(3, O), Atom(4, O...   \n",
      "3    [Atom(1, Cl), Atom(2, F), Atom(3, O), Atom(4, ...   \n",
      "4    [Atom(1, Cl), Atom(2, F), Atom(3, F), Atom(4, ...   \n",
      "..                                                 ...   \n",
      "218  [Atom(1, S), Atom(2, P), Atom(3, N), Atom(4, N...   \n",
      "219  [Atom(1, O), Atom(2, O), Atom(3, O), Atom(4, O...   \n",
      "220  [Atom(1, O), Atom(2, O), Atom(3, O), Atom(4, O...   \n",
      "221  [Atom(1, O), Atom(2, O), Atom(3, O), Atom(4, O...   \n",
      "222  [Atom(1, Zn), Atom(2, Zn), Atom(3, Zn), Atom(4...   \n",
      "\n",
      "                                           fingerprint  \\\n",
      "0    00000371E0733900000000000000000000000000000120...   \n",
      "1    0000037100633100000000000000000000000000000000...   \n",
      "2    00000371E0783800000000000000000000000000000000...   \n",
      "3    00000371E07BB100040000000000000000000000000000...   \n",
      "4    00000371C0723180040000000000000000001800000000...   \n",
      "..                                                 ...   \n",
      "218  00000371C063000240000000000000000000162C000000...   \n",
      "219  00000371C0723800000000000000000000000000000000...   \n",
      "220  00000371E07A3800000000000000000000001800000120...   \n",
      "221  00000371E07BB800000000000000000000000000000000...   \n",
      "222  0000037100000000000000020000000000000000000000...   \n",
      "\n",
      "                                    cactvs_fingerprint  \\\n",
      "0    1110000001110011001110010000000000000000000000...   \n",
      "1    0000000001100011001100010000000000000000000000...   \n",
      "2    1110000001111000001110000000000000000000000000...   \n",
      "3    1110000001111011101100010000000000000100000000...   \n",
      "4    1100000001110010001100011000000000000100000000...   \n",
      "..                                                 ...   \n",
      "218  1100000001100011000000000000001001000000000000...   \n",
      "219  1100000001110010001110000000000000000000000000...   \n",
      "220  1110000001111010001110000000000000000000000000...   \n",
      "221  1110000001111011101110000000000000000000000000...   \n",
      "222  0000000000000000000000000000000000000000000000...   \n",
      "\n",
      "                                           smile_array  \n",
      "0    [42, 42, 42, 42, 42, 48, 42, 1, 40, 48, 31, 14...  \n",
      "1    [42, 35, 40, 42, 1, 42, 1, 40, 48, 31, 14, 42,...  \n",
      "2    [42, 42, 1, 40, 48, 31, 42, 42, 1, 42, 35, 40,...  \n",
      "3    [42, 48, 42, 35, 40, 42, 1, 42, 40, 42, 4, 42,...  \n",
      "4    [42, 35, 42, 42, 35, 42, 29, 42, 53, 42, 8, 54...  \n",
      "..                                                 ...  \n",
      "218  [42, 35, 42, 14, 35, 15, 1, 40, 49, 31, 1, 14,...  \n",
      "219  [42, 35, 40, 42, 42, 1, 40, 42, 1, 42, 40, 42,...  \n",
      "220  [42, 35, 42, 42, 35, 42, 14, 4, 42, 42, 53, 42...  \n",
      "221  [42, 35, 40, 42, 42, 1, 40, 42, 42, 40, 42, 35...  \n",
      "222  [53, 19, 26, 32, 4, 54, 2, 53, 19, 26, 32, 4, ...  \n",
      "\n",
      "[223 rows x 8 columns]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "# encode smile\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "from tqdm import tqdm\n",
    "new_drug_table = pd.read_csv('../datasets/middlefile/pharmgkb_drug_table_fpfixed.csv')\n",
    "new_drug_table['smile_array'] = 0\n",
    "new_drug_table['smile_array'] = new_drug_table['smile_array'].astype(object)\n",
    "for i in tqdm(range(new_drug_table.shape[0])):\n",
    "    drugname = new_drug_table['drugname'][i]\n",
    "    smile = new_drug_table['smile'][i]\n",
    "    smile_num_list = []\n",
    "    for strr in smile:\n",
    "        smile_num_list.append(CHARISOSMISET[strr])\n",
    "    #smile_num = np.array(smile_num_list)\n",
    "    #new_drug_table['smile_array'][i] = new_drug_table['smile_array'][i].apply(lambda x: smile_num_list)\n",
    "    new_drug_table.loc[:,'smile_array'].loc[i] = np.array(smile_num_list)\n",
    "print(new_drug_table)\n",
    "new_drug_table.to_pickle('../datasets/middlefile/pharmgkb_drug_table_fpfixed_smilenum.pkl')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Part 2: label processing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>variant</th>\n",
       "      <th>drugs</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>reference_bases</th>\n",
       "      <th>variant_bases</th>\n",
       "      <th>representative_transcript</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>P124S</td>\n",
       "      <td>Selumetinib</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>15</td>\n",
       "      <td>66729162.0</td>\n",
       "      <td>66729162.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000307102.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>Q56P</td>\n",
       "      <td>Selumetinib</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>15</td>\n",
       "      <td>66727451.0</td>\n",
       "      <td>66727451.0</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>ENST00000307102.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PDGFRA</td>\n",
       "      <td>D842V</td>\n",
       "      <td>Imatinib</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>4</td>\n",
       "      <td>55152093.0</td>\n",
       "      <td>55152093.0</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000257290.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ARAF</td>\n",
       "      <td>S214C</td>\n",
       "      <td>Sorafenib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>X</td>\n",
       "      <td>47426121.0</td>\n",
       "      <td>47426121.0</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>ENST00000377045.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JAK2</td>\n",
       "      <td>V617F</td>\n",
       "      <td>Fedratinib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>9</td>\n",
       "      <td>5073770.0</td>\n",
       "      <td>5073770.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000381652.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>439</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>G250E</td>\n",
       "      <td>Dasatinib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>9</td>\n",
       "      <td>133738349.0</td>\n",
       "      <td>133738349.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ENST00000318560.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>440</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>H396P</td>\n",
       "      <td>Dasatinib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>9</td>\n",
       "      <td>133750356.0</td>\n",
       "      <td>133750356.0</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>ENST00000318560.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>441</th>\n",
       "      <td>EGFR</td>\n",
       "      <td>G598V</td>\n",
       "      <td>Erlotinib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>7</td>\n",
       "      <td>55233043.0</td>\n",
       "      <td>55233043.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000275493.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>442</th>\n",
       "      <td>IDH1</td>\n",
       "      <td>R132C</td>\n",
       "      <td>Olaparib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>2</td>\n",
       "      <td>209113113.0</td>\n",
       "      <td>209113113.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ENST00000415913.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>443</th>\n",
       "      <td>FGFR1</td>\n",
       "      <td>N546K</td>\n",
       "      <td>Pemigatinib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>8</td>\n",
       "      <td>38274849.0</td>\n",
       "      <td>38274849.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000341462.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>444 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       gene variant        drugs clinical_significance chromosome  \\\n",
       "0    MAP2K1   P124S  Selumetinib            Resistance         15   \n",
       "1    MAP2K1    Q56P  Selumetinib            Resistance         15   \n",
       "2    PDGFRA   D842V     Imatinib            Resistance          4   \n",
       "3      ARAF   S214C    Sorafenib  Sensitivity/Response          X   \n",
       "4      JAK2   V617F   Fedratinib  Sensitivity/Response          9   \n",
       "..      ...     ...          ...                   ...        ...   \n",
       "439    ABL1   G250E    Dasatinib  Sensitivity/Response          9   \n",
       "440    ABL1   H396P    Dasatinib  Sensitivity/Response          9   \n",
       "441    EGFR   G598V    Erlotinib  Sensitivity/Response          7   \n",
       "442    IDH1   R132C     Olaparib  Sensitivity/Response          2   \n",
       "443   FGFR1   N546K  Pemigatinib  Sensitivity/Response          8   \n",
       "\n",
       "           start         stop reference_bases variant_bases  \\\n",
       "0     66729162.0   66729162.0               C             T   \n",
       "1     66727451.0   66727451.0               A             C   \n",
       "2     55152093.0   55152093.0               A             T   \n",
       "3     47426121.0   47426121.0               C             G   \n",
       "4      5073770.0    5073770.0               G             T   \n",
       "..           ...          ...             ...           ...   \n",
       "439  133738349.0  133738349.0               G             A   \n",
       "440  133750356.0  133750356.0               A             C   \n",
       "441   55233043.0   55233043.0               G             T   \n",
       "442  209113113.0  209113113.0               G             A   \n",
       "443   38274849.0   38274849.0               G             T   \n",
       "\n",
       "    representative_transcript  \n",
       "0           ENST00000307102.5  \n",
       "1           ENST00000307102.5  \n",
       "2           ENST00000257290.5  \n",
       "3           ENST00000377045.4  \n",
       "4           ENST00000381652.3  \n",
       "..                        ...  \n",
       "439         ENST00000318560.5  \n",
       "440         ENST00000318560.5  \n",
       "441         ENST00000275493.2  \n",
       "442         ENST00000415913.1  \n",
       "443         ENST00000341462.5  \n",
       "\n",
       "[444 rows x 10 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# civic label preprocessing\n",
    "civic_evidence_table = pd.read_csv('../datasets/middlefile/civic_evidence_table.csv')\n",
    "civic_evidence_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Resistance              217\n",
       "Sensitivity/Response    215\n",
       "Reduced Sensitivity      10\n",
       "Adverse Response          2\n",
       "Name: clinical_significance, dtype: int64"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "civic_evidence_table['clinical_significance'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "civic_label_dict = {\n",
    "    'Resistance': 0, \n",
    "    'Sensitivity/Response': 1, 'Reduced Sensitivity': 1, 'Adverse Response': 1\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>uniprotac</th>\n",
       "      <th>variant</th>\n",
       "      <th>drug</th>\n",
       "      <th>smile</th>\n",
       "      <th>label</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>Q02750</td>\n",
       "      <td>P124S</td>\n",
       "      <td>Selumetinib</td>\n",
       "      <td>CN1C=NC2=C1C=C(C(=C2F)NC3=C(C=C(C=C3)Br)Cl)C(=...</td>\n",
       "      <td>0</td>\n",
       "      <td>civic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>Q02750</td>\n",
       "      <td>Q56P</td>\n",
       "      <td>Selumetinib</td>\n",
       "      <td>CN1C=NC2=C1C=C(C(=C2F)NC3=C(C=C(C=C3)Br)Cl)C(=...</td>\n",
       "      <td>0</td>\n",
       "      <td>civic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PDGFRA</td>\n",
       "      <td>P16234</td>\n",
       "      <td>D842V</td>\n",
       "      <td>Imatinib</td>\n",
       "      <td>CC1=C(C=C(C=C1)NC(=O)C2=CC=C(C=C2)CN3CCN(CC3)C...</td>\n",
       "      <td>0</td>\n",
       "      <td>civic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ARAF</td>\n",
       "      <td>P10398</td>\n",
       "      <td>S214C</td>\n",
       "      <td>Sorafenib</td>\n",
       "      <td>CNC(=O)C1=NC=CC(=C1)OC2=CC=C(C=C2)NC(=O)NC3=CC...</td>\n",
       "      <td>1</td>\n",
       "      <td>civic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JAK2</td>\n",
       "      <td>O60674</td>\n",
       "      <td>V617F</td>\n",
       "      <td>Fedratinib</td>\n",
       "      <td>CC1=CN=C(N=C1NC2=CC(=CC=C2)S(=O)(=O)NC(C)(C)C)...</td>\n",
       "      <td>1</td>\n",
       "      <td>civic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>439</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>P00519</td>\n",
       "      <td>G250E</td>\n",
       "      <td>Dasatinib</td>\n",
       "      <td>CC1=C(C(=CC=C1)Cl)NC(=O)C2=CN=C(S2)NC3=CC(=NC(...</td>\n",
       "      <td>1</td>\n",
       "      <td>civic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>440</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>P00519</td>\n",
       "      <td>H396P</td>\n",
       "      <td>Dasatinib</td>\n",
       "      <td>CC1=C(C(=CC=C1)Cl)NC(=O)C2=CN=C(S2)NC3=CC(=NC(...</td>\n",
       "      <td>1</td>\n",
       "      <td>civic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>441</th>\n",
       "      <td>EGFR</td>\n",
       "      <td>P00533</td>\n",
       "      <td>G598V</td>\n",
       "      <td>Erlotinib</td>\n",
       "      <td>COCCOC1=C(C=C2C(=C1)C(=NC=N2)NC3=CC=CC(=C3)C#C...</td>\n",
       "      <td>1</td>\n",
       "      <td>civic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>442</th>\n",
       "      <td>IDH1</td>\n",
       "      <td>O75874</td>\n",
       "      <td>R132C</td>\n",
       "      <td>Olaparib</td>\n",
       "      <td>C1CC1C(=O)N2CCN(CC2)C(=O)C3=C(C=CC(=C3)CC4=NNC...</td>\n",
       "      <td>1</td>\n",
       "      <td>civic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>443</th>\n",
       "      <td>FGFR1</td>\n",
       "      <td>P11362</td>\n",
       "      <td>N546K</td>\n",
       "      <td>Pemigatinib</td>\n",
       "      <td>CCN1C2=C3C=C(NC3=NC=C2CN(C1=O)C4=C(C(=CC(=C4F)...</td>\n",
       "      <td>1</td>\n",
       "      <td>civic</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>444 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       gene uniprotac variant         drug  \\\n",
       "0    MAP2K1    Q02750   P124S  Selumetinib   \n",
       "1    MAP2K1    Q02750    Q56P  Selumetinib   \n",
       "2    PDGFRA    P16234   D842V     Imatinib   \n",
       "3      ARAF    P10398   S214C    Sorafenib   \n",
       "4      JAK2    O60674   V617F   Fedratinib   \n",
       "..      ...       ...     ...          ...   \n",
       "439    ABL1    P00519   G250E    Dasatinib   \n",
       "440    ABL1    P00519   H396P    Dasatinib   \n",
       "441    EGFR    P00533   G598V    Erlotinib   \n",
       "442    IDH1    O75874   R132C     Olaparib   \n",
       "443   FGFR1    P11362   N546K  Pemigatinib   \n",
       "\n",
       "                                                 smile label source  \n",
       "0    CN1C=NC2=C1C=C(C(=C2F)NC3=C(C=C(C=C3)Br)Cl)C(=...     0  civic  \n",
       "1    CN1C=NC2=C1C=C(C(=C2F)NC3=C(C=C(C=C3)Br)Cl)C(=...     0  civic  \n",
       "2    CC1=C(C=C(C=C1)NC(=O)C2=CC=C(C=C2)CN3CCN(CC3)C...     0  civic  \n",
       "3    CNC(=O)C1=NC=CC(=C1)OC2=CC=C(C=C2)NC(=O)NC3=CC...     1  civic  \n",
       "4    CC1=CN=C(N=C1NC2=CC(=CC=C2)S(=O)(=O)NC(C)(C)C)...     1  civic  \n",
       "..                                                 ...   ...    ...  \n",
       "439  CC1=C(C(=CC=C1)Cl)NC(=O)C2=CN=C(S2)NC3=CC(=NC(...     1  civic  \n",
       "440  CC1=C(C(=CC=C1)Cl)NC(=O)C2=CN=C(S2)NC3=CC(=NC(...     1  civic  \n",
       "441  COCCOC1=C(C=C2C(=C1)C(=NC=N2)NC3=CC=CC(=C3)C#C...     1  civic  \n",
       "442  C1CC1C(=O)N2CCN(CC2)C(=O)C3=C(C=CC(=C3)CC4=NNC...     1  civic  \n",
       "443  CCN1C2=C3C=C(NC3=NC=C2CN(C1=O)C4=C(C(=CC(=C4F)...     1  civic  \n",
       "\n",
       "[444 rows x 7 columns]"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# generate check table\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "civic_gene_table = pd.read_csv('../datasets/middlefile/civic_gene_table_fixed.csv')\n",
    "civic_drug_table = pd.read_pickle('../datasets/middlefile/civic_drug_table_fpfixed_smilenum.pkl')\n",
    "civic_labeled_evidence = pd.DataFrame(columns=['gene', 'uniprotac', 'variant', 'drug', 'smile', 'label', 'source'])\n",
    "for i in range(civic_evidence_table.shape[0]):\n",
    "    gene = civic_evidence_table['gene'][i]\n",
    "    uniprotac = civic_gene_table[civic_gene_table['gene'] == gene]['uniprotac'].values[0]\n",
    "    variant = civic_evidence_table['variant'][i]\n",
    "    drug = civic_evidence_table['drugs'][i]\n",
    "    smile = civic_drug_table[civic_drug_table['drugname'] == drug]['smile'].values[0]\n",
    "    label = civic_label_dict[civic_evidence_table['clinical_significance'][i]]\n",
    "    source = 'civic'\n",
    "    civic_labeled_evidence = civic_labeled_evidence.append([{'gene': gene, 'uniprotac': uniprotac, 'variant': variant, 'drug': drug, \n",
    "                                                            'smile': smile, 'label': label, 'source': source}], ignore_index=True)\n",
    "civic_labeled_evidence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>variant</th>\n",
       "      <th>rsid</th>\n",
       "      <th>drugs</th>\n",
       "      <th>type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DPYD</td>\n",
       "      <td>M166V</td>\n",
       "      <td>rs2297595</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DPYD</td>\n",
       "      <td>C29R</td>\n",
       "      <td>rs1801265</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DPYD</td>\n",
       "      <td>V732I</td>\n",
       "      <td>rs1801160</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DPYD</td>\n",
       "      <td>I543V</td>\n",
       "      <td>rs1801159</td>\n",
       "      <td>capecitabine</td>\n",
       "      <td>Toxicity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>DPYD</td>\n",
       "      <td>S534N</td>\n",
       "      <td>rs1801158</td>\n",
       "      <td>fluorouracil</td>\n",
       "      <td>Toxicity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>856</th>\n",
       "      <td>MTHFR</td>\n",
       "      <td>A222V</td>\n",
       "      <td>rs1801133</td>\n",
       "      <td>methotrexate</td>\n",
       "      <td>Efficacy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>857</th>\n",
       "      <td>ABCC4</td>\n",
       "      <td>K1116N</td>\n",
       "      <td>rs1751034</td>\n",
       "      <td>tenofovir</td>\n",
       "      <td>Metabolism/PK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>858</th>\n",
       "      <td>ABCB1</td>\n",
       "      <td>I1145M</td>\n",
       "      <td>rs1045642</td>\n",
       "      <td>tramadol</td>\n",
       "      <td>Toxicity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>859</th>\n",
       "      <td>SLCO1B1</td>\n",
       "      <td>V174A</td>\n",
       "      <td>rs4149056</td>\n",
       "      <td>Glucarpidase</td>\n",
       "      <td>Toxicity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>860</th>\n",
       "      <td>VEGFB</td>\n",
       "      <td>D136E</td>\n",
       "      <td>rs12366035</td>\n",
       "      <td>sorafenib</td>\n",
       "      <td>Toxicity</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>861 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        gene variant        rsid         drugs           type\n",
       "0       DPYD   M166V   rs2297595  capecitabine       Toxicity\n",
       "1       DPYD    C29R   rs1801265  capecitabine       Toxicity\n",
       "2       DPYD   V732I   rs1801160  capecitabine       Toxicity\n",
       "3       DPYD   I543V   rs1801159  capecitabine       Toxicity\n",
       "4       DPYD   S534N   rs1801158  fluorouracil       Toxicity\n",
       "..       ...     ...         ...           ...            ...\n",
       "856    MTHFR   A222V   rs1801133  methotrexate       Efficacy\n",
       "857    ABCC4  K1116N   rs1751034     tenofovir  Metabolism/PK\n",
       "858    ABCB1  I1145M   rs1045642      tramadol       Toxicity\n",
       "859  SLCO1B1   V174A   rs4149056  Glucarpidase       Toxicity\n",
       "860    VEGFB   D136E  rs12366035     sorafenib       Toxicity\n",
       "\n",
       "[861 rows x 5 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# pharmgkb label preprocessing\n",
    "pharmgkb_evidence_table = pd.read_csv('../datasets/middlefile/pharmgkb_evidence_table.csv')\n",
    "pharmgkb_evidence_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Efficacy                           311\n",
       "Toxicity                           235\n",
       "Metabolism/PK                      187\n",
       "Other                               65\n",
       "Dosage                              44\n",
       "Efficacy,Toxicity                    8\n",
       "Efficacy,Metabolism/PK               5\n",
       "Dosage,Toxicity                      3\n",
       "Efficacy,Toxicity,Metabolism/PK      2\n",
       "Dosage,Efficacy                      1\n",
       "Name: type, dtype: int64"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pharmgkb_evidence_table['type'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>uniprotac</th>\n",
       "      <th>variant</th>\n",
       "      <th>drug</th>\n",
       "      <th>smile</th>\n",
       "      <th>label</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>EGFR</td>\n",
       "      <td>P00533</td>\n",
       "      <td>L858R</td>\n",
       "      <td>gefitinib</td>\n",
       "      <td>COC1=C(C=C2C(=C1)N=CN=C2NC3=CC(=C(C=C3)F)Cl)OC...</td>\n",
       "      <td>1</td>\n",
       "      <td>pharmgkb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>P13569</td>\n",
       "      <td>S1251N</td>\n",
       "      <td>ivacaftor</td>\n",
       "      <td>CC(C)(C)C1=CC(=C(C=C1NC(=O)C2=CNC3=CC=CC=C3C2=...</td>\n",
       "      <td>1</td>\n",
       "      <td>pharmgkb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>P13569</td>\n",
       "      <td>G1244V</td>\n",
       "      <td>ivacaftor</td>\n",
       "      <td>CC(C)(C)C1=CC(=C(C=C1NC(=O)C2=CNC3=CC=CC=C3C2=...</td>\n",
       "      <td>1</td>\n",
       "      <td>pharmgkb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>P13569</td>\n",
       "      <td>D110H</td>\n",
       "      <td>ivacaftor</td>\n",
       "      <td>CC(C)(C)C1=CC(=C(C=C1NC(=O)C2=CNC3=CC=CC=C3C2=...</td>\n",
       "      <td>1</td>\n",
       "      <td>pharmgkb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CFTR</td>\n",
       "      <td>P13569</td>\n",
       "      <td>L206W</td>\n",
       "      <td>ivacaftor</td>\n",
       "      <td>CC(C)(C)C1=CC(=C(C=C1NC(=O)C2=CNC3=CC=CC=C3C2=...</td>\n",
       "      <td>1</td>\n",
       "      <td>pharmgkb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>306</th>\n",
       "      <td>NQO1</td>\n",
       "      <td>P15559</td>\n",
       "      <td>P187S</td>\n",
       "      <td>warfarin</td>\n",
       "      <td>CC(=O)CC(C1=CC=CC=C1)C2=C(C3=CC=CC=C3OC2=O)O</td>\n",
       "      <td>1</td>\n",
       "      <td>pharmgkb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>307</th>\n",
       "      <td>MTR</td>\n",
       "      <td>Q99707</td>\n",
       "      <td>D919G</td>\n",
       "      <td>methotrexate</td>\n",
       "      <td>CN(CC1=CN=C2C(=N1)C(=NC(=N2)N)N)C3=CC=C(C=C3)C...</td>\n",
       "      <td>1</td>\n",
       "      <td>pharmgkb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>308</th>\n",
       "      <td>ADH1C</td>\n",
       "      <td>P00326</td>\n",
       "      <td>I350V</td>\n",
       "      <td>naltrexone</td>\n",
       "      <td>C1CC1CN2CC[C@]34[C@@H]5C(=O)CC[C@]3([C@H]2CC6=...</td>\n",
       "      <td>1</td>\n",
       "      <td>pharmgkb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309</th>\n",
       "      <td>ADH1B</td>\n",
       "      <td>P00325</td>\n",
       "      <td>R370C</td>\n",
       "      <td>naltrexone</td>\n",
       "      <td>C1CC1CN2CC[C@]34[C@@H]5C(=O)CC[C@]3([C@H]2CC6=...</td>\n",
       "      <td>1</td>\n",
       "      <td>pharmgkb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>310</th>\n",
       "      <td>MTHFR</td>\n",
       "      <td>P42898</td>\n",
       "      <td>A222V</td>\n",
       "      <td>methotrexate</td>\n",
       "      <td>CN(CC1=CN=C2C(=N1)C(=NC(=N2)N)N)C3=CC=C(C=C3)C...</td>\n",
       "      <td>1</td>\n",
       "      <td>pharmgkb</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>311 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      gene uniprotac variant          drug  \\\n",
       "0     EGFR    P00533   L858R     gefitinib   \n",
       "1     CFTR    P13569  S1251N     ivacaftor   \n",
       "2     CFTR    P13569  G1244V     ivacaftor   \n",
       "3     CFTR    P13569   D110H     ivacaftor   \n",
       "4     CFTR    P13569   L206W     ivacaftor   \n",
       "..     ...       ...     ...           ...   \n",
       "306   NQO1    P15559   P187S      warfarin   \n",
       "307    MTR    Q99707   D919G  methotrexate   \n",
       "308  ADH1C    P00326   I350V    naltrexone   \n",
       "309  ADH1B    P00325   R370C    naltrexone   \n",
       "310  MTHFR    P42898   A222V  methotrexate   \n",
       "\n",
       "                                                 smile label    source  \n",
       "0    COC1=C(C=C2C(=C1)N=CN=C2NC3=CC(=C(C=C3)F)Cl)OC...     1  pharmgkb  \n",
       "1    CC(C)(C)C1=CC(=C(C=C1NC(=O)C2=CNC3=CC=CC=C3C2=...     1  pharmgkb  \n",
       "2    CC(C)(C)C1=CC(=C(C=C1NC(=O)C2=CNC3=CC=CC=C3C2=...     1  pharmgkb  \n",
       "3    CC(C)(C)C1=CC(=C(C=C1NC(=O)C2=CNC3=CC=CC=C3C2=...     1  pharmgkb  \n",
       "4    CC(C)(C)C1=CC(=C(C=C1NC(=O)C2=CNC3=CC=CC=C3C2=...     1  pharmgkb  \n",
       "..                                                 ...   ...       ...  \n",
       "306       CC(=O)CC(C1=CC=CC=C1)C2=C(C3=CC=CC=C3OC2=O)O     1  pharmgkb  \n",
       "307  CN(CC1=CN=C2C(=N1)C(=NC(=N2)N)N)C3=CC=C(C=C3)C...     1  pharmgkb  \n",
       "308  C1CC1CN2CC[C@]34[C@@H]5C(=O)CC[C@]3([C@H]2CC6=...     1  pharmgkb  \n",
       "309  C1CC1CN2CC[C@]34[C@@H]5C(=O)CC[C@]3([C@H]2CC6=...     1  pharmgkb  \n",
       "310  CN(CC1=CN=C2C(=N1)C(=NC(=N2)N)N)C3=CC=C(C=C3)C...     1  pharmgkb  \n",
       "\n",
       "[311 rows x 7 columns]"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# generate check table\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "pharmgkb_gene_table = pd.read_csv('../datasets/middlefile/pharmgkb_gene_table_fixed.csv')\n",
    "pharmgkb_drug_table = pd.read_pickle('../datasets/middlefile/pharmgkb_drug_table_fpfixed_smilenum.pkl')\n",
    "pharmgkb_labeled_evidence = pd.DataFrame(columns=['gene', 'uniprotac', 'variant', 'drug', 'smile', 'label', 'source'])\n",
    "for i in range(pharmgkb_evidence_table.shape[0]):\n",
    "    gene = pharmgkb_evidence_table['gene'][i]\n",
    "    uniprotac = pharmgkb_gene_table[pharmgkb_gene_table['gene'] == gene]['uniprotac'].values[0]\n",
    "    variant = pharmgkb_evidence_table['variant'][i]\n",
    "    drug = pharmgkb_evidence_table['drugs'][i]\n",
    "    smile = pharmgkb_drug_table[pharmgkb_drug_table['drugname'] == drug]['smile'].values[0]\n",
    "    if(pharmgkb_evidence_table['type'][i] == 'Efficacy'):\n",
    "        label = 1\n",
    "    else:\n",
    "        continue\n",
    "    source = 'pharmgkb'\n",
    "    pharmgkb_labeled_evidence = pharmgkb_labeled_evidence.append([{'gene': gene, 'uniprotac': uniprotac, 'variant': variant, 'drug': drug, \n",
    "                                                            'smile': smile, 'label': label, 'source': source}], ignore_index=True)\n",
    "pharmgkb_labeled_evidence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alpha</th>\n",
       "      <th>num</th>\n",
       "      <th>hhh_x</th>\n",
       "      <th>hhh_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>a</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>b</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  alpha  num  hhh_x  hhh_y\n",
       "0     a    5      5      5\n",
       "1     b    6      5      5"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# test\n",
    "df1 = pd.DataFrame({'alpha':['a','b','c'],'num':[5,6,7],'hhh':[5,5,5]}, columns=['alpha','num','hhh'])\n",
    "df2 = pd.DataFrame({'alpha':['a','b','h'],'num':[5,6,2],'hhh':[5,5,5]}, columns=['alpha','num','hhh'])\n",
    "inner_df = pd.merge(df1, df2, on=['alpha','num'],how='inner')\n",
    "inner_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_x</th>\n",
       "      <th>uniprotac</th>\n",
       "      <th>variant</th>\n",
       "      <th>drug_x</th>\n",
       "      <th>smile</th>\n",
       "      <th>label_x</th>\n",
       "      <th>source_x</th>\n",
       "      <th>gene_y</th>\n",
       "      <th>drug_y</th>\n",
       "      <th>label_y</th>\n",
       "      <th>source_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [gene_x, uniprotac, variant, drug_x, smile, label_x, source_x, gene_y, drug_y, label_y, source_y]\n",
       "Index: []"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check overlap and conflicts\n",
    "inner_df = pd.merge(civic_labeled_evidence, pharmgkb_labeled_evidence, on=['uniprotac','variant','smile'],how='inner')\n",
    "inner_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove error evidences\n",
    "civic_labeled_evidence = civic_labeled_evidence[~((civic_labeled_evidence['gene']=='EGFR') & (civic_labeled_evidence['variant']=='T790M') & (civic_labeled_evidence['drug']=='Erlotinib'))]\n",
    "civic_labeled_evidence = civic_labeled_evidence[~((civic_labeled_evidence['gene']=='EGFR') & (civic_labeled_evidence['variant']=='T790M') & (civic_labeled_evidence['drug']=='Gefitinib'))]\n",
    "civic_labeled_evidence = civic_labeled_evidence[~((civic_labeled_evidence['gene']=='EGFR') & (civic_labeled_evidence['variant']=='L858R') & (civic_labeled_evidence['drug']=='Gefitinib'))]\n",
    "civic_labeled_evidence = civic_labeled_evidence[~((civic_labeled_evidence['gene']=='FLT3') & (civic_labeled_evidence['variant']=='T227M') & (civic_labeled_evidence['drug']=='Sunitinib'))]\n",
    "pharmgkb_labeled_evidence = pharmgkb_labeled_evidence[~((pharmgkb_labeled_evidence['gene']=='EGFR') & (pharmgkb_labeled_evidence['variant']=='T790M') & (pharmgkb_labeled_evidence['drug']=='erlotinib'))]\n",
    "pharmgkb_labeled_evidence = pharmgkb_labeled_evidence[~((pharmgkb_labeled_evidence['gene']=='EGFR') & (pharmgkb_labeled_evidence['variant']=='T790M') & (pharmgkb_labeled_evidence['drug']=='gefitinib'))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>uniprotac</th>\n",
       "      <th>variant</th>\n",
       "      <th>drug</th>\n",
       "      <th>smile</th>\n",
       "      <th>label</th>\n",
       "      <th>source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>Q02750</td>\n",
       "      <td>P124S</td>\n",
       "      <td>Selumetinib</td>\n",
       "      <td>CN1C=NC2=C1C=C(C(=C2F)NC3=C(C=C(C=C3)Br)Cl)C(=...</td>\n",
       "      <td>0</td>\n",
       "      <td>civic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>Q02750</td>\n",
       "      <td>Q56P</td>\n",
       "      <td>Selumetinib</td>\n",
       "      <td>CN1C=NC2=C1C=C(C(=C2F)NC3=C(C=C(C=C3)Br)Cl)C(=...</td>\n",
       "      <td>0</td>\n",
       "      <td>civic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PDGFRA</td>\n",
       "      <td>P16234</td>\n",
       "      <td>D842V</td>\n",
       "      <td>Imatinib</td>\n",
       "      <td>CC1=C(C=C(C=C1)NC(=O)C2=CC=C(C=C2)CN3CCN(CC3)C...</td>\n",
       "      <td>0</td>\n",
       "      <td>civic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ARAF</td>\n",
       "      <td>P10398</td>\n",
       "      <td>S214C</td>\n",
       "      <td>Sorafenib</td>\n",
       "      <td>CNC(=O)C1=NC=CC(=C1)OC2=CC=C(C=C2)NC(=O)NC3=CC...</td>\n",
       "      <td>1</td>\n",
       "      <td>civic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JAK2</td>\n",
       "      <td>O60674</td>\n",
       "      <td>V617F</td>\n",
       "      <td>Fedratinib</td>\n",
       "      <td>CC1=CN=C(N=C1NC2=CC(=CC=C2)S(=O)(=O)NC(C)(C)C)...</td>\n",
       "      <td>1</td>\n",
       "      <td>civic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>750</th>\n",
       "      <td>NQO1</td>\n",
       "      <td>P15559</td>\n",
       "      <td>P187S</td>\n",
       "      <td>warfarin</td>\n",
       "      <td>CC(=O)CC(C1=CC=CC=C1)C2=C(C3=CC=CC=C3OC2=O)O</td>\n",
       "      <td>1</td>\n",
       "      <td>pharmgkb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>751</th>\n",
       "      <td>MTR</td>\n",
       "      <td>Q99707</td>\n",
       "      <td>D919G</td>\n",
       "      <td>methotrexate</td>\n",
       "      <td>CN(CC1=CN=C2C(=N1)C(=NC(=N2)N)N)C3=CC=C(C=C3)C...</td>\n",
       "      <td>1</td>\n",
       "      <td>pharmgkb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>752</th>\n",
       "      <td>ADH1C</td>\n",
       "      <td>P00326</td>\n",
       "      <td>I350V</td>\n",
       "      <td>naltrexone</td>\n",
       "      <td>C1CC1CN2CC[C@]34[C@@H]5C(=O)CC[C@]3([C@H]2CC6=...</td>\n",
       "      <td>1</td>\n",
       "      <td>pharmgkb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>753</th>\n",
       "      <td>ADH1B</td>\n",
       "      <td>P00325</td>\n",
       "      <td>R370C</td>\n",
       "      <td>naltrexone</td>\n",
       "      <td>C1CC1CN2CC[C@]34[C@@H]5C(=O)CC[C@]3([C@H]2CC6=...</td>\n",
       "      <td>1</td>\n",
       "      <td>pharmgkb</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>754</th>\n",
       "      <td>MTHFR</td>\n",
       "      <td>P42898</td>\n",
       "      <td>A222V</td>\n",
       "      <td>methotrexate</td>\n",
       "      <td>CN(CC1=CN=C2C(=N1)C(=NC(=N2)N)N)C3=CC=C(C=C3)C...</td>\n",
       "      <td>1</td>\n",
       "      <td>pharmgkb</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>755 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       gene uniprotac variant          drug  \\\n",
       "0    MAP2K1    Q02750   P124S   Selumetinib   \n",
       "1    MAP2K1    Q02750    Q56P   Selumetinib   \n",
       "2    PDGFRA    P16234   D842V      Imatinib   \n",
       "3      ARAF    P10398   S214C     Sorafenib   \n",
       "4      JAK2    O60674   V617F    Fedratinib   \n",
       "..      ...       ...     ...           ...   \n",
       "750    NQO1    P15559   P187S      warfarin   \n",
       "751     MTR    Q99707   D919G  methotrexate   \n",
       "752   ADH1C    P00326   I350V    naltrexone   \n",
       "753   ADH1B    P00325   R370C    naltrexone   \n",
       "754   MTHFR    P42898   A222V  methotrexate   \n",
       "\n",
       "                                                 smile label    source  \n",
       "0    CN1C=NC2=C1C=C(C(=C2F)NC3=C(C=C(C=C3)Br)Cl)C(=...     0     civic  \n",
       "1    CN1C=NC2=C1C=C(C(=C2F)NC3=C(C=C(C=C3)Br)Cl)C(=...     0     civic  \n",
       "2    CC1=C(C=C(C=C1)NC(=O)C2=CC=C(C=C2)CN3CCN(CC3)C...     0     civic  \n",
       "3    CNC(=O)C1=NC=CC(=C1)OC2=CC=C(C=C2)NC(=O)NC3=CC...     1     civic  \n",
       "4    CC1=CN=C(N=C1NC2=CC(=CC=C2)S(=O)(=O)NC(C)(C)C)...     1     civic  \n",
       "..                                                 ...   ...       ...  \n",
       "750       CC(=O)CC(C1=CC=CC=C1)C2=C(C3=CC=CC=C3OC2=O)O     1  pharmgkb  \n",
       "751  CN(CC1=CN=C2C(=N1)C(=NC(=N2)N)N)C3=CC=C(C=C3)C...     1  pharmgkb  \n",
       "752  C1CC1CN2CC[C@]34[C@@H]5C(=O)CC[C@]3([C@H]2CC6=...     1  pharmgkb  \n",
       "753  C1CC1CN2CC[C@]34[C@@H]5C(=O)CC[C@]3([C@H]2CC6=...     1  pharmgkb  \n",
       "754  CN(CC1=CN=C2C(=N1)C(=NC(=N2)N)N)C3=CC=C(C=C3)C...     1  pharmgkb  \n",
       "\n",
       "[755 rows x 7 columns]"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# merge dataset\n",
    "merged_dataset = civic_labeled_evidence.append(pharmgkb_labeled_evidence)\n",
    "merged_dataset = merged_dataset.reset_index(drop=True)\n",
    "merged_dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_dataset.to_csv('../datasets/merged_evidence.csv', index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'12345A54321'"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# test\n",
    "seqsample = 'OOOOO12345A54321DDG'\n",
    "mut = 'A11B'\n",
    "pos = int(mut[1:-1])\n",
    "windlen = 5\n",
    "seq5 = seqsample[pos-windlen-1:pos+windlen]\n",
    "seq5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "eyes = np.eye(20)\n",
    "protein_dict = {'C':eyes[0], 'D':eyes[1], 'S':eyes[2], 'Q':eyes[3], 'K':eyes[4],\n",
    "    'I':eyes[5], 'P':eyes[6], 'T':eyes[7], 'F':eyes[8], 'N':eyes[9],\n",
    "    'G':eyes[10], 'H':eyes[11], 'L':eyes[12], 'R':eyes[13], 'W':eyes[14],\n",
    "    'A':eyes[15], 'V':eyes[16], 'E':eyes[17], 'Y':eyes[18], 'M':eyes[19]}\n",
    "fingerprint_dict = {'1':1,'0':0}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 753/753 [00:28<00:00, 26.61it/s]\n"
     ]
    }
   ],
   "source": [
    "# feature coding & dataset generate\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "\n",
    "from tqdm import tqdm\n",
    "window_len = 25\n",
    "\n",
    "merged_dataset = pd.read_csv('../datasets/merged_evidence.csv')\n",
    "civic_drug_table = pd.read_pickle('../datasets/middlefile/civic_drug_table_fpfixed_smilenum.pkl')\n",
    "pharmgkb_drug_table = pd.read_pickle('../datasets/middlefile/pharmgkb_drug_table_fpfixed_smilenum.pkl')\n",
    "dataset_feature = pd.DataFrame(columns=['gene', 'uniprotac', 'variant', 'drug', \n",
    "                                        'smile', 'smile_num' ,'cactvs_fingerprint', 'molecular_weight', # drug features\n",
    "                                        'onehot_before', 'onehot_after', 'hhm_before', 'hhm_after', 'ss', 'rasa', # sequence features\n",
    "                                        'label', 'source'])\n",
    "for i in tqdm(range(merged_dataset.shape[0])):\n",
    "    gene = merged_dataset['gene'][i]\n",
    "    uniprotac = merged_dataset['uniprotac'][i]\n",
    "    variant = merged_dataset['variant'][i]\n",
    "    drug = merged_dataset['drug'][i]\n",
    "    smile = merged_dataset['smile'][i]\n",
    "    label = merged_dataset['label'][i]\n",
    "    source = merged_dataset['source'][i]\n",
    "    smile_array = np.zeros(shape=(85,))\n",
    "    # drug features from file\n",
    "    if(source == 'civic'):\n",
    "        smile_num = civic_drug_table[civic_drug_table['drugname'] == drug]['smile_array'].values[0]\n",
    "        for j in range(min(85, smile_num.shape[0])):\n",
    "            smile_array[j] = smile_num[j]\n",
    "        cactvs_fingerprint_str = civic_drug_table[civic_drug_table['drugname'] == drug]['cactvs_fingerprint'].values[0]\n",
    "        molecular_weight = civic_drug_table[civic_drug_table['drugname'] == drug]['molecular_weight'].values[0]\n",
    "    else:\n",
    "        smile_num = pharmgkb_drug_table[pharmgkb_drug_table['drugname'] == drug]['smile_array'].values[0]\n",
    "        for j in range(min(85, smile_num.shape[0])):\n",
    "            smile_array[j] = smile_num[j]\n",
    "        cactvs_fingerprint_str = pharmgkb_drug_table[pharmgkb_drug_table['drugname'] == drug]['cactvs_fingerprint'].values[0]\n",
    "        molecular_weight = pharmgkb_drug_table[pharmgkb_drug_table['drugname'] == drug]['molecular_weight'].values[0]\n",
    "    cactvs_fingerprint = []\n",
    "    for strr in cactvs_fingerprint_str:\n",
    "        cactvs_fingerprint.append(fingerprint_dict[strr])\n",
    "    cactvs_fingerprint = np.array(cactvs_fingerprint)\n",
    "    # sequence features from file\n",
    "    pos = int(variant[1:-1])\n",
    "    pos_after = variant[-1]\n",
    "    # fasta before\n",
    "    with open('../datasets/middlefile/fasta/' + uniprotac + '.fasta') as file:\n",
    "        fasta_file = file.readlines()\n",
    "        fasta_full = fasta_file[1]\n",
    "        seq_len = len(fasta_full)\n",
    "        onehot_before = []\n",
    "        if(pos <= window_len - 1):\n",
    "            for j in range(window_len-pos+1): # padding head\n",
    "                onehot_before.append(np.zeros(20))\n",
    "            for strr in fasta_full[0:pos+window_len]:\n",
    "                onehot_before.append(protein_dict[strr])\n",
    "        elif(seq_len - pos < window_len):\n",
    "            for strr in fasta_full[pos-window_len-1:]:\n",
    "                onehot_before.append(protein_dict[strr])\n",
    "            for j in range(window_len-seq_len+pos): # padding end\n",
    "                onehot_before.append(np.zeros(20))\n",
    "        else:        \n",
    "            for strr in fasta_full[pos-window_len-1:pos+window_len]:\n",
    "                onehot_before.append(protein_dict[strr])\n",
    "    onehot_before = np.array(onehot_before)\n",
    "    # fasta after\n",
    "    with open('../datasets/middlefile/fasta/' + gene + '_' + variant + '.fasta') as file:\n",
    "        fasta_file = file.readlines()\n",
    "        fasta_full = fasta_file[1]\n",
    "        #fasta_after = fasta_full[pos-window_len-1:pos+window_len]\n",
    "        onehot_after = []\n",
    "        if(pos <= window_len - 1):\n",
    "            for j in range(window_len-pos+1): # padding head\n",
    "                onehot_after.append(np.zeros(20))\n",
    "            for strr in fasta_full[0:pos+window_len]:\n",
    "                onehot_after.append(protein_dict[strr])\n",
    "        elif(seq_len - pos < window_len):\n",
    "            for strr in fasta_full[pos-window_len-1:]:\n",
    "                onehot_after.append(protein_dict[strr])\n",
    "            for j in range(window_len-seq_len+pos): # padding end\n",
    "                onehot_after.append(np.zeros(20))\n",
    "        else:        \n",
    "            for strr in fasta_full[pos-window_len-1:pos+window_len]:\n",
    "                onehot_after.append(protein_dict[strr])\n",
    "    fasta_len = len(fasta_full)\n",
    "    onehot_after = np.array(onehot_after)\n",
    "    # hhm before\n",
    "    with open('../datasets/middlefile/hhm/' + uniprotac + '.hhm') as hhm_file:     \n",
    "        hhm_matrix = np.zeros([fasta_len, 30], float)\n",
    "        hhm_line = hhm_file.readline()\n",
    "        idxx = 0\n",
    "        while(hhm_line[0] != '#'):\n",
    "            hhm_line = hhm_file.readline()\n",
    "        for i in range(0,5):\n",
    "            hhm_line = hhm_file.readline()\n",
    "        while hhm_line:\n",
    "            if(len(hhm_line.split()) == 23):\n",
    "                idxx += 1\n",
    "                if(idxx == fasta_len + 1):\n",
    "                    break\n",
    "                each_item = hhm_line.split()[2:22]\n",
    "                for idx, s in enumerate(each_item):\n",
    "                    if(s == '*'):\n",
    "                        each_item[idx] = '99999'                            \n",
    "                for j in range(0, 20):\n",
    "                    hhm_matrix[idxx - 1, j] = int(each_item[j])\n",
    "                    #hhm_matrix[idxx - 1, j] = 10/(1 + math.exp(-1 * int(each_item[j])/2000))                                              \n",
    "            elif(len(hhm_line.split()) == 10):\n",
    "                each_item = hhm_line.split()[0:10]\n",
    "                for idx, s in enumerate(each_item):\n",
    "                    if(s == '*'):\n",
    "                        each_item[idx] = '99999'                             \n",
    "                for j in range(20, 30):\n",
    "                    hhm_matrix[idxx - 1, j] = int(each_item[j - 20]) \n",
    "                    #hhm_matrix[idxx - 1, j] = 10/(1 + math.exp(-1 * int(each_item[j - 20])/2000))                                                               \n",
    "            hhm_line = hhm_file.readline()\n",
    "        if(pos <= window_len - 1):\n",
    "            padding = np.zeros(shape=[window_len-pos+1,30])\n",
    "            hhm_before_array = np.vstack((padding,hhm_matrix[0:pos+window_len, :])) \n",
    "        elif(seq_len - pos < window_len):\n",
    "            padding = np.zeros(shape=[window_len-seq_len+pos,30])\n",
    "            hhm_before_array = np.vstack((hhm_matrix[pos-window_len-1:, :],padding))\n",
    "        else:\n",
    "            hhm_before_array = hhm_matrix[pos-window_len-1:pos+window_len, :]\n",
    "        #hhm_before = hhm_before_array.tolist()\n",
    "    # hhm after\n",
    "    with open('../datasets/middlefile/hhm/' + gene + '_' + variant + '.hhm') as hhm_file:     \n",
    "        hhm_matrix = np.zeros([fasta_len, 30], float)\n",
    "        hhm_line = hhm_file.readline()\n",
    "        idxx = 0\n",
    "        while(hhm_line[0] != '#'):\n",
    "            hhm_line = hhm_file.readline()\n",
    "        for i in range(0,5):\n",
    "            hhm_line = hhm_file.readline()\n",
    "        while hhm_line:\n",
    "            if(len(hhm_line.split()) == 23):\n",
    "                idxx += 1\n",
    "                if(idxx == fasta_len + 1):\n",
    "                    break\n",
    "                each_item = hhm_line.split()[2:22]\n",
    "                for idx, s in enumerate(each_item):\n",
    "                    if(s == '*'):\n",
    "                        each_item[idx] = '99999'                            \n",
    "                for j in range(0, 20):\n",
    "                    hhm_matrix[idxx - 1, j] = int(each_item[j])\n",
    "                    #hhm_matrix[idxx - 1, j] = 10/(1 + math.exp(-1 * int(each_item[j])/2000))                                              \n",
    "            elif(len(hhm_line.split()) == 10):\n",
    "                each_item = hhm_line.split()[0:10]\n",
    "                for idx, s in enumerate(each_item):\n",
    "                    if(s == '*'):\n",
    "                        each_item[idx] = '99999'                             \n",
    "                for j in range(20, 30):\n",
    "                    hhm_matrix[idxx - 1, j] = int(each_item[j - 20]) \n",
    "                    #hhm_matrix[idxx - 1, j] = 10/(1 + math.exp(-1 * int(each_item[j - 20])/2000))                                                                        \n",
    "            hhm_line = hhm_file.readline()\n",
    "        if(pos <= window_len - 1):\n",
    "            padding = np.zeros(shape=[window_len-pos+1,30])\n",
    "            hhm_after_array = np.vstack((padding,hhm_matrix[0:pos+window_len, :])) \n",
    "        elif(seq_len - pos < window_len):\n",
    "            padding = np.zeros(shape=[window_len-seq_len+pos,30])\n",
    "            hhm_after_array = np.vstack((hhm_matrix[pos-window_len-1:, :],padding))\n",
    "        else:\n",
    "            hhm_after_array = hhm_matrix[pos-window_len-1:pos+window_len, :]\n",
    "        #hhm_after = hhm_after_array.tolist()\n",
    "    # rasa\n",
    "    rasa_array = np.loadtxt('../datasets/middlefile/rASA/' + uniprotac + '.rasa')\n",
    "    if(pos <= window_len - 1):\n",
    "        padding = np.zeros(window_len-pos+1)\n",
    "        rasa_array = np.append(padding,rasa_array[0:pos+window_len])\n",
    "    elif(seq_len - pos < window_len):\n",
    "        padding = np.zeros(window_len-seq_len+pos)\n",
    "        rasa_array = np.append(rasa_array[pos-window_len-1:],padding)\n",
    "    else:\n",
    "        rasa_array = rasa_array[pos-window_len-1:pos+window_len]\n",
    "    #rasa = rasa_array.tolist()\n",
    "    # ss\n",
    "    ss_array = np.loadtxt('../datasets/middlefile/SS/' + uniprotac + '.ss')\n",
    "    if(pos <= window_len - 1):\n",
    "        padding = np.zeros(shape=[window_len-pos+1,3])\n",
    "        ss_array = np.vstack((padding,ss_array[0:pos+window_len]))\n",
    "    elif(seq_len - pos < window_len):\n",
    "        padding = np.zeros(shape=[window_len-seq_len+pos,3])\n",
    "        ss_array = np.vstack((ss_array[pos-window_len-1:],padding))\n",
    "    else:\n",
    "        ss_array = ss_array[pos-window_len-1:pos+window_len, :]\n",
    "    #ss = ss_array.tolist()\n",
    "    dataset_feature = dataset_feature.append([{'gene': gene, 'uniprotac': uniprotac, 'variant': variant, 'drug': drug, \n",
    "                                        'smile': smile, 'smile_num':smile_array, 'cactvs_fingerprint': cactvs_fingerprint, 'molecular_weight': molecular_weight, \n",
    "                                        'onehot_before':onehot_before, 'onehot_after':onehot_after, \n",
    "                                        'hhm_before': hhm_before_array, 'hhm_after': hhm_after_array, 'ss': ss_array, 'rasa': rasa_array, \n",
    "                                        'label': label, 'source':source}], ignore_index=True)\n",
    "\n",
    "#dataset_feature"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(753, 16)\n"
     ]
    }
   ],
   "source": [
    "filtered_dataset_feature = dataset_feature.copy()\n",
    "#filtered_dataset_feature = dataset_feature[~(dataset_feature['smile'] == \"[Pt]\")]\n",
    "filtered_dataset_feature['smile'] = filtered_dataset_feature['smile'].replace('[Pt]', '[Pt]=[Pt]')\n",
    "print(filtered_dataset_feature.shape)\n",
    "filtered_dataset_feature = filtered_dataset_feature.reset_index(drop=True)\n",
    "#dataset_feature.to_csv('../datasets/dataset_featurecode.csv', index=None)\n",
    "filtered_dataset_feature.to_pickle('../datasets/dataset_featurecode.dataset')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       gene uniprotac variant          drug  \\\n",
      "0    MAP2K1    Q02750   P124S   Selumetinib   \n",
      "1    MAP2K1    Q02750    Q56P   Selumetinib   \n",
      "2    PDGFRA    P16234   D842V      Imatinib   \n",
      "3      ARAF    P10398   S214C     Sorafenib   \n",
      "4      JAK2    O60674   V617F    Fedratinib   \n",
      "..      ...       ...     ...           ...   \n",
      "749    NQO1    P15559   P187S      warfarin   \n",
      "750     MTR    Q99707   D919G  methotrexate   \n",
      "751   ADH1C    P00326   I350V    naltrexone   \n",
      "752   ADH1B    P00325   R370C    naltrexone   \n",
      "753   MTHFR    P42898   A222V  methotrexate   \n",
      "\n",
      "                                                 smile  \\\n",
      "0    CN1C=NC2=C1C=C(C(=C2F)NC3=C(C=C(C=C3)Br)Cl)C(=...   \n",
      "1    CN1C=NC2=C1C=C(C(=C2F)NC3=C(C=C(C=C3)Br)Cl)C(=...   \n",
      "2    CC1=C(C=C(C=C1)NC(=O)C2=CC=C(C=C2)CN3CCN(CC3)C...   \n",
      "3    CNC(=O)C1=NC=CC(=C1)OC2=CC=C(C=C2)NC(=O)NC3=CC...   \n",
      "4    CC1=CN=C(N=C1NC2=CC(=CC=C2)S(=O)(=O)NC(C)(C)C)...   \n",
      "..                                                 ...   \n",
      "749       CC(=O)CC(C1=CC=CC=C1)C2=C(C3=CC=CC=C3OC2=O)O   \n",
      "750  CN(CC1=CN=C2C(=N1)C(=NC(=N2)N)N)C3=CC=C(C=C3)C...   \n",
      "751  C1CC1CN2CC[C@]34[C@@H]5C(=O)CC[C@]3([C@H]2CC6=...   \n",
      "752  C1CC1CN2CC[C@]34[C@@H]5C(=O)CC[C@]3([C@H]2CC6=...   \n",
      "753  CN(CC1=CN=C2C(=N1)C(=NC(=N2)N)N)C3=CC=C(C=C3)C...   \n",
      "\n",
      "                                             smile_num  \\\n",
      "0    [42.0, 14.0, 35.0, 42.0, 40.0, 14.0, 42.0, 4.0...   \n",
      "1    [42.0, 14.0, 35.0, 42.0, 40.0, 14.0, 42.0, 4.0...   \n",
      "2    [42.0, 42.0, 35.0, 40.0, 42.0, 1.0, 42.0, 40.0...   \n",
      "3    [42.0, 14.0, 42.0, 1.0, 40.0, 48.0, 31.0, 42.0...   \n",
      "4    [42.0, 42.0, 35.0, 40.0, 42.0, 14.0, 40.0, 42....   \n",
      "..                                                 ...   \n",
      "749  [42.0, 42.0, 1.0, 40.0, 48.0, 31.0, 42.0, 42.0...   \n",
      "750  [42.0, 14.0, 1.0, 42.0, 42.0, 35.0, 40.0, 42.0...   \n",
      "751  [42.0, 35.0, 42.0, 42.0, 35.0, 42.0, 14.0, 4.0...   \n",
      "752  [42.0, 35.0, 42.0, 42.0, 35.0, 42.0, 14.0, 4.0...   \n",
      "753  [42.0, 14.0, 1.0, 42.0, 42.0, 35.0, 40.0, 42.0...   \n",
      "\n",
      "                                    cactvs_fingerprint  molecular_weight  \\\n",
      "0    [1, 1, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 1, ...             457.7   \n",
      "1    [1, 1, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 1, ...             457.7   \n",
      "2    [1, 1, 1, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 1, ...             493.6   \n",
      "3    [1, 1, 1, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 1, ...             464.8   \n",
      "4    [1, 1, 1, 1, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 1, ...             524.7   \n",
      "..                                                 ...               ...   \n",
      "749  [1, 1, 1, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, ...             308.3   \n",
      "750  [1, 1, 1, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 1, ...             454.4   \n",
      "751  [1, 1, 1, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 1, ...             341.4   \n",
      "752  [1, 1, 1, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 1, ...             341.4   \n",
      "753  [1, 1, 1, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 1, ...             454.4   \n",
      "\n",
      "    fasta_before fasta_after  \\\n",
      "0            NaN         NaN   \n",
      "1            NaN         NaN   \n",
      "2            NaN         NaN   \n",
      "3            NaN         NaN   \n",
      "4            NaN         NaN   \n",
      "..           ...         ...   \n",
      "749          NaN         NaN   \n",
      "750          NaN         NaN   \n",
      "751          NaN         NaN   \n",
      "752          NaN         NaN   \n",
      "753          NaN         NaN   \n",
      "\n",
      "                                         onehot_before  \\\n",
      "0    [[0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0,...   \n",
      "1    [[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,...   \n",
      "2    [[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,...   \n",
      "3    [[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0,...   \n",
      "4    [[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,...   \n",
      "..                                                 ...   \n",
      "749  [[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,...   \n",
      "750  [[0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,...   \n",
      "751  [[0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,...   \n",
      "752  [[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,...   \n",
      "753  [[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,...   \n",
      "\n",
      "                                          onehot_after  \\\n",
      "0    [[0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0,...   \n",
      "1    [[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,...   \n",
      "2    [[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,...   \n",
      "3    [[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0,...   \n",
      "4    [[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,...   \n",
      "..                                                 ...   \n",
      "749  [[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,...   \n",
      "750  [[0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,...   \n",
      "751  [[0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,...   \n",
      "752  [[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,...   \n",
      "753  [[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,...   \n",
      "\n",
      "                                            hhm_before  \\\n",
      "0    [[3801.0, 5836.0, 6623.0, 5140.0, 3578.0, 6501...   \n",
      "1    [[3356.0, 99999.0, 2348.0, 2687.0, 99999.0, 49...   \n",
      "2    [[4818.0, 4708.0, 6560.0, 6523.0, 6288.0, 1738...   \n",
      "3    [[4229.0, 99999.0, 5293.0, 4284.0, 6424.0, 510...   \n",
      "4    [[3961.0, 7091.0, 3693.0, 2289.0, 7040.0, 5510...   \n",
      "..                                                 ...   \n",
      "749  [[3350.0, 6099.0, 4389.0, 3257.0, 4894.0, 4639...   \n",
      "750  [[3526.0, 7493.0, 3529.0, 2827.0, 9552.0, 3701...   \n",
      "751  [[4024.0, 6677.0, 3578.0, 4213.0, 5167.0, 4849...   \n",
      "752  [[3535.0, 6324.0, 4388.0, 4474.0, 5453.0, 5161...   \n",
      "753  [[6864.0, 9124.0, 4049.0, 5979.0, 4470.0, 6720...   \n",
      "\n",
      "                                             hhm_after  \\\n",
      "0    [[3857.0, 5881.0, 6728.0, 4966.0, 3547.0, 6372...   \n",
      "1    [[3209.0, 99999.0, 2845.0, 2667.0, 99999.0, 50...   \n",
      "2    [[4645.0, 4869.0, 6738.0, 6428.0, 6171.0, 1749...   \n",
      "3    [[6783.0, 99999.0, 5219.0, 99999.0, 99999.0, 3...   \n",
      "4    [[3790.0, 6862.0, 3765.0, 2364.0, 7474.0, 5624...   \n",
      "..                                                 ...   \n",
      "749  [[3381.0, 6456.0, 4537.0, 3760.0, 4702.0, 4936...   \n",
      "750  [[3559.0, 7519.0, 3496.0, 2813.0, 9578.0, 3718...   \n",
      "751  [[3999.0, 6814.0, 3655.0, 4080.0, 5390.0, 4921...   \n",
      "752  [[3762.0, 6211.0, 4502.0, 4546.0, 5520.0, 4992...   \n",
      "753  [[6904.0, 9064.0, 4122.0, 5951.0, 4488.0, 6695...   \n",
      "\n",
      "                                                    ss  \\\n",
      "0    [[0.0, 1.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, ...   \n",
      "1    [[1.0, 0.0, 0.0], [1.0, 0.0, 0.0], [1.0, 0.0, ...   \n",
      "2    [[0.0, 0.0, 1.0], [0.0, 0.0, 0.0], [0.0, 0.0, ...   \n",
      "3    [[0.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 0.0, ...   \n",
      "4    [[1.0, 0.0, 0.0], [1.0, 0.0, 0.0], [1.0, 0.0, ...   \n",
      "..                                                 ...   \n",
      "749  [[0.0, 0.0, 1.0], [0.0, 0.0, 0.0], [0.0, 0.0, ...   \n",
      "750  [[1.0, 0.0, 0.0], [1.0, 0.0, 0.0], [1.0, 0.0, ...   \n",
      "751  [[1.0, 0.0, 0.0], [1.0, 0.0, 0.0], [1.0, 0.0, ...   \n",
      "752  [[1.0, 0.0, 0.0], [1.0, 0.0, 0.0], [0.0, 1.0, ...   \n",
      "753  [[0.0, 0.0, 0.0], [0.0, 0.0, 1.0], [0.0, 0.0, ...   \n",
      "\n",
      "                                                  rasa label    source  \n",
      "0    [0.041, 0.504, 0.07, 0.776, 0.157, 0.674, 0.58...     0     civic  \n",
      "1    [0.565, 0.357, 0.095, 0.333, 0.576, 0.542, 0.0...     0     civic  \n",
      "2    [0.157, 0.005, 0.0, 0.0, 0.0, 0.011, 0.0, 0.0,...     0     civic  \n",
      "3    [0.797, 0.818, 0.817, 0.677, 0.741, 0.635, 0.8...     1     civic  \n",
      "4    [0.596, 0.032, 0.004, 0.15, 0.395, 0.039, 0.00...     1     civic  \n",
      "..                                                 ...   ...       ...  \n",
      "749  [0.317, 0.269, 0.15, 0.009, 0.005, 0.437, 0.13...     1  pharmgkb  \n",
      "750  [0.284, 0.231, 0.025, 0.284, 0.316, 0.807, 0.7...     1  pharmgkb  \n",
      "751  [0.006, 0.195, 0.439, 0.11, 0.0, 0.258, 0.564,...     1  pharmgkb  \n",
      "752  [0.101, 0.01, 0.162, 0.32, 0.263, 0.517, 0.154...     1  pharmgkb  \n",
      "753  [0.221, 0.063, 0.479, 0.173, 0.125, 0.453, 0.3...     1  pharmgkb  \n",
      "\n",
      "[754 rows x 18 columns]\n"
     ]
    }
   ],
   "source": [
    "print(dataset_feature)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.9.12 64-bit",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "6fff98fc3b3d81bd655c2cc48858186e4d9e2db7b515bf1c3221888f12a62f87"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
