{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Variant & clinical annotations overview"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Part 1: pharmgkb variant overview"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       Study Parameters ID  Variant Annotation ID     Study Type  Study Cases  \\\n",
      "0               1449169927             1449169911   case/control         15.0   \n",
      "1                982022171              982022165   case/control         99.0   \n",
      "2               1451833880             1451833860         cohort          1.0   \n",
      "3                982022155              982022148   case/control         99.0   \n",
      "4               1451283504             1451283480  meta-analysis          NaN   \n",
      "...                    ...                    ...            ...          ...   \n",
      "32021           1451293460             1451293440    case series         39.0   \n",
      "32022           1451293520             1451293500    case series         39.0   \n",
      "32023           1451293504             1451293521    case series         39.0   \n",
      "32024           1451293420             1451293404    case series         39.0   \n",
      "32025           1451566940             1451566922            NaN          NaN   \n",
      "\n",
      "       Study Controls                                    Characteristics  \\\n",
      "0                50.0                SCAR, MPE, SJS (cases and controls)   \n",
      "1                99.0  Immunocompromised patients were excluded. Pers...   \n",
      "2                 NaN                                                NaN   \n",
      "3                99.0  Immunocompromised patients were excluded. Pers...   \n",
      "4                 NaN  meta-analysis of 27 studies, median minor alle...   \n",
      "...               ...                                                ...   \n",
      "32021             NaN  39 patients with genotype–phenotype discrepanc...   \n",
      "32022             NaN  39 patients with genotype–phenotype discrepanc...   \n",
      "32023             NaN  39 patients with genotype–phenotype discrepanc...   \n",
      "32024             NaN  39 patients with genotype–phenotype discrepanc...   \n",
      "32025             NaN                                                NaN   \n",
      "\n",
      "      Characteristics Type  Frequency In Cases Allele Of Frequency In Cases  \\\n",
      "0                  Disease               0.060                    *35:08:01   \n",
      "1             Study Cohort               0.000                            T   \n",
      "2                      NaN                 NaN                          NaN   \n",
      "3             Study Cohort               0.252                            A   \n",
      "4             Study Cohort               0.350                            A   \n",
      "...                    ...                 ...                          ...   \n",
      "32021         Study Cohort                 NaN                          NaN   \n",
      "32022         Study Cohort                 NaN                          NaN   \n",
      "32023         Study Cohort                 NaN                          NaN   \n",
      "32024         Study Cohort                 NaN                          NaN   \n",
      "32025              Unknown                 NaN                          NaN   \n",
      "\n",
      "       Frequency In Controls Allele Of Frequency In Controls  P Value  \\\n",
      "0                      0.001                       *35:08:01  = 0.231   \n",
      "1                      0.005                               T      NaN   \n",
      "2                        NaN                             NaN      NaN   \n",
      "3                      0.308                               A   = 0.46   \n",
      "4                        NaN                             NaN  < 0.001   \n",
      "...                      ...                             ...      ...   \n",
      "32021                    NaN                             NaN      NaN   \n",
      "32022                    NaN                             NaN      NaN   \n",
      "32023                    NaN                             NaN      NaN   \n",
      "32024                    NaN                             NaN      NaN   \n",
      "32025                    NaN                             NaN   < 0.01   \n",
      "\n",
      "      Ratio Stat Type  Ratio Stat  Confidence Interval Start  \\\n",
      "0                  OR       10.45                        0.4   \n",
      "1                  OR         NaN                        NaN   \n",
      "2                 NaN         NaN                        NaN   \n",
      "3                  OR        0.74                        0.4   \n",
      "4                 NaN         NaN                        NaN   \n",
      "...               ...         ...                        ...   \n",
      "32021             NaN         NaN                        NaN   \n",
      "32022             NaN         NaN                        NaN   \n",
      "32023             NaN         NaN                        NaN   \n",
      "32024             NaN         NaN                        NaN   \n",
      "32025             NaN         NaN                        NaN   \n",
      "\n",
      "       Confidence Interval Stop  \\\n",
      "0                        270.41   \n",
      "1                           NaN   \n",
      "2                           NaN   \n",
      "3                          1.38   \n",
      "4                           NaN   \n",
      "...                         ...   \n",
      "32021                       NaN   \n",
      "32022                       NaN   \n",
      "32023                       NaN   \n",
      "32024                       NaN   \n",
      "32025                       NaN   \n",
      "\n",
      "                                  Biogeographical Groups  \n",
      "0                                             East Asian  \n",
      "1      Multiple groups, Majority Caucasian, several N...  \n",
      "2                                                Unknown  \n",
      "3      Multiple groups, Majority Caucasian, several N...  \n",
      "4                                        Multiple groups  \n",
      "...                                                  ...  \n",
      "32021                                            Unknown  \n",
      "32022                                            Unknown  \n",
      "32023                                            Unknown  \n",
      "32024                                            Unknown  \n",
      "32025                                            Unknown  \n",
      "\n",
      "[32026 rows x 17 columns]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Study Parameters ID</th>\n",
       "      <th>Variant Annotation ID</th>\n",
       "      <th>Study Type</th>\n",
       "      <th>Study Cases</th>\n",
       "      <th>Study Controls</th>\n",
       "      <th>Characteristics</th>\n",
       "      <th>Characteristics Type</th>\n",
       "      <th>Frequency In Cases</th>\n",
       "      <th>Allele Of Frequency In Cases</th>\n",
       "      <th>Frequency In Controls</th>\n",
       "      <th>Allele Of Frequency In Controls</th>\n",
       "      <th>P Value</th>\n",
       "      <th>Ratio Stat Type</th>\n",
       "      <th>Ratio Stat</th>\n",
       "      <th>Confidence Interval Start</th>\n",
       "      <th>Confidence Interval Stop</th>\n",
       "      <th>Biogeographical Groups</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1449169927</td>\n",
       "      <td>1449169911</td>\n",
       "      <td>case/control</td>\n",
       "      <td>15.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>SCAR, MPE, SJS (cases and controls)</td>\n",
       "      <td>Disease</td>\n",
       "      <td>0.060</td>\n",
       "      <td>*35:08:01</td>\n",
       "      <td>0.001</td>\n",
       "      <td>*35:08:01</td>\n",
       "      <td>= 0.231</td>\n",
       "      <td>OR</td>\n",
       "      <td>10.45</td>\n",
       "      <td>0.4</td>\n",
       "      <td>270.41</td>\n",
       "      <td>East Asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>982022171</td>\n",
       "      <td>982022165</td>\n",
       "      <td>case/control</td>\n",
       "      <td>99.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>Immunocompromised patients were excluded. Pers...</td>\n",
       "      <td>Study Cohort</td>\n",
       "      <td>0.000</td>\n",
       "      <td>T</td>\n",
       "      <td>0.005</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Multiple groups, Majority Caucasian, several N...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1451833880</td>\n",
       "      <td>1451833860</td>\n",
       "      <td>cohort</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>982022155</td>\n",
       "      <td>982022148</td>\n",
       "      <td>case/control</td>\n",
       "      <td>99.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>Immunocompromised patients were excluded. Pers...</td>\n",
       "      <td>Study Cohort</td>\n",
       "      <td>0.252</td>\n",
       "      <td>A</td>\n",
       "      <td>0.308</td>\n",
       "      <td>A</td>\n",
       "      <td>= 0.46</td>\n",
       "      <td>OR</td>\n",
       "      <td>0.74</td>\n",
       "      <td>0.4</td>\n",
       "      <td>1.38</td>\n",
       "      <td>Multiple groups, Majority Caucasian, several N...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1451283504</td>\n",
       "      <td>1451283480</td>\n",
       "      <td>meta-analysis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>meta-analysis of 27 studies, median minor alle...</td>\n",
       "      <td>Study Cohort</td>\n",
       "      <td>0.350</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>&lt; 0.001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Multiple groups</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Study Parameters ID  Variant Annotation ID     Study Type  Study Cases  \\\n",
       "0           1449169927             1449169911   case/control         15.0   \n",
       "1            982022171              982022165   case/control         99.0   \n",
       "2           1451833880             1451833860         cohort          1.0   \n",
       "3            982022155              982022148   case/control         99.0   \n",
       "4           1451283504             1451283480  meta-analysis          NaN   \n",
       "\n",
       "   Study Controls                                    Characteristics  \\\n",
       "0            50.0                SCAR, MPE, SJS (cases and controls)   \n",
       "1            99.0  Immunocompromised patients were excluded. Pers...   \n",
       "2             NaN                                                NaN   \n",
       "3            99.0  Immunocompromised patients were excluded. Pers...   \n",
       "4             NaN  meta-analysis of 27 studies, median minor alle...   \n",
       "\n",
       "  Characteristics Type  Frequency In Cases Allele Of Frequency In Cases  \\\n",
       "0              Disease               0.060                    *35:08:01   \n",
       "1         Study Cohort               0.000                            T   \n",
       "2                  NaN                 NaN                          NaN   \n",
       "3         Study Cohort               0.252                            A   \n",
       "4         Study Cohort               0.350                            A   \n",
       "\n",
       "   Frequency In Controls Allele Of Frequency In Controls  P Value  \\\n",
       "0                  0.001                       *35:08:01  = 0.231   \n",
       "1                  0.005                               T      NaN   \n",
       "2                    NaN                             NaN      NaN   \n",
       "3                  0.308                               A   = 0.46   \n",
       "4                    NaN                             NaN  < 0.001   \n",
       "\n",
       "  Ratio Stat Type  Ratio Stat  Confidence Interval Start  \\\n",
       "0              OR       10.45                        0.4   \n",
       "1              OR         NaN                        NaN   \n",
       "2             NaN         NaN                        NaN   \n",
       "3              OR        0.74                        0.4   \n",
       "4             NaN         NaN                        NaN   \n",
       "\n",
       "   Confidence Interval Stop                             Biogeographical Groups  \n",
       "0                    270.41                                         East Asian  \n",
       "1                       NaN  Multiple groups, Majority Caucasian, several N...  \n",
       "2                       NaN                                            Unknown  \n",
       "3                      1.38  Multiple groups, Majority Caucasian, several N...  \n",
       "4                       NaN                                    Multiple groups  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Contains information about the study population size, \n",
    "# biogeographical group and statistics for the variant annotations; \n",
    "# this file is cross-referenced against the 3 variant annotation files\n",
    "study_parameter_df = pd.read_csv(\"../datasets/rawData-21-Dec-2022/pharmgkb/variantAnnotations/study_parameters.tsv\",sep='\\t')\n",
    "print(study_parameter_df)\n",
    "study_parameter_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       Variant Annotation ID                Variant/Haplotypes    Gene  \\\n",
      "0                 1449169911                    HLA-B*35:08:01   HLA-B   \n",
      "1                  982022165                        rs45607939    NAT2   \n",
      "2                 1451833860  CYP3A poor metabolizer phenotype   CYP3A   \n",
      "3                  982022148                         rs1799930    NAT2   \n",
      "4                 1451283480                        rs16969968  CHRNA5   \n",
      "...                      ...                               ...     ...   \n",
      "12820              827695506                         rs3745274  CYP2B6   \n",
      "12821             1447984440                         rs2472677   NR1I2   \n",
      "12822             1444936025                         rs2654754    DRD3   \n",
      "12823             1450807601                         rs2472677   NR1I2   \n",
      "12824             1450807610                         rs2472677   NR1I2   \n",
      "\n",
      "                                             Drug(s)      PMID  \\\n",
      "0                                        lamotrigine  29238301   \n",
      "1                    sulfamethoxazole / trimethoprim  22850190   \n",
      "2                                         loperamide  35815036   \n",
      "3                    sulfamethoxazole / trimethoprim  22850190   \n",
      "4                                                NaN  22071378   \n",
      "...                                              ...       ...   \n",
      "12820                                      efavirenz  16267764   \n",
      "12821                                     atazanavir  26857335   \n",
      "12822                                         heroin  25521358   \n",
      "12823  efavirenz,\"ethambutol\",\"isoniazid\",\"rifampin\"  30779340   \n",
      "12824  efavirenz,\"ethambutol\",\"isoniazid\",\"rifampin\"  30779340   \n",
      "\n",
      "      Phenotype Category Significance  \\\n",
      "0               Toxicity           no   \n",
      "1               Toxicity           no   \n",
      "2               Toxicity   not stated   \n",
      "3               Toxicity           no   \n",
      "4                  Other          yes   \n",
      "...                  ...          ...   \n",
      "12820           Efficacy           no   \n",
      "12821           Efficacy           no   \n",
      "12822              Other           no   \n",
      "12823           Toxicity          yes   \n",
      "12824           Toxicity          yes   \n",
      "\n",
      "                                                   Notes  \\\n",
      "0      The allele was not significant when comparing ...   \n",
      "1      Minor allele frequencies were compared between...   \n",
      "2      A patient with opioid use disorder showed slow...   \n",
      "3      Minor allele frequencies were compared between...   \n",
      "4      this was from meta-analysis of 27 studies but ...   \n",
      "...                                                  ...   \n",
      "12820                                                NaN   \n",
      "12821  The study authors investigated whether the SNP...   \n",
      "12822  Patients with the AG or GG genotype had an inc...   \n",
      "12823  There was a significantly higher probability o...   \n",
      "12824  This variant is associated with worsening peri...   \n",
      "\n",
      "                                                Sentence    Alleles  \\\n",
      "0      HLA-B *35:08:01 is not associated with likelih...  *35:08:01   \n",
      "1      Allele T is not associated with increased risk...          T   \n",
      "2      CYP3A poor metabolizer phenotype is associated...        NaN   \n",
      "3      Allele A is not associated with increased risk...          A   \n",
      "4      Allele A is associated with increased severity...          A   \n",
      "...                                                  ...        ...   \n",
      "12820  Genotype TT is not associated with increased l...         TT   \n",
      "12821  Allele T is not associated with likelihood of ...          T   \n",
      "12822  Genotypes AG + GG is associated with increased...    AG + GG   \n",
      "12823  Genotype TT is associated with increased likel...         TT   \n",
      "12824  Genotype TT is associated with increased sever...         TT   \n",
      "\n",
      "      Specialty Population  \n",
      "0                      NaN  \n",
      "1                      NaN  \n",
      "2                      NaN  \n",
      "3                      NaN  \n",
      "4                      NaN  \n",
      "...                    ...  \n",
      "12820                  NaN  \n",
      "12821                  NaN  \n",
      "12822                  NaN  \n",
      "12823                  NaN  \n",
      "12824                  NaN  \n",
      "\n",
      "[12825 rows x 11 columns]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant Annotation ID</th>\n",
       "      <th>Variant/Haplotypes</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Drug(s)</th>\n",
       "      <th>PMID</th>\n",
       "      <th>Phenotype Category</th>\n",
       "      <th>Significance</th>\n",
       "      <th>Notes</th>\n",
       "      <th>Sentence</th>\n",
       "      <th>Alleles</th>\n",
       "      <th>Specialty Population</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1449169911</td>\n",
       "      <td>HLA-B*35:08:01</td>\n",
       "      <td>HLA-B</td>\n",
       "      <td>lamotrigine</td>\n",
       "      <td>29238301</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>no</td>\n",
       "      <td>The allele was not significant when comparing ...</td>\n",
       "      <td>HLA-B *35:08:01 is not associated with likelih...</td>\n",
       "      <td>*35:08:01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>982022165</td>\n",
       "      <td>rs45607939</td>\n",
       "      <td>NAT2</td>\n",
       "      <td>sulfamethoxazole / trimethoprim</td>\n",
       "      <td>22850190</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>no</td>\n",
       "      <td>Minor allele frequencies were compared between...</td>\n",
       "      <td>Allele T is not associated with increased risk...</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1451833860</td>\n",
       "      <td>CYP3A poor metabolizer phenotype</td>\n",
       "      <td>CYP3A</td>\n",
       "      <td>loperamide</td>\n",
       "      <td>35815036</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>not stated</td>\n",
       "      <td>A patient with opioid use disorder showed slow...</td>\n",
       "      <td>CYP3A poor metabolizer phenotype is associated...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>982022148</td>\n",
       "      <td>rs1799930</td>\n",
       "      <td>NAT2</td>\n",
       "      <td>sulfamethoxazole / trimethoprim</td>\n",
       "      <td>22850190</td>\n",
       "      <td>Toxicity</td>\n",
       "      <td>no</td>\n",
       "      <td>Minor allele frequencies were compared between...</td>\n",
       "      <td>Allele A is not associated with increased risk...</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1451283480</td>\n",
       "      <td>rs16969968</td>\n",
       "      <td>CHRNA5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22071378</td>\n",
       "      <td>Other</td>\n",
       "      <td>yes</td>\n",
       "      <td>this was from meta-analysis of 27 studies but ...</td>\n",
       "      <td>Allele A is associated with increased severity...</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Variant Annotation ID                Variant/Haplotypes    Gene  \\\n",
       "0             1449169911                    HLA-B*35:08:01   HLA-B   \n",
       "1              982022165                        rs45607939    NAT2   \n",
       "2             1451833860  CYP3A poor metabolizer phenotype   CYP3A   \n",
       "3              982022148                         rs1799930    NAT2   \n",
       "4             1451283480                        rs16969968  CHRNA5   \n",
       "\n",
       "                           Drug(s)      PMID Phenotype Category Significance  \\\n",
       "0                      lamotrigine  29238301           Toxicity           no   \n",
       "1  sulfamethoxazole / trimethoprim  22850190           Toxicity           no   \n",
       "2                       loperamide  35815036           Toxicity   not stated   \n",
       "3  sulfamethoxazole / trimethoprim  22850190           Toxicity           no   \n",
       "4                              NaN  22071378              Other          yes   \n",
       "\n",
       "                                               Notes  \\\n",
       "0  The allele was not significant when comparing ...   \n",
       "1  Minor allele frequencies were compared between...   \n",
       "2  A patient with opioid use disorder showed slow...   \n",
       "3  Minor allele frequencies were compared between...   \n",
       "4  this was from meta-analysis of 27 studies but ...   \n",
       "\n",
       "                                            Sentence    Alleles  \\\n",
       "0  HLA-B *35:08:01 is not associated with likelih...  *35:08:01   \n",
       "1  Allele T is not associated with increased risk...          T   \n",
       "2  CYP3A poor metabolizer phenotype is associated...        NaN   \n",
       "3  Allele A is not associated with increased risk...          A   \n",
       "4  Allele A is associated with increased severity...          A   \n",
       "\n",
       "  Specialty Population  \n",
       "0                  NaN  \n",
       "1                  NaN  \n",
       "2                  NaN  \n",
       "3                  NaN  \n",
       "4                  NaN  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Contains associations in which the variant affects a phenotype, with or without drug information.\n",
    "var_pheno_ann_df = pd.read_csv(\"../datasets/rawData-21-Dec-2022/pharmgkb/variantAnnotations/var_pheno_ann.tsv\",sep='\\t', on_bad_lines='skip')\n",
    "print(var_pheno_ann_df)\n",
    "var_pheno_ann_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       Variant Annotation ID                      Variant/Haplotypes    Gene  \\\n",
      "0                 1451834452                     CYP3A4*1, CYP3A4*17  CYP3A4   \n",
      "1                 1451159680                               rs5031016  CYP2A6   \n",
      "2                 1183684657  CYP2D6 ultrarapid metabolizer genotype  CYP2D6   \n",
      "3                 1451306860                               CYP2C9*11  CYP2C9   \n",
      "4                 1448997750                     CYP2B6*1, CYP2B6*18  CYP2B6   \n",
      "...                      ...                                     ...     ...   \n",
      "11192             1448997209                               rs3745274  CYP2B6   \n",
      "11193             1448617505                               rs3745274  CYP2B6   \n",
      "11194             1448998426                               rs3745274  CYP2B6   \n",
      "11195             1448107499                               rs3745274  CYP2B6   \n",
      "11196             1450807575                               rs2472677   NR1I2   \n",
      "\n",
      "          Drug(s)      PMID     Phenotype Category Significance  \\\n",
      "0      nifedipine  15634941  Other,\"Metabolism/PK\"   not stated   \n",
      "1        warfarin  22248286                 Dosage           no   \n",
      "2        tramadol  18204346          Metabolism/PK          yes   \n",
      "3        warfarin  33350885                 Dosage   not stated   \n",
      "4       efavirenz  16495778          Metabolism/PK          yes   \n",
      "...           ...       ...                    ...          ...   \n",
      "11192   efavirenz  25882300          Metabolism/PK          yes   \n",
      "11193   efavirenz  28481785               Efficacy          yes   \n",
      "11194   efavirenz  15825040          Metabolism/PK   not stated   \n",
      "11195   efavirenz  26196596                 Dosage   not stated   \n",
      "11196   isoniazid  30779340          Metabolism/PK          yes   \n",
      "\n",
      "                                                   Notes  \\\n",
      "0      in vitro expression of the recombinant CYP3A4*...   \n",
      "1      No association was found between this variant ...   \n",
      "2      Median (+)R,R-tramadol area under the curve wa...   \n",
      "3      This case suggests that CYP2C9 *11/*11 carrier...   \n",
      "4      Please note that in the paper the allele was r...   \n",
      "...                                                  ...   \n",
      "11192  CYP2B6 516G>T was independently associated wit...   \n",
      "11193  CYPB2B6 516 T-allele was protective against la...   \n",
      "11194                                                NaN   \n",
      "11195  This study provided efavirenz dosing advice fo...   \n",
      "11196  This variant is associated with increased AUC ...   \n",
      "\n",
      "                                                Sentence  Alleles  \\\n",
      "0      CYP3A4 *17 is associated with decreased metabo...      *17   \n",
      "1      Allele G is not associated with increased dose...        G   \n",
      "2      CYP2D6 ultra-metabolizer genotype is associate...      NaN   \n",
      "3      CYP2C9 *11/*11 is associated with decreased do...  *11/*11   \n",
      "4      CYP2B6 *1/*18 is associated with increased con...   *1/*18   \n",
      "...                                                  ...      ...   \n",
      "11192  Genotypes GT + TT are associated with increase...  GT + TT   \n",
      "11193  Genotypes GT + TT are associated with decrease...  GT + TT   \n",
      "11194  Genotypes GT + TT are associated with increase...  GT + TT   \n",
      "11195  Genotype GG is associated with increased dose ...       GG   \n",
      "11196  Genotype TT is associated with increased expos...       TT   \n",
      "\n",
      "      Specialty Population  \n",
      "0                      NaN  \n",
      "1                      NaN  \n",
      "2                      NaN  \n",
      "3                      NaN  \n",
      "4                      NaN  \n",
      "...                    ...  \n",
      "11192                  NaN  \n",
      "11193                  NaN  \n",
      "11194                  NaN  \n",
      "11195                  NaN  \n",
      "11196                  NaN  \n",
      "\n",
      "[11197 rows x 11 columns]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant Annotation ID</th>\n",
       "      <th>Variant/Haplotypes</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Drug(s)</th>\n",
       "      <th>PMID</th>\n",
       "      <th>Phenotype Category</th>\n",
       "      <th>Significance</th>\n",
       "      <th>Notes</th>\n",
       "      <th>Sentence</th>\n",
       "      <th>Alleles</th>\n",
       "      <th>Specialty Population</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1451834452</td>\n",
       "      <td>CYP3A4*1, CYP3A4*17</td>\n",
       "      <td>CYP3A4</td>\n",
       "      <td>nifedipine</td>\n",
       "      <td>15634941</td>\n",
       "      <td>Other,\"Metabolism/PK\"</td>\n",
       "      <td>not stated</td>\n",
       "      <td>in vitro expression of the recombinant CYP3A4*...</td>\n",
       "      <td>CYP3A4 *17 is associated with decreased metabo...</td>\n",
       "      <td>*17</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1451159680</td>\n",
       "      <td>rs5031016</td>\n",
       "      <td>CYP2A6</td>\n",
       "      <td>warfarin</td>\n",
       "      <td>22248286</td>\n",
       "      <td>Dosage</td>\n",
       "      <td>no</td>\n",
       "      <td>No association was found between this variant ...</td>\n",
       "      <td>Allele G is not associated with increased dose...</td>\n",
       "      <td>G</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1183684657</td>\n",
       "      <td>CYP2D6 ultrarapid metabolizer genotype</td>\n",
       "      <td>CYP2D6</td>\n",
       "      <td>tramadol</td>\n",
       "      <td>18204346</td>\n",
       "      <td>Metabolism/PK</td>\n",
       "      <td>yes</td>\n",
       "      <td>Median (+)R,R-tramadol area under the curve wa...</td>\n",
       "      <td>CYP2D6 ultra-metabolizer genotype is associate...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1451306860</td>\n",
       "      <td>CYP2C9*11</td>\n",
       "      <td>CYP2C9</td>\n",
       "      <td>warfarin</td>\n",
       "      <td>33350885</td>\n",
       "      <td>Dosage</td>\n",
       "      <td>not stated</td>\n",
       "      <td>This case suggests that CYP2C9 *11/*11 carrier...</td>\n",
       "      <td>CYP2C9 *11/*11 is associated with decreased do...</td>\n",
       "      <td>*11/*11</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1448997750</td>\n",
       "      <td>CYP2B6*1, CYP2B6*18</td>\n",
       "      <td>CYP2B6</td>\n",
       "      <td>efavirenz</td>\n",
       "      <td>16495778</td>\n",
       "      <td>Metabolism/PK</td>\n",
       "      <td>yes</td>\n",
       "      <td>Please note that in the paper the allele was r...</td>\n",
       "      <td>CYP2B6 *1/*18 is associated with increased con...</td>\n",
       "      <td>*1/*18</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Variant Annotation ID                      Variant/Haplotypes    Gene  \\\n",
       "0             1451834452                     CYP3A4*1, CYP3A4*17  CYP3A4   \n",
       "1             1451159680                               rs5031016  CYP2A6   \n",
       "2             1183684657  CYP2D6 ultrarapid metabolizer genotype  CYP2D6   \n",
       "3             1451306860                               CYP2C9*11  CYP2C9   \n",
       "4             1448997750                     CYP2B6*1, CYP2B6*18  CYP2B6   \n",
       "\n",
       "      Drug(s)      PMID     Phenotype Category Significance  \\\n",
       "0  nifedipine  15634941  Other,\"Metabolism/PK\"   not stated   \n",
       "1    warfarin  22248286                 Dosage           no   \n",
       "2    tramadol  18204346          Metabolism/PK          yes   \n",
       "3    warfarin  33350885                 Dosage   not stated   \n",
       "4   efavirenz  16495778          Metabolism/PK          yes   \n",
       "\n",
       "                                               Notes  \\\n",
       "0  in vitro expression of the recombinant CYP3A4*...   \n",
       "1  No association was found between this variant ...   \n",
       "2  Median (+)R,R-tramadol area under the curve wa...   \n",
       "3  This case suggests that CYP2C9 *11/*11 carrier...   \n",
       "4  Please note that in the paper the allele was r...   \n",
       "\n",
       "                                            Sentence  Alleles  \\\n",
       "0  CYP3A4 *17 is associated with decreased metabo...      *17   \n",
       "1  Allele G is not associated with increased dose...        G   \n",
       "2  CYP2D6 ultra-metabolizer genotype is associate...      NaN   \n",
       "3  CYP2C9 *11/*11 is associated with decreased do...  *11/*11   \n",
       "4  CYP2B6 *1/*18 is associated with increased con...   *1/*18   \n",
       "\n",
       "  Specialty Population  \n",
       "0                  NaN  \n",
       "1                  NaN  \n",
       "2                  NaN  \n",
       "3                  NaN  \n",
       "4                  NaN  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Contains associations in which the variant affects a drug dose, response, metabolism, etc\n",
    "var_drug_ann_df = pd.read_csv(\"../datasets/rawData-21-Dec-2022/pharmgkb/variantAnnotations/var_drug_ann.tsv\",sep='\\t', on_bad_lines='skip')\n",
    "print(var_drug_ann_df)\n",
    "var_drug_ann_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      Variant Annotation ID  \\\n",
      "0                1451148445   \n",
      "1                1447814273   \n",
      "2                1447814277   \n",
      "3                1447990384   \n",
      "4                1448281185   \n",
      "...                     ...   \n",
      "1805             1451914664   \n",
      "1806             1451914660   \n",
      "1807             1451914680   \n",
      "1808             1451566922   \n",
      "1809             1451566980   \n",
      "\n",
      "                                     Variant/Haplotypes     Gene  \\\n",
      "0                                 CYP2C19*1, CYP2C19*17  CYP2C19   \n",
      "1                                             rs9923231   VKORC1   \n",
      "2                                            rs56314408   VKORC1   \n",
      "3                                             rs1065852   CYP2D6   \n",
      "4                                    CYP2B6*1, CYP2B6*6   CYP2B6   \n",
      "...                                                 ...      ...   \n",
      "1805                                            rs11615    ERCC1   \n",
      "1806                                            rs13181    ERCC2   \n",
      "1807                                             rs1695    GSTP1   \n",
      "1808  CYP3A4*1, CYP3A4*2, CYP3A4*3, CYP3A4*4, CYP3A4...   CYP3A4   \n",
      "1809  CYP3A4*1, CYP3A4*6, CYP3A4*17, CYP3A4*20, CYP3...   CYP3A4   \n",
      "\n",
      "            Drug(s)      PMID Phenotype Category Significance  \\\n",
      "0     normeperidine  30902024                NaN   not stated   \n",
      "1               NaN  26847243              Other           no   \n",
      "2               NaN  26847243              Other          yes   \n",
      "3         bufuralol   2211621      Metabolism/PK   not stated   \n",
      "4         bupropion  27439448           Efficacy          yes   \n",
      "...             ...       ...                ...          ...   \n",
      "1805      cisplatin  36233089           Toxicity          yes   \n",
      "1806      cisplatin  36233089           Toxicity          yes   \n",
      "1807      cisplatin  36233089           Toxicity          yes   \n",
      "1808      oxycodone  33393779      Metabolism/PK          yes   \n",
      "1809      oxycodone  33393779      Metabolism/PK          yes   \n",
      "\n",
      "                                                  Notes  \\\n",
      "0     In other in vitro experiments, normeperidine f...   \n",
      "1                                                   NaN   \n",
      "2     In the European population, this SNPs is in hi...   \n",
      "3     In vitro experiments showed a significant decr...   \n",
      "4     The ratio of hydroxybupropion versus bupropion...   \n",
      "...                                                 ...   \n",
      "1805  ERCC1 rs11615, which was heterozygous variant ...   \n",
      "1806  For ERCC2 rs13181, the genotype of the detecte...   \n",
      "1807  Ile105Val caused by the GSTP1 rs1695 variant a...   \n",
      "1808  These alleles showed decreased, but still dete...   \n",
      "1809  These alleles did not show any detectable enzy...   \n",
      "\n",
      "                                               Sentence  \\\n",
      "0     CYP2C19 *17/*17 is associated with increased f...   \n",
      "1     Allele T is not associated with transcription ...   \n",
      "2     Allele C is associated with increased transcri...   \n",
      "3     Allele A is associated with decreased activity...   \n",
      "4     CYP2B6 *1/*1 is associated with increased acti...   \n",
      "...                                                 ...   \n",
      "1805  Genotype GG is associated with decreased sensi...   \n",
      "1806  Genotype GG is associated with decreased sensi...   \n",
      "1807  Genotype AG is associated with decreased sensi...   \n",
      "1808  CYP3A4 *2 + *3 + *4+ *5 + *7 + *8 + *9 + *10 +...   \n",
      "1809  CYP3A4 *6 + *17 + *20 + *21 + *26 + *30 are as...   \n",
      "\n",
      "                                                Alleles Specialty Population  \n",
      "0                                               *17/*17                  NaN  \n",
      "1                                                     T                  NaN  \n",
      "2                                                     C                  NaN  \n",
      "3                                                     A                  NaN  \n",
      "4                                                 *1/*1                  NaN  \n",
      "...                                                 ...                  ...  \n",
      "1805                                                 GG                  NaN  \n",
      "1806                                                 GG                  NaN  \n",
      "1807                                                 AG                  NaN  \n",
      "1808  *2 + *3 + *4+ *5 + *7 + *8 + *9 + *10 + *11 + ...                  NaN  \n",
      "1809                   *6 + *17 + *20 + *21 + *26 + *30                  NaN  \n",
      "\n",
      "[1810 rows x 11 columns]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant Annotation ID</th>\n",
       "      <th>Variant/Haplotypes</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Drug(s)</th>\n",
       "      <th>PMID</th>\n",
       "      <th>Phenotype Category</th>\n",
       "      <th>Significance</th>\n",
       "      <th>Notes</th>\n",
       "      <th>Sentence</th>\n",
       "      <th>Alleles</th>\n",
       "      <th>Specialty Population</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1451148445</td>\n",
       "      <td>CYP2C19*1, CYP2C19*17</td>\n",
       "      <td>CYP2C19</td>\n",
       "      <td>normeperidine</td>\n",
       "      <td>30902024</td>\n",
       "      <td>NaN</td>\n",
       "      <td>not stated</td>\n",
       "      <td>In other in vitro experiments, normeperidine f...</td>\n",
       "      <td>CYP2C19 *17/*17 is associated with increased f...</td>\n",
       "      <td>*17/*17</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1447814273</td>\n",
       "      <td>rs9923231</td>\n",
       "      <td>VKORC1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26847243</td>\n",
       "      <td>Other</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele T is not associated with transcription ...</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1447814277</td>\n",
       "      <td>rs56314408</td>\n",
       "      <td>VKORC1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26847243</td>\n",
       "      <td>Other</td>\n",
       "      <td>yes</td>\n",
       "      <td>In the European population, this SNPs is in hi...</td>\n",
       "      <td>Allele C is associated with increased transcri...</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1447990384</td>\n",
       "      <td>rs1065852</td>\n",
       "      <td>CYP2D6</td>\n",
       "      <td>bufuralol</td>\n",
       "      <td>2211621</td>\n",
       "      <td>Metabolism/PK</td>\n",
       "      <td>not stated</td>\n",
       "      <td>In vitro experiments showed a significant decr...</td>\n",
       "      <td>Allele A is associated with decreased activity...</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1448281185</td>\n",
       "      <td>CYP2B6*1, CYP2B6*6</td>\n",
       "      <td>CYP2B6</td>\n",
       "      <td>bupropion</td>\n",
       "      <td>27439448</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>yes</td>\n",
       "      <td>The ratio of hydroxybupropion versus bupropion...</td>\n",
       "      <td>CYP2B6 *1/*1 is associated with increased acti...</td>\n",
       "      <td>*1/*1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Variant Annotation ID     Variant/Haplotypes     Gene        Drug(s)  \\\n",
       "0             1451148445  CYP2C19*1, CYP2C19*17  CYP2C19  normeperidine   \n",
       "1             1447814273              rs9923231   VKORC1            NaN   \n",
       "2             1447814277             rs56314408   VKORC1            NaN   \n",
       "3             1447990384              rs1065852   CYP2D6      bufuralol   \n",
       "4             1448281185     CYP2B6*1, CYP2B6*6   CYP2B6      bupropion   \n",
       "\n",
       "       PMID Phenotype Category Significance  \\\n",
       "0  30902024                NaN   not stated   \n",
       "1  26847243              Other           no   \n",
       "2  26847243              Other          yes   \n",
       "3   2211621      Metabolism/PK   not stated   \n",
       "4  27439448           Efficacy          yes   \n",
       "\n",
       "                                               Notes  \\\n",
       "0  In other in vitro experiments, normeperidine f...   \n",
       "1                                                NaN   \n",
       "2  In the European population, this SNPs is in hi...   \n",
       "3  In vitro experiments showed a significant decr...   \n",
       "4  The ratio of hydroxybupropion versus bupropion...   \n",
       "\n",
       "                                            Sentence  Alleles  \\\n",
       "0  CYP2C19 *17/*17 is associated with increased f...  *17/*17   \n",
       "1  Allele T is not associated with transcription ...        T   \n",
       "2  Allele C is associated with increased transcri...        C   \n",
       "3  Allele A is associated with decreased activity...        A   \n",
       "4  CYP2B6 *1/*1 is associated with increased acti...    *1/*1   \n",
       "\n",
       "  Specialty Population  \n",
       "0                  NaN  \n",
       "1                  NaN  \n",
       "2                  NaN  \n",
       "3                  NaN  \n",
       "4                  NaN  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Contains in vitro and functional analysis-type associations\n",
    "var_fa_ann_df = pd.read_csv(\"../datasets/rawData-21-Dec-2022/pharmgkb/variantAnnotations/var_fa_ann.tsv\",sep='\\t', on_bad_lines='skip')\n",
    "print(var_fa_ann_df)\n",
    "var_fa_ann_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Part 2: pharmgkb clinical overview"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      Clinical Annotation ID  \\\n",
      "0                  981755803   \n",
      "1                 1449311190   \n",
      "2                  981204774   \n",
      "3                 1449191690   \n",
      "4                 1449191746   \n",
      "...                      ...   \n",
      "5028              1451553326   \n",
      "5029              1451553595   \n",
      "5030              1451553580   \n",
      "5031              1451566760   \n",
      "5032              1451567040   \n",
      "\n",
      "                                     Variant/Haplotypes           Gene  \\\n",
      "0                                            rs75527207           CFTR   \n",
      "1                                             rs4149056        SLCO1B1   \n",
      "2                                             rs1799971          OPRM1   \n",
      "3                                           rs141033578           CFTR   \n",
      "4                                            rs78769542           CFTR   \n",
      "...                                                 ...            ...   \n",
      "5028                                         rs45445694  C18orf56;TYMS   \n",
      "5029                                          rs1045642          ABCB1   \n",
      "5030                                          rs1045642          ABCB1   \n",
      "5031                                          rs1800497     ANKK1;DRD2   \n",
      "5032  CYP3A4*1, CYP3A4*2, CYP3A4*3, CYP3A4*4, CYP3A4...         CYP3A4   \n",
      "\n",
      "     Level of Evidence Level Override           Level Modifiers    Score  \\\n",
      "0                   1A            NaN  Rare Variant; Tier 1 VIP  234.875   \n",
      "1                    3            NaN                Tier 1 VIP    2.000   \n",
      "2                    4            NaN                       NaN   -2.000   \n",
      "3                   1A            NaN  Rare Variant; Tier 1 VIP  200.000   \n",
      "4                   1A            NaN  Rare Variant; Tier 1 VIP  200.000   \n",
      "...                ...            ...                       ...      ...   \n",
      "5028                 4            NaN                Tier 1 VIP   -1.000   \n",
      "5029                 4            NaN                Tier 1 VIP   -1.250   \n",
      "5030                 3            NaN                Tier 1 VIP    0.000   \n",
      "5031                 3            NaN                Tier 1 VIP    1.500   \n",
      "5032                 3            NaN                Tier 1 VIP    2.500   \n",
      "\n",
      "     Phenotype Category  PMID Count  Evidence Count  \\\n",
      "0              Efficacy          28              30   \n",
      "1                Dosage           1               1   \n",
      "2              Efficacy           2               3   \n",
      "3              Efficacy           1               3   \n",
      "4              Efficacy           1               3   \n",
      "...                 ...         ...             ...   \n",
      "5028      Metabolism/PK           2               2   \n",
      "5029           Efficacy           2               3   \n",
      "5030      Metabolism/PK           5               5   \n",
      "5031           Efficacy           1               1   \n",
      "5032      Metabolism/PK           2               3   \n",
      "\n",
      "                                         Drug(s)  \\\n",
      "0                                      ivacaftor   \n",
      "1                    mercaptopurine;methotrexate   \n",
      "2     Drugs used in nicotine dependence;nicotine   \n",
      "3                                      ivacaftor   \n",
      "4                                      ivacaftor   \n",
      "...                                          ...   \n",
      "5028                                methotrexate   \n",
      "5029                                methotrexate   \n",
      "5030                                methotrexate   \n",
      "5031                        bupropion;naltrexone   \n",
      "5032                                   oxycodone   \n",
      "\n",
      "                                           Phenotype(s)  \\\n",
      "0                                       Cystic Fibrosis   \n",
      "1        Precursor Cell Lymphoblastic Leukemia-Lymphoma   \n",
      "2                                  Tobacco Use Disorder   \n",
      "3                                       Cystic Fibrosis   \n",
      "4                                       Cystic Fibrosis   \n",
      "...                                                 ...   \n",
      "5028     Precursor Cell Lymphoblastic Leukemia-Lymphoma   \n",
      "5029     Precursor Cell Lymphoblastic Leukemia-Lymphoma   \n",
      "5030  Burkitt Lymphoma;Lymphoma, T-Cell;Precursor Ce...   \n",
      "5031                                            Obesity   \n",
      "5032                                                NaN   \n",
      "\n",
      "     Latest History Date (YYYY-MM-DD)  \\\n",
      "0                          2021-03-24   \n",
      "1                          2021-03-24   \n",
      "2                          2021-03-24   \n",
      "3                          2021-03-24   \n",
      "4                          2021-03-24   \n",
      "...                               ...   \n",
      "5028                       2021-10-21   \n",
      "5029                       2021-10-21   \n",
      "5030                       2021-12-13   \n",
      "5031                       2021-10-29   \n",
      "5032                       2022-01-13   \n",
      "\n",
      "                                                    URL Specialty Population  \n",
      "0     https://www.pharmgkb.org/clinicalAnnotation/98...            Pediatric  \n",
      "1     https://www.pharmgkb.org/clinicalAnnotation/14...            Pediatric  \n",
      "2     https://www.pharmgkb.org/clinicalAnnotation/98...                  NaN  \n",
      "3     https://www.pharmgkb.org/clinicalAnnotation/14...                  NaN  \n",
      "4     https://www.pharmgkb.org/clinicalAnnotation/14...                  NaN  \n",
      "...                                                 ...                  ...  \n",
      "5028  https://www.pharmgkb.org/clinicalAnnotation/14...            Pediatric  \n",
      "5029  https://www.pharmgkb.org/clinicalAnnotation/14...            Pediatric  \n",
      "5030  https://www.pharmgkb.org/clinicalAnnotation/14...            Pediatric  \n",
      "5031  https://www.pharmgkb.org/clinicalAnnotation/14...                  NaN  \n",
      "5032  https://www.pharmgkb.org/clinicalAnnotation/14...                  NaN  \n",
      "\n",
      "[5033 rows x 15 columns]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Clinical Annotation ID</th>\n",
       "      <th>Variant/Haplotypes</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Level of Evidence</th>\n",
       "      <th>Level Override</th>\n",
       "      <th>Level Modifiers</th>\n",
       "      <th>Score</th>\n",
       "      <th>Phenotype Category</th>\n",
       "      <th>PMID Count</th>\n",
       "      <th>Evidence Count</th>\n",
       "      <th>Drug(s)</th>\n",
       "      <th>Phenotype(s)</th>\n",
       "      <th>Latest History Date (YYYY-MM-DD)</th>\n",
       "      <th>URL</th>\n",
       "      <th>Specialty Population</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>981755803</td>\n",
       "      <td>rs75527207</td>\n",
       "      <td>CFTR</td>\n",
       "      <td>1A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rare Variant; Tier 1 VIP</td>\n",
       "      <td>234.875</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>28</td>\n",
       "      <td>30</td>\n",
       "      <td>ivacaftor</td>\n",
       "      <td>Cystic Fibrosis</td>\n",
       "      <td>2021-03-24</td>\n",
       "      <td>https://www.pharmgkb.org/clinicalAnnotation/98...</td>\n",
       "      <td>Pediatric</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1449311190</td>\n",
       "      <td>rs4149056</td>\n",
       "      <td>SLCO1B1</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Tier 1 VIP</td>\n",
       "      <td>2.000</td>\n",
       "      <td>Dosage</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>mercaptopurine;methotrexate</td>\n",
       "      <td>Precursor Cell Lymphoblastic Leukemia-Lymphoma</td>\n",
       "      <td>2021-03-24</td>\n",
       "      <td>https://www.pharmgkb.org/clinicalAnnotation/14...</td>\n",
       "      <td>Pediatric</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>981204774</td>\n",
       "      <td>rs1799971</td>\n",
       "      <td>OPRM1</td>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.000</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>Drugs used in nicotine dependence;nicotine</td>\n",
       "      <td>Tobacco Use Disorder</td>\n",
       "      <td>2021-03-24</td>\n",
       "      <td>https://www.pharmgkb.org/clinicalAnnotation/98...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1449191690</td>\n",
       "      <td>rs141033578</td>\n",
       "      <td>CFTR</td>\n",
       "      <td>1A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rare Variant; Tier 1 VIP</td>\n",
       "      <td>200.000</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>ivacaftor</td>\n",
       "      <td>Cystic Fibrosis</td>\n",
       "      <td>2021-03-24</td>\n",
       "      <td>https://www.pharmgkb.org/clinicalAnnotation/14...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1449191746</td>\n",
       "      <td>rs78769542</td>\n",
       "      <td>CFTR</td>\n",
       "      <td>1A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Rare Variant; Tier 1 VIP</td>\n",
       "      <td>200.000</td>\n",
       "      <td>Efficacy</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>ivacaftor</td>\n",
       "      <td>Cystic Fibrosis</td>\n",
       "      <td>2021-03-24</td>\n",
       "      <td>https://www.pharmgkb.org/clinicalAnnotation/14...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Clinical Annotation ID Variant/Haplotypes     Gene Level of Evidence  \\\n",
       "0               981755803         rs75527207     CFTR                1A   \n",
       "1              1449311190          rs4149056  SLCO1B1                 3   \n",
       "2               981204774          rs1799971    OPRM1                 4   \n",
       "3              1449191690        rs141033578     CFTR                1A   \n",
       "4              1449191746         rs78769542     CFTR                1A   \n",
       "\n",
       "  Level Override           Level Modifiers    Score Phenotype Category  \\\n",
       "0            NaN  Rare Variant; Tier 1 VIP  234.875           Efficacy   \n",
       "1            NaN                Tier 1 VIP    2.000             Dosage   \n",
       "2            NaN                       NaN   -2.000           Efficacy   \n",
       "3            NaN  Rare Variant; Tier 1 VIP  200.000           Efficacy   \n",
       "4            NaN  Rare Variant; Tier 1 VIP  200.000           Efficacy   \n",
       "\n",
       "   PMID Count  Evidence Count                                     Drug(s)  \\\n",
       "0          28              30                                   ivacaftor   \n",
       "1           1               1                 mercaptopurine;methotrexate   \n",
       "2           2               3  Drugs used in nicotine dependence;nicotine   \n",
       "3           1               3                                   ivacaftor   \n",
       "4           1               3                                   ivacaftor   \n",
       "\n",
       "                                     Phenotype(s)  \\\n",
       "0                                 Cystic Fibrosis   \n",
       "1  Precursor Cell Lymphoblastic Leukemia-Lymphoma   \n",
       "2                            Tobacco Use Disorder   \n",
       "3                                 Cystic Fibrosis   \n",
       "4                                 Cystic Fibrosis   \n",
       "\n",
       "  Latest History Date (YYYY-MM-DD)  \\\n",
       "0                       2021-03-24   \n",
       "1                       2021-03-24   \n",
       "2                       2021-03-24   \n",
       "3                       2021-03-24   \n",
       "4                       2021-03-24   \n",
       "\n",
       "                                                 URL Specialty Population  \n",
       "0  https://www.pharmgkb.org/clinicalAnnotation/98...            Pediatric  \n",
       "1  https://www.pharmgkb.org/clinicalAnnotation/14...            Pediatric  \n",
       "2  https://www.pharmgkb.org/clinicalAnnotation/98...                  NaN  \n",
       "3  https://www.pharmgkb.org/clinicalAnnotation/14...                  NaN  \n",
       "4  https://www.pharmgkb.org/clinicalAnnotation/14...                  NaN  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Contains all of the meta-data about each clinical annotation\n",
    "clinical_annotations_df = pd.read_csv(\"../datasets/rawData-21-Dec-2022/pharmgkb/clinicalAnnotations/clinical_annotations.tsv\",sep='\\t', on_bad_lines='skip')\n",
    "print(clinical_annotations_df)\n",
    "clinical_annotations_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       Clinical Annotation ID Genotype/Allele  \\\n",
      "0                   981755803              AA   \n",
      "1                   981755803              AG   \n",
      "2                   981755803              GG   \n",
      "3                  1449311190              CC   \n",
      "4                  1449311190              CT   \n",
      "...                       ...             ...   \n",
      "15540              1451567040             *30   \n",
      "15541              1451567040             *31   \n",
      "15542              1451567040             *32   \n",
      "15543              1451567040             *33   \n",
      "15544              1451567040             *34   \n",
      "\n",
      "                                         Annotation Text Allele Function  \n",
      "0      Patients with the rs75527207 AA genotype (two ...             NaN  \n",
      "1      Patients with the rs75527207 AG genotype (one ...             NaN  \n",
      "2      Patients with the rs75527207 GG genotype (do n...             NaN  \n",
      "3      Patients with the CC genotype and Precursor Ce...             NaN  \n",
      "4      Patients with the CT genotype and Precursor Ce...             NaN  \n",
      "...                                                  ...             ...  \n",
      "15540  Patients carrying the CYP3A4*30 allele may hav...             NaN  \n",
      "15541  Patients carrying the CYP3A4*31 allele may hav...             NaN  \n",
      "15542  Patients carrying the CYP3A4*32 allele may hav...             NaN  \n",
      "15543  Patients carrying the CYP3A4*33 allele may hav...             NaN  \n",
      "15544  Patients carrying the CYP3A4*34 allele may hav...             NaN  \n",
      "\n",
      "[15545 rows x 4 columns]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Clinical Annotation ID</th>\n",
       "      <th>Genotype/Allele</th>\n",
       "      <th>Annotation Text</th>\n",
       "      <th>Allele Function</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>981755803</td>\n",
       "      <td>AA</td>\n",
       "      <td>Patients with the rs75527207 AA genotype (two ...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>981755803</td>\n",
       "      <td>AG</td>\n",
       "      <td>Patients with the rs75527207 AG genotype (one ...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>981755803</td>\n",
       "      <td>GG</td>\n",
       "      <td>Patients with the rs75527207 GG genotype (do n...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1449311190</td>\n",
       "      <td>CC</td>\n",
       "      <td>Patients with the CC genotype and Precursor Ce...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1449311190</td>\n",
       "      <td>CT</td>\n",
       "      <td>Patients with the CT genotype and Precursor Ce...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Clinical Annotation ID Genotype/Allele  \\\n",
       "0               981755803              AA   \n",
       "1               981755803              AG   \n",
       "2               981755803              GG   \n",
       "3              1449311190              CC   \n",
       "4              1449311190              CT   \n",
       "\n",
       "                                     Annotation Text Allele Function  \n",
       "0  Patients with the rs75527207 AA genotype (two ...             NaN  \n",
       "1  Patients with the rs75527207 AG genotype (one ...             NaN  \n",
       "2  Patients with the rs75527207 GG genotype (do n...             NaN  \n",
       "3  Patients with the CC genotype and Precursor Ce...             NaN  \n",
       "4  Patients with the CT genotype and Precursor Ce...             NaN  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Contains the genotype- or allele-based annotation text and CPIC-assigned allele function,if available\n",
    "clinical_ann_alleles_df = pd.read_csv(\"../datasets/rawData-21-Dec-2022/pharmgkb/clinicalAnnotations/clinical_ann_alleles.tsv\",sep='\\t', on_bad_lines='skip')\n",
    "print(clinical_ann_alleles_df)\n",
    "clinical_ann_alleles_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       Clinical Annotation ID  Evidence ID  \\\n",
      "0                   981755803  PA166114461   \n",
      "1                   981755803  PA166104890   \n",
      "2                   981755803    981755665   \n",
      "3                   981755803    981755678   \n",
      "4                   981755803    982009991   \n",
      "...                       ...          ...   \n",
      "14761              1451553580   1451614946   \n",
      "14762              1451566760   1451566720   \n",
      "14763              1451567040   1451566980   \n",
      "14764              1451567040   1451647340   \n",
      "14765              1451567040   1451566922   \n",
      "\n",
      "                             Evidence Type  \\\n",
      "0                     Guideline Annotation   \n",
      "1                         Label Annotation   \n",
      "2                  Variant Drug Annotation   \n",
      "3                  Variant Drug Annotation   \n",
      "4                  Variant Drug Annotation   \n",
      "...                                    ...   \n",
      "14761              Variant Drug Annotation   \n",
      "14762              Variant Drug Annotation   \n",
      "14763  Variant Functional Assay Annotation   \n",
      "14764              Variant Drug Annotation   \n",
      "14765  Variant Functional Assay Annotation   \n",
      "\n",
      "                                            Evidence URL        PMID  \\\n",
      "0      https://www.pharmgkb.org/guidelineAnnotation/P...         NaN   \n",
      "1      https://www.pharmgkb.org/labelAnnotation/PA166...         NaN   \n",
      "2      https://www.pharmgkb.org/variantAnnotation/981...  21083385.0   \n",
      "3      https://www.pharmgkb.org/variantAnnotation/981...  22047557.0   \n",
      "4      https://www.pharmgkb.org/variantAnnotation/982...  23590265.0   \n",
      "...                                                  ...         ...   \n",
      "14761  https://www.pharmgkb.org/variantAnnotation/145...  33501733.0   \n",
      "14762  https://www.pharmgkb.org/variantAnnotation/145...  33236485.0   \n",
      "14763  https://www.pharmgkb.org/variantAnnotation/145...  33393779.0   \n",
      "14764  https://www.pharmgkb.org/variantAnnotation/145...  34480108.0   \n",
      "14765  https://www.pharmgkb.org/variantAnnotation/145...  33393779.0   \n",
      "\n",
      "                                                 Summary Score  \n",
      "0      Annotation of CPIC Guideline for ivacaftor and...   100  \n",
      "1         Annotation of FDA Label for ivacaftor and CFTR   100  \n",
      "2      Genotypes AA + AG are associated with response...  0.25  \n",
      "3      Genotypes AA + AG are associated with response...   2.0  \n",
      "4      Allele A is associated with response to ivacaf...  2.25  \n",
      "...                                                  ...   ...  \n",
      "14761  Allele G is not associated with exposure to me...  -1.0  \n",
      "14762  Genotypes AA + AG are associated with increase...   1.5  \n",
      "14763  CYP3A4 *6 + *17 + *20 + *21 + *26 + *30 are as...   0.0  \n",
      "14764  CYP3A4 *22 is associated with decreased metabo...   2.5  \n",
      "14765  CYP3A4 *2 + *3 + *4+ *5 + *7 + *8 + *9 + *10 +...   0.0  \n",
      "\n",
      "[14766 rows x 7 columns]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Clinical Annotation ID</th>\n",
       "      <th>Evidence ID</th>\n",
       "      <th>Evidence Type</th>\n",
       "      <th>Evidence URL</th>\n",
       "      <th>PMID</th>\n",
       "      <th>Summary</th>\n",
       "      <th>Score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>981755803</td>\n",
       "      <td>PA166114461</td>\n",
       "      <td>Guideline Annotation</td>\n",
       "      <td>https://www.pharmgkb.org/guidelineAnnotation/P...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotation of CPIC Guideline for ivacaftor and...</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>981755803</td>\n",
       "      <td>PA166104890</td>\n",
       "      <td>Label Annotation</td>\n",
       "      <td>https://www.pharmgkb.org/labelAnnotation/PA166...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Annotation of FDA Label for ivacaftor and CFTR</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>981755803</td>\n",
       "      <td>981755665</td>\n",
       "      <td>Variant Drug Annotation</td>\n",
       "      <td>https://www.pharmgkb.org/variantAnnotation/981...</td>\n",
       "      <td>21083385.0</td>\n",
       "      <td>Genotypes AA + AG are associated with response...</td>\n",
       "      <td>0.25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>981755803</td>\n",
       "      <td>981755678</td>\n",
       "      <td>Variant Drug Annotation</td>\n",
       "      <td>https://www.pharmgkb.org/variantAnnotation/981...</td>\n",
       "      <td>22047557.0</td>\n",
       "      <td>Genotypes AA + AG are associated with response...</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>981755803</td>\n",
       "      <td>982009991</td>\n",
       "      <td>Variant Drug Annotation</td>\n",
       "      <td>https://www.pharmgkb.org/variantAnnotation/982...</td>\n",
       "      <td>23590265.0</td>\n",
       "      <td>Allele A is associated with response to ivacaf...</td>\n",
       "      <td>2.25</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Clinical Annotation ID  Evidence ID            Evidence Type  \\\n",
       "0               981755803  PA166114461     Guideline Annotation   \n",
       "1               981755803  PA166104890         Label Annotation   \n",
       "2               981755803    981755665  Variant Drug Annotation   \n",
       "3               981755803    981755678  Variant Drug Annotation   \n",
       "4               981755803    982009991  Variant Drug Annotation   \n",
       "\n",
       "                                        Evidence URL        PMID  \\\n",
       "0  https://www.pharmgkb.org/guidelineAnnotation/P...         NaN   \n",
       "1  https://www.pharmgkb.org/labelAnnotation/PA166...         NaN   \n",
       "2  https://www.pharmgkb.org/variantAnnotation/981...  21083385.0   \n",
       "3  https://www.pharmgkb.org/variantAnnotation/981...  22047557.0   \n",
       "4  https://www.pharmgkb.org/variantAnnotation/982...  23590265.0   \n",
       "\n",
       "                                             Summary Score  \n",
       "0  Annotation of CPIC Guideline for ivacaftor and...   100  \n",
       "1     Annotation of FDA Label for ivacaftor and CFTR   100  \n",
       "2  Genotypes AA + AG are associated with response...  0.25  \n",
       "3  Genotypes AA + AG are associated with response...   2.0  \n",
       "4  Allele A is associated with response to ivacaf...  2.25  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Contains information about each supporting annotation (variant annotation, guideline annotation, label annotation) for every clinical annotation\n",
    "clinical_ann_evidence_df = pd.read_csv(\"../datasets/rawData-21-Dec-2022/pharmgkb/clinicalAnnotations/clinical_ann_evidence.tsv\",sep='\\t', on_bad_lines='skip')\n",
    "print(clinical_ann_evidence_df)\n",
    "clinical_ann_evidence_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "       Clinical Annotation ID Date (YYYY-MM-DD)    Type  \\\n",
      "0                   981755803        2018-03-28  Update   \n",
      "1                   981755803        2018-11-28  Update   \n",
      "2                   981755803        2020-11-04  Update   \n",
      "3                   981755803        2021-02-02  Update   \n",
      "4                   981755803        2021-02-02  Update   \n",
      "...                       ...               ...     ...   \n",
      "13188              1451553580        2021-10-21  Create   \n",
      "13189              1451553580        2021-12-13  Update   \n",
      "13190              1451566760        2021-10-29  Create   \n",
      "13191              1451567040        2021-10-29  Create   \n",
      "13192              1451567040        2022-01-13  Update   \n",
      "\n",
      "                                                 Comment  \n",
      "0                        Added PMID 25145599 to evidence  \n",
      "1                        Added PMID 23628510 to evidence  \n",
      "2      Updated text to match new format. Added guidel...  \n",
      "3      Attached CPIC guideline and FDA label for ivac...  \n",
      "4                  Small edit to text to match template.  \n",
      "...                                                  ...  \n",
      "13188                                            Created  \n",
      "13189                    Added PMID 33501733 to evidence  \n",
      "13190                                            Created  \n",
      "13191                                            Created  \n",
      "13192                    Added PMID 34480108 to evidence  \n",
      "\n",
      "[13193 rows x 4 columns]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Clinical Annotation ID</th>\n",
       "      <th>Date (YYYY-MM-DD)</th>\n",
       "      <th>Type</th>\n",
       "      <th>Comment</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>981755803</td>\n",
       "      <td>2018-03-28</td>\n",
       "      <td>Update</td>\n",
       "      <td>Added PMID 25145599 to evidence</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>981755803</td>\n",
       "      <td>2018-11-28</td>\n",
       "      <td>Update</td>\n",
       "      <td>Added PMID 23628510 to evidence</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>981755803</td>\n",
       "      <td>2020-11-04</td>\n",
       "      <td>Update</td>\n",
       "      <td>Updated text to match new format. Added guidel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>981755803</td>\n",
       "      <td>2021-02-02</td>\n",
       "      <td>Update</td>\n",
       "      <td>Attached CPIC guideline and FDA label for ivac...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>981755803</td>\n",
       "      <td>2021-02-02</td>\n",
       "      <td>Update</td>\n",
       "      <td>Small edit to text to match template.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Clinical Annotation ID Date (YYYY-MM-DD)    Type  \\\n",
       "0               981755803        2018-03-28  Update   \n",
       "1               981755803        2018-11-28  Update   \n",
       "2               981755803        2020-11-04  Update   \n",
       "3               981755803        2021-02-02  Update   \n",
       "4               981755803        2021-02-02  Update   \n",
       "\n",
       "                                             Comment  \n",
       "0                    Added PMID 25145599 to evidence  \n",
       "1                    Added PMID 23628510 to evidence  \n",
       "2  Updated text to match new format. Added guidel...  \n",
       "3  Attached CPIC guideline and FDA label for ivac...  \n",
       "4              Small edit to text to match template.  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Contains the history of the clinical annotation, including the creation date and the dates of changes or updates to the annotation\n",
    "clinical_ann_history_df = pd.read_csv(\"../datasets/rawData-21-Dec-2022/pharmgkb/clinicalAnnotations/clinical_ann_history.tsv\",sep='\\t', on_bad_lines='skip')\n",
    "print(clinical_ann_history_df)\n",
    "clinical_ann_history_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Part 3: civic annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "        gene  entrez_id variant                         disease    doid  \\\n",
      "0       JAK2       3717   V617F               Lymphoid Leukemia  1037.0   \n",
      "1     PDGFRA       5156   D842V  Gastrointestinal Stromal Tumor  9253.0   \n",
      "2     DNMT3A       1788    R882          Acute Myeloid Leukemia  9119.0   \n",
      "3     DNMT3A       1788    R882          Acute Myeloid Leukemia  9119.0   \n",
      "4       JAK2       3717   V617F        Chronic Myeloid Leukemia  8552.0   \n",
      "...      ...        ...     ...                             ...     ...   \n",
      "4030    TP53       7157   G266V                             NaN     NaN   \n",
      "4031    TP53       7157   G266E                             NaN     NaN   \n",
      "4032    TP53       7157   F270S                             NaN     NaN   \n",
      "4033    TP53       7157   V272G                             NaN     NaN   \n",
      "4034    IDH1       3417    R132              Cholangiocarcinoma  4947.0   \n",
      "\n",
      "     phenotypes       drugs drug_interaction_type evidence_type  \\\n",
      "0           NaN         NaN                   NaN    Diagnostic   \n",
      "1           NaN         NaN                   NaN    Diagnostic   \n",
      "2           NaN         NaN                   NaN    Diagnostic   \n",
      "3           NaN         NaN                   NaN    Diagnostic   \n",
      "4           NaN         NaN                   NaN    Diagnostic   \n",
      "...         ...         ...                   ...           ...   \n",
      "4030        NaN         NaN                   NaN    Functional   \n",
      "4031        NaN         NaN                   NaN    Functional   \n",
      "4032        NaN         NaN                   NaN    Functional   \n",
      "4033        NaN         NaN                   NaN    Functional   \n",
      "4034        NaN  Ivosidenib                   NaN    Predictive   \n",
      "\n",
      "     evidence_direction  ... representative_transcript2 ensembl_version  \\\n",
      "0              Supports  ...                        NaN            75.0   \n",
      "1              Supports  ...                        NaN            75.0   \n",
      "2              Supports  ...                        NaN            75.0   \n",
      "3              Supports  ...                        NaN            75.0   \n",
      "4              Supports  ...                        NaN            75.0   \n",
      "...                 ...  ...                        ...             ...   \n",
      "4030   Does Not Support  ...                        NaN             NaN   \n",
      "4031   Does Not Support  ...                        NaN             NaN   \n",
      "4032   Does Not Support  ...                        NaN             NaN   \n",
      "4033   Does Not Support  ...                        NaN             NaN   \n",
      "4034           Supports  ...                        NaN            75.0   \n",
      "\n",
      "     reference_build                                    variant_summary  \\\n",
      "0             GRCh37  JAK2 V617F is a highly recurrent mutation in m...   \n",
      "1             GRCh37  PDGFRA D842 mutations are characterized broadl...   \n",
      "2             GRCh37  DNMT3A R882 mutations are associated with cyto...   \n",
      "3             GRCh37  DNMT3A R882 mutations are associated with cyto...   \n",
      "4             GRCh37  JAK2 V617F is a highly recurrent mutation in m...   \n",
      "...              ...                                                ...   \n",
      "4030             NaN                                                NaN   \n",
      "4031             NaN                                                NaN   \n",
      "4032          GRCh37                                                NaN   \n",
      "4033             NaN                                                NaN   \n",
      "4034          GRCh37  IDH1 R132 mutations have been observed in a nu...   \n",
      "\n",
      "     variant_origin         last_review_date  \\\n",
      "0           Somatic  2020-11-20 16:18:21 UTC   \n",
      "1           Somatic  2015-06-21 16:49:38 UTC   \n",
      "2           Somatic  2016-03-26 22:10:59 UTC   \n",
      "3           Somatic  2016-03-16 22:09:27 UTC   \n",
      "4           Somatic  2015-06-21 16:49:39 UTC   \n",
      "...             ...                      ...   \n",
      "4030        Somatic  2022-07-21 17:34:32 UTC   \n",
      "4031        Somatic  2022-07-21 17:19:50 UTC   \n",
      "4032        Somatic  2022-07-21 17:31:52 UTC   \n",
      "4033        Somatic  2022-07-21 17:30:41 UTC   \n",
      "4034        Somatic  2022-10-27 19:55:06 UTC   \n",
      "\n",
      "                                  evidence_civic_url  \\\n",
      "0         https://civicdb.org/links/evidence_items/1   \n",
      "1         https://civicdb.org/links/evidence_items/2   \n",
      "2         https://civicdb.org/links/evidence_items/3   \n",
      "3         https://civicdb.org/links/evidence_items/4   \n",
      "4         https://civicdb.org/links/evidence_items/5   \n",
      "...                                              ...   \n",
      "4030  https://civicdb.org/links/evidence_items/10574   \n",
      "4031  https://civicdb.org/links/evidence_items/10575   \n",
      "4032  https://civicdb.org/links/evidence_items/10576   \n",
      "4033  https://civicdb.org/links/evidence_items/10577   \n",
      "4034  https://civicdb.org/links/evidence_items/10886   \n",
      "\n",
      "                            variant_civic_url  \\\n",
      "0       https://civicdb.org/links/variants/64   \n",
      "1       https://civicdb.org/links/variants/99   \n",
      "2       https://civicdb.org/links/variants/32   \n",
      "3       https://civicdb.org/links/variants/32   \n",
      "4       https://civicdb.org/links/variants/64   \n",
      "...                                       ...   \n",
      "4030  https://civicdb.org/links/variants/4122   \n",
      "4031  https://civicdb.org/links/variants/3774   \n",
      "4032  https://civicdb.org/links/variants/1053   \n",
      "4033  https://civicdb.org/links/variants/4123   \n",
      "4034    https://civicdb.org/links/variants/58   \n",
      "\n",
      "                          gene_civic_url is_flagged  \n",
      "0     https://civicdb.org/links/genes/28      False  \n",
      "1     https://civicdb.org/links/genes/38      False  \n",
      "2     https://civicdb.org/links/genes/18      False  \n",
      "3     https://civicdb.org/links/genes/18      False  \n",
      "4     https://civicdb.org/links/genes/28      False  \n",
      "...                                  ...        ...  \n",
      "4030  https://civicdb.org/links/genes/45      False  \n",
      "4031  https://civicdb.org/links/genes/45      False  \n",
      "4032  https://civicdb.org/links/genes/45      False  \n",
      "4033  https://civicdb.org/links/genes/45      False  \n",
      "4034  https://civicdb.org/links/genes/26      False  \n",
      "\n",
      "[4035 rows x 42 columns]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>entrez_id</th>\n",
       "      <th>variant</th>\n",
       "      <th>disease</th>\n",
       "      <th>doid</th>\n",
       "      <th>phenotypes</th>\n",
       "      <th>drugs</th>\n",
       "      <th>drug_interaction_type</th>\n",
       "      <th>evidence_type</th>\n",
       "      <th>evidence_direction</th>\n",
       "      <th>...</th>\n",
       "      <th>representative_transcript2</th>\n",
       "      <th>ensembl_version</th>\n",
       "      <th>reference_build</th>\n",
       "      <th>variant_summary</th>\n",
       "      <th>variant_origin</th>\n",
       "      <th>last_review_date</th>\n",
       "      <th>evidence_civic_url</th>\n",
       "      <th>variant_civic_url</th>\n",
       "      <th>gene_civic_url</th>\n",
       "      <th>is_flagged</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>JAK2</td>\n",
       "      <td>3717</td>\n",
       "      <td>V617F</td>\n",
       "      <td>Lymphoid Leukemia</td>\n",
       "      <td>1037.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>JAK2 V617F is a highly recurrent mutation in m...</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2020-11-20 16:18:21 UTC</td>\n",
       "      <td>https://civicdb.org/links/evidence_items/1</td>\n",
       "      <td>https://civicdb.org/links/variants/64</td>\n",
       "      <td>https://civicdb.org/links/genes/28</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PDGFRA</td>\n",
       "      <td>5156</td>\n",
       "      <td>D842V</td>\n",
       "      <td>Gastrointestinal Stromal Tumor</td>\n",
       "      <td>9253.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>PDGFRA D842 mutations are characterized broadl...</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2015-06-21 16:49:38 UTC</td>\n",
       "      <td>https://civicdb.org/links/evidence_items/2</td>\n",
       "      <td>https://civicdb.org/links/variants/99</td>\n",
       "      <td>https://civicdb.org/links/genes/38</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DNMT3A</td>\n",
       "      <td>1788</td>\n",
       "      <td>R882</td>\n",
       "      <td>Acute Myeloid Leukemia</td>\n",
       "      <td>9119.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>DNMT3A R882 mutations are associated with cyto...</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2016-03-26 22:10:59 UTC</td>\n",
       "      <td>https://civicdb.org/links/evidence_items/3</td>\n",
       "      <td>https://civicdb.org/links/variants/32</td>\n",
       "      <td>https://civicdb.org/links/genes/18</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DNMT3A</td>\n",
       "      <td>1788</td>\n",
       "      <td>R882</td>\n",
       "      <td>Acute Myeloid Leukemia</td>\n",
       "      <td>9119.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>DNMT3A R882 mutations are associated with cyto...</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2016-03-16 22:09:27 UTC</td>\n",
       "      <td>https://civicdb.org/links/evidence_items/4</td>\n",
       "      <td>https://civicdb.org/links/variants/32</td>\n",
       "      <td>https://civicdb.org/links/genes/18</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JAK2</td>\n",
       "      <td>3717</td>\n",
       "      <td>V617F</td>\n",
       "      <td>Chronic Myeloid Leukemia</td>\n",
       "      <td>8552.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>JAK2 V617F is a highly recurrent mutation in m...</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2015-06-21 16:49:39 UTC</td>\n",
       "      <td>https://civicdb.org/links/evidence_items/5</td>\n",
       "      <td>https://civicdb.org/links/variants/64</td>\n",
       "      <td>https://civicdb.org/links/genes/28</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene  entrez_id variant                         disease    doid  \\\n",
       "0    JAK2       3717   V617F               Lymphoid Leukemia  1037.0   \n",
       "1  PDGFRA       5156   D842V  Gastrointestinal Stromal Tumor  9253.0   \n",
       "2  DNMT3A       1788    R882          Acute Myeloid Leukemia  9119.0   \n",
       "3  DNMT3A       1788    R882          Acute Myeloid Leukemia  9119.0   \n",
       "4    JAK2       3717   V617F        Chronic Myeloid Leukemia  8552.0   \n",
       "\n",
       "  phenotypes drugs drug_interaction_type evidence_type evidence_direction  \\\n",
       "0        NaN   NaN                   NaN    Diagnostic           Supports   \n",
       "1        NaN   NaN                   NaN    Diagnostic           Supports   \n",
       "2        NaN   NaN                   NaN    Diagnostic           Supports   \n",
       "3        NaN   NaN                   NaN    Diagnostic           Supports   \n",
       "4        NaN   NaN                   NaN    Diagnostic           Supports   \n",
       "\n",
       "   ... representative_transcript2 ensembl_version reference_build  \\\n",
       "0  ...                        NaN            75.0          GRCh37   \n",
       "1  ...                        NaN            75.0          GRCh37   \n",
       "2  ...                        NaN            75.0          GRCh37   \n",
       "3  ...                        NaN            75.0          GRCh37   \n",
       "4  ...                        NaN            75.0          GRCh37   \n",
       "\n",
       "                                     variant_summary variant_origin  \\\n",
       "0  JAK2 V617F is a highly recurrent mutation in m...        Somatic   \n",
       "1  PDGFRA D842 mutations are characterized broadl...        Somatic   \n",
       "2  DNMT3A R882 mutations are associated with cyto...        Somatic   \n",
       "3  DNMT3A R882 mutations are associated with cyto...        Somatic   \n",
       "4  JAK2 V617F is a highly recurrent mutation in m...        Somatic   \n",
       "\n",
       "          last_review_date                          evidence_civic_url  \\\n",
       "0  2020-11-20 16:18:21 UTC  https://civicdb.org/links/evidence_items/1   \n",
       "1  2015-06-21 16:49:38 UTC  https://civicdb.org/links/evidence_items/2   \n",
       "2  2016-03-26 22:10:59 UTC  https://civicdb.org/links/evidence_items/3   \n",
       "3  2016-03-16 22:09:27 UTC  https://civicdb.org/links/evidence_items/4   \n",
       "4  2015-06-21 16:49:39 UTC  https://civicdb.org/links/evidence_items/5   \n",
       "\n",
       "                       variant_civic_url                      gene_civic_url  \\\n",
       "0  https://civicdb.org/links/variants/64  https://civicdb.org/links/genes/28   \n",
       "1  https://civicdb.org/links/variants/99  https://civicdb.org/links/genes/38   \n",
       "2  https://civicdb.org/links/variants/32  https://civicdb.org/links/genes/18   \n",
       "3  https://civicdb.org/links/variants/32  https://civicdb.org/links/genes/18   \n",
       "4  https://civicdb.org/links/variants/64  https://civicdb.org/links/genes/28   \n",
       "\n",
       "  is_flagged  \n",
       "0      False  \n",
       "1      False  \n",
       "2      False  \n",
       "3      False  \n",
       "4      False  \n",
       "\n",
       "[5 rows x 42 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ClinicalEvidence_df = pd.read_csv(\"../datasets/rawData-21-Dec-2022/civic/nightly-ClinicalEvidenceSummaries.tsv\",sep='\\t', on_bad_lines='skip')\n",
    "print(ClinicalEvidence_df)\n",
    "ClinicalEvidence_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     gene_id                         gene_civic_url    name  entrez_id  \\\n",
      "0          1      https://civicdb.org/links/genes/1     ALK        238   \n",
      "1          2      https://civicdb.org/links/genes/2    AKT1        207   \n",
      "2          3      https://civicdb.org/links/genes/3    ARAF        369   \n",
      "3          4      https://civicdb.org/links/genes/4    ABL1         25   \n",
      "4          5      https://civicdb.org/links/genes/5    BRAF        673   \n",
      "..       ...                                    ...     ...        ...   \n",
      "484    26599  https://civicdb.org/links/genes/26599   ECSCR     641700   \n",
      "485    32083  https://civicdb.org/links/genes/32083    KLLN  100144748   \n",
      "486    34321  https://civicdb.org/links/genes/34321    DUX4  100288687   \n",
      "487    41522  https://civicdb.org/links/genes/41522  PRNCR1  101867536   \n",
      "488    56400  https://civicdb.org/links/genes/56400    COX2       4513   \n",
      "\n",
      "                                           description  \\\n",
      "0    ALK amplifications, fusions and mutations have...   \n",
      "1    AKT1, also referred to as protein kinase B, is...   \n",
      "2    ARAF has recently become increasingly consider...   \n",
      "3    ABL1 is most relevant to cancer in its role in...   \n",
      "4    BRAF mutations are found to be recurrent in ma...   \n",
      "..                                                 ...   \n",
      "484                                                NaN   \n",
      "485                                                NaN   \n",
      "486                                                NaN   \n",
      "487                                                NaN   \n",
      "488                                                NaN   \n",
      "\n",
      "            last_review_date  is_flagged  \n",
      "0    2017-03-06 00:00:15 UTC       False  \n",
      "1    2017-02-09 21:58:06 UTC       False  \n",
      "2    2015-06-21 16:49:19 UTC       False  \n",
      "3    2015-06-21 16:49:20 UTC       False  \n",
      "4    2019-09-19 19:32:44 UTC       False  \n",
      "..                       ...         ...  \n",
      "484  2015-09-03 21:57:43 UTC       False  \n",
      "485  2020-02-01 00:06:12 UTC       False  \n",
      "486  2015-09-29 16:07:57 UTC       False  \n",
      "487  2015-09-29 16:33:53 UTC       False  \n",
      "488  2015-09-29 17:28:08 UTC       False  \n",
      "\n",
      "[489 rows x 7 columns]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_civic_url</th>\n",
       "      <th>name</th>\n",
       "      <th>entrez_id</th>\n",
       "      <th>description</th>\n",
       "      <th>last_review_date</th>\n",
       "      <th>is_flagged</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>https://civicdb.org/links/genes/1</td>\n",
       "      <td>ALK</td>\n",
       "      <td>238</td>\n",
       "      <td>ALK amplifications, fusions and mutations have...</td>\n",
       "      <td>2017-03-06 00:00:15 UTC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>https://civicdb.org/links/genes/2</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>207</td>\n",
       "      <td>AKT1, also referred to as protein kinase B, is...</td>\n",
       "      <td>2017-02-09 21:58:06 UTC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>https://civicdb.org/links/genes/3</td>\n",
       "      <td>ARAF</td>\n",
       "      <td>369</td>\n",
       "      <td>ARAF has recently become increasingly consider...</td>\n",
       "      <td>2015-06-21 16:49:19 UTC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>https://civicdb.org/links/genes/4</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>25</td>\n",
       "      <td>ABL1 is most relevant to cancer in its role in...</td>\n",
       "      <td>2015-06-21 16:49:20 UTC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>https://civicdb.org/links/genes/5</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>673</td>\n",
       "      <td>BRAF mutations are found to be recurrent in ma...</td>\n",
       "      <td>2019-09-19 19:32:44 UTC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   gene_id                     gene_civic_url  name  entrez_id  \\\n",
       "0        1  https://civicdb.org/links/genes/1   ALK        238   \n",
       "1        2  https://civicdb.org/links/genes/2  AKT1        207   \n",
       "2        3  https://civicdb.org/links/genes/3  ARAF        369   \n",
       "3        4  https://civicdb.org/links/genes/4  ABL1         25   \n",
       "4        5  https://civicdb.org/links/genes/5  BRAF        673   \n",
       "\n",
       "                                         description         last_review_date  \\\n",
       "0  ALK amplifications, fusions and mutations have...  2017-03-06 00:00:15 UTC   \n",
       "1  AKT1, also referred to as protein kinase B, is...  2017-02-09 21:58:06 UTC   \n",
       "2  ARAF has recently become increasingly consider...  2015-06-21 16:49:19 UTC   \n",
       "3  ABL1 is most relevant to cancer in its role in...  2015-06-21 16:49:20 UTC   \n",
       "4  BRAF mutations are found to be recurrent in ma...  2019-09-19 19:32:44 UTC   \n",
       "\n",
       "   is_flagged  \n",
       "0       False  \n",
       "1       False  \n",
       "2       False  \n",
       "3       False  \n",
       "4       False  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "GeneSummaries_df = pd.read_csv(\"../datasets/rawData-21-Dec-2022/civic/nightly-GeneSummaries.tsv\",sep='\\t', on_bad_lines='skip')\n",
    "print(GeneSummaries_df)\n",
    "GeneSummaries_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    variant_group_id                      variant_group_civic_url  \\\n",
      "0                  1   https://civicdb.org/links/variant_groups/1   \n",
      "1                  2   https://civicdb.org/links/variant_groups/2   \n",
      "2                  3   https://civicdb.org/links/variant_groups/3   \n",
      "3                  4   https://civicdb.org/links/variant_groups/4   \n",
      "4                  5   https://civicdb.org/links/variant_groups/5   \n",
      "5                  6   https://civicdb.org/links/variant_groups/6   \n",
      "6                  7   https://civicdb.org/links/variant_groups/7   \n",
      "7                  8   https://civicdb.org/links/variant_groups/8   \n",
      "8                  9   https://civicdb.org/links/variant_groups/9   \n",
      "9                 10  https://civicdb.org/links/variant_groups/10   \n",
      "10                11  https://civicdb.org/links/variant_groups/11   \n",
      "11                12  https://civicdb.org/links/variant_groups/12   \n",
      "12                13  https://civicdb.org/links/variant_groups/13   \n",
      "13                14  https://civicdb.org/links/variant_groups/14   \n",
      "14                15  https://civicdb.org/links/variant_groups/15   \n",
      "15                16  https://civicdb.org/links/variant_groups/16   \n",
      "16                17  https://civicdb.org/links/variant_groups/17   \n",
      "17                18  https://civicdb.org/links/variant_groups/18   \n",
      "18                19  https://civicdb.org/links/variant_groups/19   \n",
      "19                20  https://civicdb.org/links/variant_groups/20   \n",
      "20                21  https://civicdb.org/links/variant_groups/21   \n",
      "21                22  https://civicdb.org/links/variant_groups/22   \n",
      "22                23  https://civicdb.org/links/variant_groups/23   \n",
      "23                24  https://civicdb.org/links/variant_groups/24   \n",
      "24                25  https://civicdb.org/links/variant_groups/25   \n",
      "25                26  https://civicdb.org/links/variant_groups/26   \n",
      "26                27  https://civicdb.org/links/variant_groups/27   \n",
      "\n",
      "                                    variant_group  \\\n",
      "0                             Imatinib Resistance   \n",
      "1                                     KIT Exon 17   \n",
      "2                           Crizotinib Resistance   \n",
      "3                                     KIT Exon 11   \n",
      "4                                    Other V600's   \n",
      "5                            Motesanib Resistance   \n",
      "6                                    NPM1 Exon 12   \n",
      "7                                    FGFR fusions   \n",
      "8                                        TSC Loss   \n",
      "9                          BRCA Germline Variants   \n",
      "10                                    ALK Fusions   \n",
      "11                            EGFR TKI Resistance   \n",
      "12                     ESR1 Ligand-Binding Domain   \n",
      "13                                HER2 Activating   \n",
      "14                          PTEN Loss-of-Function   \n",
      "15                                SYT-SSX fusions   \n",
      "16                             PML-RARa B2 Domain   \n",
      "17                                   BRAF Fusions   \n",
      "18                      Kinase Dead BRAF Mutation   \n",
      "19                           HEAT domain mutation   \n",
      "20                              NRG1 gene fusions   \n",
      "21                                  NTRK1 Fusions   \n",
      "22                   ACVR1 kinase domain mutation   \n",
      "23                                   ECD MUTATION   \n",
      "24                          JAK2 fusions in B-ALL   \n",
      "25                          ABL1 fusions in B-ALL   \n",
      "26  MiT Family Translocation Renal Cell Carcinoma   \n",
      "\n",
      "                                          description  \\\n",
      "0   While imatinib has shown to be incredibly succ...   \n",
      "1                                                 NaN   \n",
      "2   The ALK oncogene has long been considered a dr...   \n",
      "3                                                 NaN   \n",
      "4   While BRAF V600E is nearly ubiquitous in many ...   \n",
      "5   RET activation is a common oncogenic marker of...   \n",
      "6   NPM1 exon 12 mutations are frequently identifi...   \n",
      "7   FGFR fusions have been demonstrated across man...   \n",
      "8   Inactivating events in TSC1 and TSC2 have been...   \n",
      "9   BRCA germline variants have long been recogniz...   \n",
      "10  ALK fusion positive non-small cell lung cancer...   \n",
      "11  EGFR pathway activation is a nearly ubiquitous...   \n",
      "12  ER-positive breast cancer is the most common o...   \n",
      "13  HER2-positive breast cancer is widely recogniz...   \n",
      "14  PTEN loss is a common event in breast cancer, ...   \n",
      "15  Several fusions between SYT (SS18) and various...   \n",
      "16  The B2 domain of PML is the binding site for A...   \n",
      "17  BRAF fusions are a rare event in several cance...   \n",
      "18  Several BRAF mutations lead to an inactive for...   \n",
      "19                                                NaN   \n",
      "20  Emerging evidence indicates that NRG1gene fusi...   \n",
      "21  NTRK fusions are increasingly recognized as th...   \n",
      "22                                                NaN   \n",
      "23  After EGFR blockade using anti-epidermal growt...   \n",
      "24  In B-ALL JAK2 fusions with different gene part...   \n",
      "25                                                NaN   \n",
      "26  The MiT family (microphthalmia-associated tran...   \n",
      "\n",
      "           last_review_date  is_flagged  \n",
      "0   2015-06-21 16:49:55 UTC       False  \n",
      "1   2016-04-15 17:15:01 UTC       False  \n",
      "2   2015-06-21 16:49:55 UTC       False  \n",
      "3   2016-04-15 17:15:01 UTC       False  \n",
      "4   2015-06-21 16:49:55 UTC       False  \n",
      "5   2015-06-21 16:49:55 UTC       False  \n",
      "6   2015-06-21 16:49:55 UTC       False  \n",
      "7   2015-06-21 16:49:55 UTC       False  \n",
      "8   2015-06-21 16:49:55 UTC       False  \n",
      "9   2015-06-21 16:49:55 UTC       False  \n",
      "10  2015-06-21 16:49:55 UTC       False  \n",
      "11  2015-06-21 16:49:55 UTC       False  \n",
      "12  2015-06-21 16:49:55 UTC       False  \n",
      "13  2016-01-04 23:22:15 UTC       False  \n",
      "14  2015-06-21 16:49:55 UTC       False  \n",
      "15  2016-02-22 18:18:50 UTC       False  \n",
      "16  2016-02-24 00:20:12 UTC       False  \n",
      "17  2016-09-14 08:07:24 UTC       False  \n",
      "18  2020-02-20 19:37:32 UTC       False  \n",
      "19  2016-12-05 18:54:15 UTC       False  \n",
      "20  2020-04-14 17:40:24 UTC       False  \n",
      "21  2018-04-11 12:41:05 UTC       False  \n",
      "22  2018-10-15 23:37:03 UTC       False  \n",
      "23  2019-02-08 15:51:16 UTC       False  \n",
      "24  2019-12-03 20:53:35 UTC       False  \n",
      "25  2021-05-07 19:55:23 UTC       False  \n",
      "26  2022-04-12 03:21:37 UTC       False  \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_group_id</th>\n",
       "      <th>variant_group_civic_url</th>\n",
       "      <th>variant_group</th>\n",
       "      <th>description</th>\n",
       "      <th>last_review_date</th>\n",
       "      <th>is_flagged</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>https://civicdb.org/links/variant_groups/1</td>\n",
       "      <td>Imatinib Resistance</td>\n",
       "      <td>While imatinib has shown to be incredibly succ...</td>\n",
       "      <td>2015-06-21 16:49:55 UTC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>https://civicdb.org/links/variant_groups/2</td>\n",
       "      <td>KIT Exon 17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2016-04-15 17:15:01 UTC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>https://civicdb.org/links/variant_groups/3</td>\n",
       "      <td>Crizotinib Resistance</td>\n",
       "      <td>The ALK oncogene has long been considered a dr...</td>\n",
       "      <td>2015-06-21 16:49:55 UTC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>https://civicdb.org/links/variant_groups/4</td>\n",
       "      <td>KIT Exon 11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2016-04-15 17:15:01 UTC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>https://civicdb.org/links/variant_groups/5</td>\n",
       "      <td>Other V600's</td>\n",
       "      <td>While BRAF V600E is nearly ubiquitous in many ...</td>\n",
       "      <td>2015-06-21 16:49:55 UTC</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   variant_group_id                     variant_group_civic_url  \\\n",
       "0                 1  https://civicdb.org/links/variant_groups/1   \n",
       "1                 2  https://civicdb.org/links/variant_groups/2   \n",
       "2                 3  https://civicdb.org/links/variant_groups/3   \n",
       "3                 4  https://civicdb.org/links/variant_groups/4   \n",
       "4                 5  https://civicdb.org/links/variant_groups/5   \n",
       "\n",
       "           variant_group                                        description  \\\n",
       "0    Imatinib Resistance  While imatinib has shown to be incredibly succ...   \n",
       "1            KIT Exon 17                                                NaN   \n",
       "2  Crizotinib Resistance  The ALK oncogene has long been considered a dr...   \n",
       "3            KIT Exon 11                                                NaN   \n",
       "4           Other V600's  While BRAF V600E is nearly ubiquitous in many ...   \n",
       "\n",
       "          last_review_date  is_flagged  \n",
       "0  2015-06-21 16:49:55 UTC       False  \n",
       "1  2016-04-15 17:15:01 UTC       False  \n",
       "2  2015-06-21 16:49:55 UTC       False  \n",
       "3  2016-04-15 17:15:01 UTC       False  \n",
       "4  2015-06-21 16:49:55 UTC       False  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "VariantGroupSummaries_df = pd.read_csv(\"../datasets/rawData-21-Dec-2022/civic/nightly-VariantGroupSummaries.tsv\",sep='\\t', on_bad_lines='skip')\n",
    "print(VariantGroupSummaries_df)\n",
    "VariantGroupSummaries_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      variant_id                        variant_civic_url  gene  entrez_id  \\\n",
      "0              1     https://civicdb.org/links/variants/1  ABL1         25   \n",
      "1              2     https://civicdb.org/links/variants/2  ABL1         25   \n",
      "2              3     https://civicdb.org/links/variants/3  ABL1         25   \n",
      "3              4     https://civicdb.org/links/variants/4  AKT1        207   \n",
      "4              5     https://civicdb.org/links/variants/5   ALK        238   \n",
      "...          ...                                      ...   ...        ...   \n",
      "1610        4113  https://civicdb.org/links/variants/4113  TP53       7157   \n",
      "1611        4114  https://civicdb.org/links/variants/4114  TP53       7157   \n",
      "1612        4115  https://civicdb.org/links/variants/4115  TP53       7157   \n",
      "1613        4122  https://civicdb.org/links/variants/4122  TP53       7157   \n",
      "1614        4123  https://civicdb.org/links/variants/4123  TP53       7157   \n",
      "\n",
      "             variant                                            summary  \\\n",
      "0           BCR::ABL  The BCR-ABL fusion protein, commonly referred ...   \n",
      "1     BCR::ABL T315I  While the efficacy of imatinib has revolutioni...   \n",
      "2     BCR::ABL E255K  While the efficacy of imatinib has revolutioni...   \n",
      "3               E17K  AKT1 E17K is a recurrent mutation that has bee...   \n",
      "4          EML4::ALK  The EML4-ALK fusion variant 1 consisting of AL...   \n",
      "...              ...                                                ...   \n",
      "1610           C124R                                                NaN   \n",
      "1611           L139N                                                NaN   \n",
      "1612           R158G                                                NaN   \n",
      "1613           G266V                                                NaN   \n",
      "1614           V272G                                                NaN   \n",
      "\n",
      "             variant_groups chromosome        start         stop  ...  \\\n",
      "0     ABL1 fusions in B-ALL         22   23522397.0   23632600.0  ...   \n",
      "1       Imatinib Resistance          9  133748283.0  133748283.0  ...   \n",
      "2       Imatinib Resistance          9  133738363.0  133738363.0  ...   \n",
      "3                       NaN         14  105246551.0  105246551.0  ...   \n",
      "4               ALK Fusions          2   42396490.0   42522656.0  ...   \n",
      "...                     ...        ...          ...          ...  ...   \n",
      "1610                    NaN        NaN          NaN          NaN  ...   \n",
      "1611                    NaN        NaN          NaN          NaN  ...   \n",
      "1612                    NaN        NaN          NaN          NaN  ...   \n",
      "1613                    NaN        NaN          NaN          NaN  ...   \n",
      "1614                    NaN        NaN          NaN          NaN  ...   \n",
      "\n",
      "                           variant_types  \\\n",
      "0                      transcript_fusion   \n",
      "1     missense_variant,transcript_fusion   \n",
      "2     missense_variant,transcript_fusion   \n",
      "3                       missense_variant   \n",
      "4                      transcript_fusion   \n",
      "...                                  ...   \n",
      "1610                                 NaN   \n",
      "1611                                 NaN   \n",
      "1612                                 NaN   \n",
      "1613                                 NaN   \n",
      "1614                                 NaN   \n",
      "\n",
      "                                       hgvs_expressions  \\\n",
      "0                                                   NaN   \n",
      "1     NM_007313.2:c.1001C>T,NP_005148.2:p.Thr315Ile,...   \n",
      "2     NC_000009.11:g.133738363G>A,NM_005157.5:c.763G...   \n",
      "3     NC_000014.8:g.105246551C>T,NM_001014432.1:c.49...   \n",
      "4                                                   NaN   \n",
      "...                                                 ...   \n",
      "1610                                                NaN   \n",
      "1611                                                NaN   \n",
      "1612                                                NaN   \n",
      "1613                                                NaN   \n",
      "1614                                                NaN   \n",
      "\n",
      "             last_review_date  civic_variant_evidence_score  \\\n",
      "0     2022-03-10 22:27:25 UTC                         255.5   \n",
      "1     2022-03-10 19:55:19 UTC                         146.0   \n",
      "2     2022-03-11 04:36:22 UTC                          83.0   \n",
      "3     2021-11-12 18:53:34 UTC                          33.5   \n",
      "4     2022-03-10 21:08:52 UTC                          38.0   \n",
      "...                       ...                           ...   \n",
      "1610  2022-07-21 17:01:43 UTC                           3.0   \n",
      "1611  2022-07-21 16:52:29 UTC                           3.0   \n",
      "1612  2022-07-21 17:00:19 UTC                           3.0   \n",
      "1613  2022-07-21 17:34:32 UTC                           3.0   \n",
      "1614  2022-07-21 17:30:41 UTC                           3.0   \n",
      "\n",
      "     allele_registry_id clinvar_ids  \\\n",
      "0                   NaN         NaN   \n",
      "1              CA122575       12624   \n",
      "2            CA16602551      376090   \n",
      "3              CA123660       13983   \n",
      "4                   NaN         NaN   \n",
      "...                 ...         ...   \n",
      "1610                NaN         NaN   \n",
      "1611                NaN         NaN   \n",
      "1612                NaN         NaN   \n",
      "1613                NaN         NaN   \n",
      "1614                NaN         NaN   \n",
      "\n",
      "                                        variant_aliases  assertion_ids  \\\n",
      "0                     T(9;22)(Q34;Q11),BCR-ABL1,BCR-ABL            NaN   \n",
      "1     THR334ILE,RS121913459,BCR-ABL THR315ILE,BCR-AB...            NaN   \n",
      "2     E274K,RS121913448,BCR-ABL GLU255LYS,BCR-ABL1 E...            NaN   \n",
      "3                                   GLU17LYS,RS34409589            NaN   \n",
      "4                                              EML4-ALK            NaN   \n",
      "...                                                 ...            ...   \n",
      "1610                                                NaN            NaN   \n",
      "1611                                                NaN            NaN   \n",
      "1612                                                NaN            NaN   \n",
      "1613                                                NaN            NaN   \n",
      "1614                                                NaN            NaN   \n",
      "\n",
      "     assertion_civic_urls is_flagged  \n",
      "0                     NaN      False  \n",
      "1                     NaN      False  \n",
      "2                     NaN      False  \n",
      "3                     NaN      False  \n",
      "4                     NaN      False  \n",
      "...                   ...        ...  \n",
      "1610                  NaN      False  \n",
      "1611                  NaN      False  \n",
      "1612                  NaN      False  \n",
      "1613                  NaN      False  \n",
      "1614                  NaN      False  \n",
      "\n",
      "[1615 rows x 29 columns]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>variant_civic_url</th>\n",
       "      <th>gene</th>\n",
       "      <th>entrez_id</th>\n",
       "      <th>variant</th>\n",
       "      <th>summary</th>\n",
       "      <th>variant_groups</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>...</th>\n",
       "      <th>variant_types</th>\n",
       "      <th>hgvs_expressions</th>\n",
       "      <th>last_review_date</th>\n",
       "      <th>civic_variant_evidence_score</th>\n",
       "      <th>allele_registry_id</th>\n",
       "      <th>clinvar_ids</th>\n",
       "      <th>variant_aliases</th>\n",
       "      <th>assertion_ids</th>\n",
       "      <th>assertion_civic_urls</th>\n",
       "      <th>is_flagged</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>https://civicdb.org/links/variants/1</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>25</td>\n",
       "      <td>BCR::ABL</td>\n",
       "      <td>The BCR-ABL fusion protein, commonly referred ...</td>\n",
       "      <td>ABL1 fusions in B-ALL</td>\n",
       "      <td>22</td>\n",
       "      <td>23522397.0</td>\n",
       "      <td>23632600.0</td>\n",
       "      <td>...</td>\n",
       "      <td>transcript_fusion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2022-03-10 22:27:25 UTC</td>\n",
       "      <td>255.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T(9;22)(Q34;Q11),BCR-ABL1,BCR-ABL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>https://civicdb.org/links/variants/2</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>25</td>\n",
       "      <td>BCR::ABL T315I</td>\n",
       "      <td>While the efficacy of imatinib has revolutioni...</td>\n",
       "      <td>Imatinib Resistance</td>\n",
       "      <td>9</td>\n",
       "      <td>133748283.0</td>\n",
       "      <td>133748283.0</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant,transcript_fusion</td>\n",
       "      <td>NM_007313.2:c.1001C&gt;T,NP_005148.2:p.Thr315Ile,...</td>\n",
       "      <td>2022-03-10 19:55:19 UTC</td>\n",
       "      <td>146.0</td>\n",
       "      <td>CA122575</td>\n",
       "      <td>12624</td>\n",
       "      <td>THR334ILE,RS121913459,BCR-ABL THR315ILE,BCR-AB...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>https://civicdb.org/links/variants/3</td>\n",
       "      <td>ABL1</td>\n",
       "      <td>25</td>\n",
       "      <td>BCR::ABL E255K</td>\n",
       "      <td>While the efficacy of imatinib has revolutioni...</td>\n",
       "      <td>Imatinib Resistance</td>\n",
       "      <td>9</td>\n",
       "      <td>133738363.0</td>\n",
       "      <td>133738363.0</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant,transcript_fusion</td>\n",
       "      <td>NC_000009.11:g.133738363G&gt;A,NM_005157.5:c.763G...</td>\n",
       "      <td>2022-03-11 04:36:22 UTC</td>\n",
       "      <td>83.0</td>\n",
       "      <td>CA16602551</td>\n",
       "      <td>376090</td>\n",
       "      <td>E274K,RS121913448,BCR-ABL GLU255LYS,BCR-ABL1 E...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>https://civicdb.org/links/variants/4</td>\n",
       "      <td>AKT1</td>\n",
       "      <td>207</td>\n",
       "      <td>E17K</td>\n",
       "      <td>AKT1 E17K is a recurrent mutation that has bee...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>14</td>\n",
       "      <td>105246551.0</td>\n",
       "      <td>105246551.0</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NC_000014.8:g.105246551C&gt;T,NM_001014432.1:c.49...</td>\n",
       "      <td>2021-11-12 18:53:34 UTC</td>\n",
       "      <td>33.5</td>\n",
       "      <td>CA123660</td>\n",
       "      <td>13983</td>\n",
       "      <td>GLU17LYS,RS34409589</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>https://civicdb.org/links/variants/5</td>\n",
       "      <td>ALK</td>\n",
       "      <td>238</td>\n",
       "      <td>EML4::ALK</td>\n",
       "      <td>The EML4-ALK fusion variant 1 consisting of AL...</td>\n",
       "      <td>ALK Fusions</td>\n",
       "      <td>2</td>\n",
       "      <td>42396490.0</td>\n",
       "      <td>42522656.0</td>\n",
       "      <td>...</td>\n",
       "      <td>transcript_fusion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2022-03-10 21:08:52 UTC</td>\n",
       "      <td>38.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EML4-ALK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   variant_id                     variant_civic_url  gene  entrez_id  \\\n",
       "0           1  https://civicdb.org/links/variants/1  ABL1         25   \n",
       "1           2  https://civicdb.org/links/variants/2  ABL1         25   \n",
       "2           3  https://civicdb.org/links/variants/3  ABL1         25   \n",
       "3           4  https://civicdb.org/links/variants/4  AKT1        207   \n",
       "4           5  https://civicdb.org/links/variants/5   ALK        238   \n",
       "\n",
       "          variant                                            summary  \\\n",
       "0        BCR::ABL  The BCR-ABL fusion protein, commonly referred ...   \n",
       "1  BCR::ABL T315I  While the efficacy of imatinib has revolutioni...   \n",
       "2  BCR::ABL E255K  While the efficacy of imatinib has revolutioni...   \n",
       "3            E17K  AKT1 E17K is a recurrent mutation that has bee...   \n",
       "4       EML4::ALK  The EML4-ALK fusion variant 1 consisting of AL...   \n",
       "\n",
       "          variant_groups chromosome        start         stop  ...  \\\n",
       "0  ABL1 fusions in B-ALL         22   23522397.0   23632600.0  ...   \n",
       "1    Imatinib Resistance          9  133748283.0  133748283.0  ...   \n",
       "2    Imatinib Resistance          9  133738363.0  133738363.0  ...   \n",
       "3                    NaN         14  105246551.0  105246551.0  ...   \n",
       "4            ALK Fusions          2   42396490.0   42522656.0  ...   \n",
       "\n",
       "                        variant_types  \\\n",
       "0                   transcript_fusion   \n",
       "1  missense_variant,transcript_fusion   \n",
       "2  missense_variant,transcript_fusion   \n",
       "3                    missense_variant   \n",
       "4                   transcript_fusion   \n",
       "\n",
       "                                    hgvs_expressions         last_review_date  \\\n",
       "0                                                NaN  2022-03-10 22:27:25 UTC   \n",
       "1  NM_007313.2:c.1001C>T,NP_005148.2:p.Thr315Ile,...  2022-03-10 19:55:19 UTC   \n",
       "2  NC_000009.11:g.133738363G>A,NM_005157.5:c.763G...  2022-03-11 04:36:22 UTC   \n",
       "3  NC_000014.8:g.105246551C>T,NM_001014432.1:c.49...  2021-11-12 18:53:34 UTC   \n",
       "4                                                NaN  2022-03-10 21:08:52 UTC   \n",
       "\n",
       "   civic_variant_evidence_score allele_registry_id clinvar_ids  \\\n",
       "0                         255.5                NaN         NaN   \n",
       "1                         146.0           CA122575       12624   \n",
       "2                          83.0         CA16602551      376090   \n",
       "3                          33.5           CA123660       13983   \n",
       "4                          38.0                NaN         NaN   \n",
       "\n",
       "                                     variant_aliases  assertion_ids  \\\n",
       "0                  T(9;22)(Q34;Q11),BCR-ABL1,BCR-ABL            NaN   \n",
       "1  THR334ILE,RS121913459,BCR-ABL THR315ILE,BCR-AB...            NaN   \n",
       "2  E274K,RS121913448,BCR-ABL GLU255LYS,BCR-ABL1 E...            NaN   \n",
       "3                                GLU17LYS,RS34409589            NaN   \n",
       "4                                           EML4-ALK            NaN   \n",
       "\n",
       "  assertion_civic_urls is_flagged  \n",
       "0                  NaN      False  \n",
       "1                  NaN      False  \n",
       "2                  NaN      False  \n",
       "3                  NaN      False  \n",
       "4                  NaN      False  \n",
       "\n",
       "[5 rows x 29 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "VariantSummaries_df = pd.read_csv(\"../datasets/rawData-21-Dec-2022/civic/nightly-VariantSummaries.tsv\",sep='\\t', on_bad_lines='skip')\n",
    "print(VariantSummaries_df)\n",
    "VariantSummaries_df.head()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.9.12 64-bit",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "6fff98fc3b3d81bd655c2cc48858186e4d9e2db7b515bf1c3221888f12a62f87"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
