{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### preprocess civic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "        gene  entrez_id variant                         disease    doid  \\\n",
      "0       JAK2       3717   V617F               Lymphoid Leukemia  1037.0   \n",
      "1     PDGFRA       5156   D842V  Gastrointestinal Stromal Tumor  9253.0   \n",
      "2     DNMT3A       1788    R882          Acute Myeloid Leukemia  9119.0   \n",
      "3     DNMT3A       1788    R882          Acute Myeloid Leukemia  9119.0   \n",
      "4       JAK2       3717   V617F        Chronic Myeloid Leukemia  8552.0   \n",
      "...      ...        ...     ...                             ...     ...   \n",
      "4030    TP53       7157   G266V                             NaN     NaN   \n",
      "4031    TP53       7157   G266E                             NaN     NaN   \n",
      "4032    TP53       7157   F270S                             NaN     NaN   \n",
      "4033    TP53       7157   V272G                             NaN     NaN   \n",
      "4034    IDH1       3417    R132              Cholangiocarcinoma  4947.0   \n",
      "\n",
      "     phenotypes       drugs drug_interaction_type evidence_type  \\\n",
      "0           NaN         NaN                   NaN    Diagnostic   \n",
      "1           NaN         NaN                   NaN    Diagnostic   \n",
      "2           NaN         NaN                   NaN    Diagnostic   \n",
      "3           NaN         NaN                   NaN    Diagnostic   \n",
      "4           NaN         NaN                   NaN    Diagnostic   \n",
      "...         ...         ...                   ...           ...   \n",
      "4030        NaN         NaN                   NaN    Functional   \n",
      "4031        NaN         NaN                   NaN    Functional   \n",
      "4032        NaN         NaN                   NaN    Functional   \n",
      "4033        NaN         NaN                   NaN    Functional   \n",
      "4034        NaN  Ivosidenib                   NaN    Predictive   \n",
      "\n",
      "     evidence_direction  ... representative_transcript2 ensembl_version  \\\n",
      "0              Supports  ...                        NaN            75.0   \n",
      "1              Supports  ...                        NaN            75.0   \n",
      "2              Supports  ...                        NaN            75.0   \n",
      "3              Supports  ...                        NaN            75.0   \n",
      "4              Supports  ...                        NaN            75.0   \n",
      "...                 ...  ...                        ...             ...   \n",
      "4030   Does Not Support  ...                        NaN             NaN   \n",
      "4031   Does Not Support  ...                        NaN             NaN   \n",
      "4032   Does Not Support  ...                        NaN             NaN   \n",
      "4033   Does Not Support  ...                        NaN             NaN   \n",
      "4034           Supports  ...                        NaN            75.0   \n",
      "\n",
      "     reference_build                                    variant_summary  \\\n",
      "0             GRCh37  JAK2 V617F is a highly recurrent mutation in m...   \n",
      "1             GRCh37  PDGFRA D842 mutations are characterized broadl...   \n",
      "2             GRCh37  DNMT3A R882 mutations are associated with cyto...   \n",
      "3             GRCh37  DNMT3A R882 mutations are associated with cyto...   \n",
      "4             GRCh37  JAK2 V617F is a highly recurrent mutation in m...   \n",
      "...              ...                                                ...   \n",
      "4030             NaN                                                NaN   \n",
      "4031             NaN                                                NaN   \n",
      "4032          GRCh37                                                NaN   \n",
      "4033             NaN                                                NaN   \n",
      "4034          GRCh37  IDH1 R132 mutations have been observed in a nu...   \n",
      "\n",
      "     variant_origin         last_review_date  \\\n",
      "0           Somatic  2020-11-20 16:18:21 UTC   \n",
      "1           Somatic  2015-06-21 16:49:38 UTC   \n",
      "2           Somatic  2016-03-26 22:10:59 UTC   \n",
      "3           Somatic  2016-03-16 22:09:27 UTC   \n",
      "4           Somatic  2015-06-21 16:49:39 UTC   \n",
      "...             ...                      ...   \n",
      "4030        Somatic  2022-07-21 17:34:32 UTC   \n",
      "4031        Somatic  2022-07-21 17:19:50 UTC   \n",
      "4032        Somatic  2022-07-21 17:31:52 UTC   \n",
      "4033        Somatic  2022-07-21 17:30:41 UTC   \n",
      "4034        Somatic  2022-10-27 19:55:06 UTC   \n",
      "\n",
      "                                  evidence_civic_url  \\\n",
      "0         https://civicdb.org/links/evidence_items/1   \n",
      "1         https://civicdb.org/links/evidence_items/2   \n",
      "2         https://civicdb.org/links/evidence_items/3   \n",
      "3         https://civicdb.org/links/evidence_items/4   \n",
      "4         https://civicdb.org/links/evidence_items/5   \n",
      "...                                              ...   \n",
      "4030  https://civicdb.org/links/evidence_items/10574   \n",
      "4031  https://civicdb.org/links/evidence_items/10575   \n",
      "4032  https://civicdb.org/links/evidence_items/10576   \n",
      "4033  https://civicdb.org/links/evidence_items/10577   \n",
      "4034  https://civicdb.org/links/evidence_items/10886   \n",
      "\n",
      "                            variant_civic_url  \\\n",
      "0       https://civicdb.org/links/variants/64   \n",
      "1       https://civicdb.org/links/variants/99   \n",
      "2       https://civicdb.org/links/variants/32   \n",
      "3       https://civicdb.org/links/variants/32   \n",
      "4       https://civicdb.org/links/variants/64   \n",
      "...                                       ...   \n",
      "4030  https://civicdb.org/links/variants/4122   \n",
      "4031  https://civicdb.org/links/variants/3774   \n",
      "4032  https://civicdb.org/links/variants/1053   \n",
      "4033  https://civicdb.org/links/variants/4123   \n",
      "4034    https://civicdb.org/links/variants/58   \n",
      "\n",
      "                          gene_civic_url is_flagged  \n",
      "0     https://civicdb.org/links/genes/28      False  \n",
      "1     https://civicdb.org/links/genes/38      False  \n",
      "2     https://civicdb.org/links/genes/18      False  \n",
      "3     https://civicdb.org/links/genes/18      False  \n",
      "4     https://civicdb.org/links/genes/28      False  \n",
      "...                                  ...        ...  \n",
      "4030  https://civicdb.org/links/genes/45      False  \n",
      "4031  https://civicdb.org/links/genes/45      False  \n",
      "4032  https://civicdb.org/links/genes/45      False  \n",
      "4033  https://civicdb.org/links/genes/45      False  \n",
      "4034  https://civicdb.org/links/genes/26      False  \n",
      "\n",
      "[4035 rows x 42 columns]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>entrez_id</th>\n",
       "      <th>variant</th>\n",
       "      <th>disease</th>\n",
       "      <th>doid</th>\n",
       "      <th>phenotypes</th>\n",
       "      <th>drugs</th>\n",
       "      <th>drug_interaction_type</th>\n",
       "      <th>evidence_type</th>\n",
       "      <th>evidence_direction</th>\n",
       "      <th>...</th>\n",
       "      <th>representative_transcript2</th>\n",
       "      <th>ensembl_version</th>\n",
       "      <th>reference_build</th>\n",
       "      <th>variant_summary</th>\n",
       "      <th>variant_origin</th>\n",
       "      <th>last_review_date</th>\n",
       "      <th>evidence_civic_url</th>\n",
       "      <th>variant_civic_url</th>\n",
       "      <th>gene_civic_url</th>\n",
       "      <th>is_flagged</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>JAK2</td>\n",
       "      <td>3717</td>\n",
       "      <td>V617F</td>\n",
       "      <td>Lymphoid Leukemia</td>\n",
       "      <td>1037.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>JAK2 V617F is a highly recurrent mutation in m...</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2020-11-20 16:18:21 UTC</td>\n",
       "      <td>https://civicdb.org/links/evidence_items/1</td>\n",
       "      <td>https://civicdb.org/links/variants/64</td>\n",
       "      <td>https://civicdb.org/links/genes/28</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PDGFRA</td>\n",
       "      <td>5156</td>\n",
       "      <td>D842V</td>\n",
       "      <td>Gastrointestinal Stromal Tumor</td>\n",
       "      <td>9253.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>PDGFRA D842 mutations are characterized broadl...</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2015-06-21 16:49:38 UTC</td>\n",
       "      <td>https://civicdb.org/links/evidence_items/2</td>\n",
       "      <td>https://civicdb.org/links/variants/99</td>\n",
       "      <td>https://civicdb.org/links/genes/38</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DNMT3A</td>\n",
       "      <td>1788</td>\n",
       "      <td>R882</td>\n",
       "      <td>Acute Myeloid Leukemia</td>\n",
       "      <td>9119.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>DNMT3A R882 mutations are associated with cyto...</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2016-03-26 22:10:59 UTC</td>\n",
       "      <td>https://civicdb.org/links/evidence_items/3</td>\n",
       "      <td>https://civicdb.org/links/variants/32</td>\n",
       "      <td>https://civicdb.org/links/genes/18</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DNMT3A</td>\n",
       "      <td>1788</td>\n",
       "      <td>R882</td>\n",
       "      <td>Acute Myeloid Leukemia</td>\n",
       "      <td>9119.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>DNMT3A R882 mutations are associated with cyto...</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2016-03-16 22:09:27 UTC</td>\n",
       "      <td>https://civicdb.org/links/evidence_items/4</td>\n",
       "      <td>https://civicdb.org/links/variants/32</td>\n",
       "      <td>https://civicdb.org/links/genes/18</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JAK2</td>\n",
       "      <td>3717</td>\n",
       "      <td>V617F</td>\n",
       "      <td>Chronic Myeloid Leukemia</td>\n",
       "      <td>8552.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>75.0</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>JAK2 V617F is a highly recurrent mutation in m...</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2015-06-21 16:49:39 UTC</td>\n",
       "      <td>https://civicdb.org/links/evidence_items/5</td>\n",
       "      <td>https://civicdb.org/links/variants/64</td>\n",
       "      <td>https://civicdb.org/links/genes/28</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene  entrez_id variant                         disease    doid  \\\n",
       "0    JAK2       3717   V617F               Lymphoid Leukemia  1037.0   \n",
       "1  PDGFRA       5156   D842V  Gastrointestinal Stromal Tumor  9253.0   \n",
       "2  DNMT3A       1788    R882          Acute Myeloid Leukemia  9119.0   \n",
       "3  DNMT3A       1788    R882          Acute Myeloid Leukemia  9119.0   \n",
       "4    JAK2       3717   V617F        Chronic Myeloid Leukemia  8552.0   \n",
       "\n",
       "  phenotypes drugs drug_interaction_type evidence_type evidence_direction  \\\n",
       "0        NaN   NaN                   NaN    Diagnostic           Supports   \n",
       "1        NaN   NaN                   NaN    Diagnostic           Supports   \n",
       "2        NaN   NaN                   NaN    Diagnostic           Supports   \n",
       "3        NaN   NaN                   NaN    Diagnostic           Supports   \n",
       "4        NaN   NaN                   NaN    Diagnostic           Supports   \n",
       "\n",
       "   ... representative_transcript2 ensembl_version reference_build  \\\n",
       "0  ...                        NaN            75.0          GRCh37   \n",
       "1  ...                        NaN            75.0          GRCh37   \n",
       "2  ...                        NaN            75.0          GRCh37   \n",
       "3  ...                        NaN            75.0          GRCh37   \n",
       "4  ...                        NaN            75.0          GRCh37   \n",
       "\n",
       "                                     variant_summary variant_origin  \\\n",
       "0  JAK2 V617F is a highly recurrent mutation in m...        Somatic   \n",
       "1  PDGFRA D842 mutations are characterized broadl...        Somatic   \n",
       "2  DNMT3A R882 mutations are associated with cyto...        Somatic   \n",
       "3  DNMT3A R882 mutations are associated with cyto...        Somatic   \n",
       "4  JAK2 V617F is a highly recurrent mutation in m...        Somatic   \n",
       "\n",
       "          last_review_date                          evidence_civic_url  \\\n",
       "0  2020-11-20 16:18:21 UTC  https://civicdb.org/links/evidence_items/1   \n",
       "1  2015-06-21 16:49:38 UTC  https://civicdb.org/links/evidence_items/2   \n",
       "2  2016-03-26 22:10:59 UTC  https://civicdb.org/links/evidence_items/3   \n",
       "3  2016-03-16 22:09:27 UTC  https://civicdb.org/links/evidence_items/4   \n",
       "4  2015-06-21 16:49:39 UTC  https://civicdb.org/links/evidence_items/5   \n",
       "\n",
       "                       variant_civic_url                      gene_civic_url  \\\n",
       "0  https://civicdb.org/links/variants/64  https://civicdb.org/links/genes/28   \n",
       "1  https://civicdb.org/links/variants/99  https://civicdb.org/links/genes/38   \n",
       "2  https://civicdb.org/links/variants/32  https://civicdb.org/links/genes/18   \n",
       "3  https://civicdb.org/links/variants/32  https://civicdb.org/links/genes/18   \n",
       "4  https://civicdb.org/links/variants/64  https://civicdb.org/links/genes/28   \n",
       "\n",
       "  is_flagged  \n",
       "0      False  \n",
       "1      False  \n",
       "2      False  \n",
       "3      False  \n",
       "4      False  \n",
       "\n",
       "[5 rows x 42 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "civic_df = pd.read_csv(\"../datasets/rawData-21-Dec-2022/civic/nightly-ClinicalEvidenceSummaries.tsv\", sep='\\t', on_bad_lines='skip')\n",
    "print(civic_df)\n",
    "civic_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Part 1: drugs & evidence_direction filtering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2485, 42)\n",
      "(2121, 42)\n"
     ]
    }
   ],
   "source": [
    "# predictive_df = civic_df[civic_df['evidence_type'] == 'Predictive']\n",
    "predictive_df = civic_df.dropna(subset=['drugs'])\n",
    "print(predictive_df.shape)\n",
    "support_df = predictive_df[predictive_df['evidence_direction'] == 'Supports']\n",
    "print(support_df.shape)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Part 2: variant filtering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['A325B']\n",
      "[]\n",
      "['A3820B']\n",
      "[]\n",
      "['B492C']\n"
     ]
    }
   ],
   "source": [
    "# check regular expression\n",
    "test_list = ['A325B','A40','A3820BC','AODBS','AB492C']\n",
    "import re\n",
    "pattern = re.compile('[A-Z][0-9]+[A-Z]')\n",
    "for item in test_list:\n",
    "    print(pattern.findall(item))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(711, 11)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>variant</th>\n",
       "      <th>drugs</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>reference_bases</th>\n",
       "      <th>variant_bases</th>\n",
       "      <th>representative_transcript</th>\n",
       "      <th>drug_interaction_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>P124S</td>\n",
       "      <td>Selumetinib</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>15</td>\n",
       "      <td>66729162.0</td>\n",
       "      <td>66729162.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000307102.5</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>Q56P</td>\n",
       "      <td>Selumetinib</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>15</td>\n",
       "      <td>66727451.0</td>\n",
       "      <td>66727451.0</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>ENST00000307102.5</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PDGFRA</td>\n",
       "      <td>D842V</td>\n",
       "      <td>Imatinib</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>4</td>\n",
       "      <td>55152093.0</td>\n",
       "      <td>55152093.0</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000257290.5</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ARAF</td>\n",
       "      <td>S214C</td>\n",
       "      <td>Sorafenib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>X</td>\n",
       "      <td>47426121.0</td>\n",
       "      <td>47426121.0</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>ENST00000377045.4</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JAK2</td>\n",
       "      <td>V617F</td>\n",
       "      <td>Peginterferon Alfa-2b</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>9</td>\n",
       "      <td>5073770.0</td>\n",
       "      <td>5073770.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000381652.3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>713</th>\n",
       "      <td>BRAF</td>\n",
       "      <td>V600E</td>\n",
       "      <td>Encorafenib,Cetuximab</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>7</td>\n",
       "      <td>140453136.0</td>\n",
       "      <td>140453136.0</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000288602.6</td>\n",
       "      <td>Combination</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>714</th>\n",
       "      <td>EGFR</td>\n",
       "      <td>G598V</td>\n",
       "      <td>Erlotinib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>7</td>\n",
       "      <td>55233043.0</td>\n",
       "      <td>55233043.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000275493.2</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>715</th>\n",
       "      <td>SF3B1</td>\n",
       "      <td>K700E</td>\n",
       "      <td>Etoposide,Olaparib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>2</td>\n",
       "      <td>198266834.0</td>\n",
       "      <td>198266834.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>ENST00000335508.6</td>\n",
       "      <td>Substitutes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>716</th>\n",
       "      <td>IDH1</td>\n",
       "      <td>R132C</td>\n",
       "      <td>Olaparib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>2</td>\n",
       "      <td>209113113.0</td>\n",
       "      <td>209113113.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ENST00000415913.1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>717</th>\n",
       "      <td>FGFR1</td>\n",
       "      <td>N546K</td>\n",
       "      <td>Pemigatinib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>8</td>\n",
       "      <td>38274849.0</td>\n",
       "      <td>38274849.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000341462.5</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>711 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       gene variant                  drugs clinical_significance chromosome  \\\n",
       "0    MAP2K1   P124S            Selumetinib            Resistance         15   \n",
       "1    MAP2K1    Q56P            Selumetinib            Resistance         15   \n",
       "2    PDGFRA   D842V               Imatinib            Resistance          4   \n",
       "3      ARAF   S214C              Sorafenib  Sensitivity/Response          X   \n",
       "4      JAK2   V617F  Peginterferon Alfa-2b  Sensitivity/Response          9   \n",
       "..      ...     ...                    ...                   ...        ...   \n",
       "713    BRAF   V600E  Encorafenib,Cetuximab  Sensitivity/Response          7   \n",
       "714    EGFR   G598V              Erlotinib  Sensitivity/Response          7   \n",
       "715   SF3B1   K700E     Etoposide,Olaparib  Sensitivity/Response          2   \n",
       "716    IDH1   R132C               Olaparib  Sensitivity/Response          2   \n",
       "717   FGFR1   N546K            Pemigatinib  Sensitivity/Response          8   \n",
       "\n",
       "           start         stop reference_bases variant_bases  \\\n",
       "0     66729162.0   66729162.0               C             T   \n",
       "1     66727451.0   66727451.0               A             C   \n",
       "2     55152093.0   55152093.0               A             T   \n",
       "3     47426121.0   47426121.0               C             G   \n",
       "4      5073770.0    5073770.0               G             T   \n",
       "..           ...          ...             ...           ...   \n",
       "713  140453136.0  140453136.0               A             T   \n",
       "714   55233043.0   55233043.0               G             T   \n",
       "715  198266834.0  198266834.0               T             C   \n",
       "716  209113113.0  209113113.0               G             A   \n",
       "717   38274849.0   38274849.0               G             T   \n",
       "\n",
       "    representative_transcript drug_interaction_type  \n",
       "0           ENST00000307102.5                   NaN  \n",
       "1           ENST00000307102.5                   NaN  \n",
       "2           ENST00000257290.5                   NaN  \n",
       "3           ENST00000377045.4                   NaN  \n",
       "4           ENST00000381652.3                   NaN  \n",
       "..                        ...                   ...  \n",
       "713         ENST00000288602.6           Combination  \n",
       "714         ENST00000275493.2                   NaN  \n",
       "715         ENST00000335508.6           Substitutes  \n",
       "716         ENST00000415913.1                   NaN  \n",
       "717         ENST00000341462.5                   NaN  \n",
       "\n",
       "[711 rows x 11 columns]"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#missense_df = support_df[support_df['start'] == support_df['stop']]\n",
    "#print(missense_df.shape)\n",
    "\n",
    "\n",
    "missense_df = support_df.dropna(subset=['start','stop'])\n",
    "columns = ['gene', 'variant', 'drugs', 'clinical_significance', 'chromosome', 'start', 'stop', 'reference_bases', 'variant_bases', 'representative_transcript', 'drug_interaction_type']\n",
    "missense_df = pd.DataFrame(data=missense_df, columns=columns)\n",
    "missense_df.drop_duplicates(inplace=True)\n",
    "\n",
    "# check missense\n",
    "missense_df['variant'] = missense_df['variant'].str.split(' ')\n",
    "missense_df['variant'] = missense_df['variant'].apply(lambda x: x[-1])\n",
    "missense_df['variant'] = missense_df['variant'].str.split('::')\n",
    "missense_df['variant'] = missense_df['variant'].apply(lambda x: x[-1])\n",
    "\n",
    "pd.options.display.max_rows = 20\n",
    "# update: regular expression filtering\n",
    "\n",
    "missense_df = missense_df[missense_df['variant'].str.contains('[A-Z][0-9]+[A-Z]')]\n",
    "missense_df = missense_df[~missense_df['variant'].str.contains('_')]\n",
    "missense_df = missense_df[~missense_df['variant'].str.contains('DEL')]\n",
    "missense_df = missense_df[~missense_df['variant'].str.contains('INS')]\n",
    "missense_df = missense_df[~missense_df['variant'].str.contains('INST')]\n",
    "missense_df = missense_df[~missense_df['variant'].str.contains('-')]\n",
    "missense_df = missense_df[~missense_df['variant'].str.contains('/')].reset_index(drop=True)\n",
    "\n",
    "sele_index = [38, 176, 252, 260, 269, 286, 550]\n",
    "missense_df.drop(index=sele_index, inplace=True)\n",
    "print(missense_df.shape)\n",
    "#print(missense_df)\n",
    "missense_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Part 3: drug filtering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Imatinib                         40\n",
       "Dasatinib                        29\n",
       "Imatinib Mesylate                28\n",
       "Erlotinib                        24\n",
       "Cetuximab                        20\n",
       "                                 ..\n",
       "MTOR Kinase Inhibitor AZD8055     1\n",
       "PLX4720                           1\n",
       "Lapatinib,Capivasertib            1\n",
       "Capivasertib,Trastuzumab          1\n",
       "Pemigatinib                       1\n",
       "Name: drugs, Length: 234, dtype: int64"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# view unique : original 234 drugs\n",
    "missense_df['drugs'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>variant</th>\n",
       "      <th>drugs</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>reference_bases</th>\n",
       "      <th>variant_bases</th>\n",
       "      <th>representative_transcript</th>\n",
       "      <th>drug_interaction_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KIT</td>\n",
       "      <td>L576P</td>\n",
       "      <td>Sorafenib,Imatinib,Nilotinib</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>4</td>\n",
       "      <td>55593661.0</td>\n",
       "      <td>55593661.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>ENST00000288135.5</td>\n",
       "      <td>Substitutes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BRAF</td>\n",
       "      <td>V600E</td>\n",
       "      <td>Sorafenib,Panitumumab</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>7</td>\n",
       "      <td>140453136.0</td>\n",
       "      <td>140453136.0</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000288602.6</td>\n",
       "      <td>Combination</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BRAF</td>\n",
       "      <td>V600E</td>\n",
       "      <td>Dabrafenib,Trametinib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>7</td>\n",
       "      <td>140453136.0</td>\n",
       "      <td>140453136.0</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000288602.6</td>\n",
       "      <td>Combination</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BRAF</td>\n",
       "      <td>V600E</td>\n",
       "      <td>Pictilisib Bismesylate,PLX4720</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>7</td>\n",
       "      <td>140453136.0</td>\n",
       "      <td>140453136.0</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000288602.6</td>\n",
       "      <td>Combination</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BRAF</td>\n",
       "      <td>V600E</td>\n",
       "      <td>Nutlin-3,PLX4720</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>7</td>\n",
       "      <td>140453136.0</td>\n",
       "      <td>140453136.0</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000288602.6</td>\n",
       "      <td>Combination</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>203</th>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>H1047R</td>\n",
       "      <td>Akt Inhibitor MK2206,Vemurafenib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>3</td>\n",
       "      <td>178952085.0</td>\n",
       "      <td>178952085.0</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>ENST00000263967.3</td>\n",
       "      <td>Combination</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>204</th>\n",
       "      <td>BRAF</td>\n",
       "      <td>V600E</td>\n",
       "      <td>Irinotecan,Vemurafenib,Cetuximab</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>7</td>\n",
       "      <td>140453136.0</td>\n",
       "      <td>140453136.0</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000288602.6</td>\n",
       "      <td>Combination</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>BRAF</td>\n",
       "      <td>V600E</td>\n",
       "      <td>Erlotinib,Vemurafenib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>7</td>\n",
       "      <td>140453136.0</td>\n",
       "      <td>140453136.0</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000288602.6</td>\n",
       "      <td>Combination</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>BRAF</td>\n",
       "      <td>V600E</td>\n",
       "      <td>Encorafenib,Cetuximab</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>7</td>\n",
       "      <td>140453136.0</td>\n",
       "      <td>140453136.0</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000288602.6</td>\n",
       "      <td>Combination</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207</th>\n",
       "      <td>SF3B1</td>\n",
       "      <td>K700E</td>\n",
       "      <td>Etoposide,Olaparib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>2</td>\n",
       "      <td>198266834.0</td>\n",
       "      <td>198266834.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>ENST00000335508.6</td>\n",
       "      <td>Substitutes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>208 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       gene variant                             drugs clinical_significance  \\\n",
       "0       KIT   L576P      Sorafenib,Imatinib,Nilotinib            Resistance   \n",
       "1      BRAF   V600E             Sorafenib,Panitumumab  Sensitivity/Response   \n",
       "2      BRAF   V600E             Dabrafenib,Trametinib  Sensitivity/Response   \n",
       "3      BRAF   V600E    Pictilisib Bismesylate,PLX4720  Sensitivity/Response   \n",
       "4      BRAF   V600E                  Nutlin-3,PLX4720  Sensitivity/Response   \n",
       "..      ...     ...                               ...                   ...   \n",
       "203  PIK3CA  H1047R  Akt Inhibitor MK2206,Vemurafenib  Sensitivity/Response   \n",
       "204    BRAF   V600E  Irinotecan,Vemurafenib,Cetuximab  Sensitivity/Response   \n",
       "205    BRAF   V600E             Erlotinib,Vemurafenib  Sensitivity/Response   \n",
       "206    BRAF   V600E             Encorafenib,Cetuximab  Sensitivity/Response   \n",
       "207   SF3B1   K700E                Etoposide,Olaparib  Sensitivity/Response   \n",
       "\n",
       "    chromosome        start         stop reference_bases variant_bases  \\\n",
       "0            4   55593661.0   55593661.0               T             C   \n",
       "1            7  140453136.0  140453136.0               A             T   \n",
       "2            7  140453136.0  140453136.0               A             T   \n",
       "3            7  140453136.0  140453136.0               A             T   \n",
       "4            7  140453136.0  140453136.0               A             T   \n",
       "..         ...          ...          ...             ...           ...   \n",
       "203          3  178952085.0  178952085.0               A             G   \n",
       "204          7  140453136.0  140453136.0               A             T   \n",
       "205          7  140453136.0  140453136.0               A             T   \n",
       "206          7  140453136.0  140453136.0               A             T   \n",
       "207          2  198266834.0  198266834.0               T             C   \n",
       "\n",
       "    representative_transcript drug_interaction_type  \n",
       "0           ENST00000288135.5           Substitutes  \n",
       "1           ENST00000288602.6           Combination  \n",
       "2           ENST00000288602.6           Combination  \n",
       "3           ENST00000288602.6           Combination  \n",
       "4           ENST00000288602.6           Combination  \n",
       "..                        ...                   ...  \n",
       "203         ENST00000263967.3           Combination  \n",
       "204         ENST00000288602.6           Combination  \n",
       "205         ENST00000288602.6           Combination  \n",
       "206         ENST00000288602.6           Combination  \n",
       "207         ENST00000335508.6           Substitutes  \n",
       "\n",
       "[208 rows x 11 columns]"
      ]
     },
     "execution_count": 144,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# search drug combination\n",
    "combination_df = missense_df.dropna(subset=['drug_interaction_type'], inplace=False).reset_index(drop=True)\n",
    "combination_df.to_csv('../datasets/middlefile/civic_drug_combination.csv', index=None)\n",
    "combination_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>variant</th>\n",
       "      <th>drugs</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>reference_bases</th>\n",
       "      <th>variant_bases</th>\n",
       "      <th>representative_transcript</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>P124S</td>\n",
       "      <td>Selumetinib</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>15</td>\n",
       "      <td>66729162.0</td>\n",
       "      <td>66729162.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000307102.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>Q56P</td>\n",
       "      <td>Selumetinib</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>15</td>\n",
       "      <td>66727451.0</td>\n",
       "      <td>66727451.0</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>ENST00000307102.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PDGFRA</td>\n",
       "      <td>D842V</td>\n",
       "      <td>Imatinib</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>4</td>\n",
       "      <td>55152093.0</td>\n",
       "      <td>55152093.0</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000257290.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ARAF</td>\n",
       "      <td>S214C</td>\n",
       "      <td>Sorafenib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>X</td>\n",
       "      <td>47426121.0</td>\n",
       "      <td>47426121.0</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>ENST00000377045.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JAK2</td>\n",
       "      <td>V617F</td>\n",
       "      <td>Peginterferon Alfa-2b</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>9</td>\n",
       "      <td>5073770.0</td>\n",
       "      <td>5073770.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000381652.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>496</th>\n",
       "      <td>CSF1R</td>\n",
       "      <td>CSF1R</td>\n",
       "      <td>Imatinib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>1</td>\n",
       "      <td>156446804.0</td>\n",
       "      <td>156470620.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000348159.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>497</th>\n",
       "      <td>EZH2</td>\n",
       "      <td>Y646H</td>\n",
       "      <td>Tazemetostat</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>7</td>\n",
       "      <td>148508728.0</td>\n",
       "      <td>148508728.0</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>ENST00000320356.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>498</th>\n",
       "      <td>EGFR</td>\n",
       "      <td>G598V</td>\n",
       "      <td>Erlotinib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>7</td>\n",
       "      <td>55233043.0</td>\n",
       "      <td>55233043.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000275493.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499</th>\n",
       "      <td>IDH1</td>\n",
       "      <td>R132C</td>\n",
       "      <td>Olaparib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>2</td>\n",
       "      <td>209113113.0</td>\n",
       "      <td>209113113.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ENST00000415913.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>500</th>\n",
       "      <td>FGFR1</td>\n",
       "      <td>N546K</td>\n",
       "      <td>Pemigatinib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>8</td>\n",
       "      <td>38274849.0</td>\n",
       "      <td>38274849.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000341462.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>501 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       gene variant                  drugs clinical_significance chromosome  \\\n",
       "0    MAP2K1   P124S            Selumetinib            Resistance         15   \n",
       "1    MAP2K1    Q56P            Selumetinib            Resistance         15   \n",
       "2    PDGFRA   D842V               Imatinib            Resistance          4   \n",
       "3      ARAF   S214C              Sorafenib  Sensitivity/Response          X   \n",
       "4      JAK2   V617F  Peginterferon Alfa-2b  Sensitivity/Response          9   \n",
       "..      ...     ...                    ...                   ...        ...   \n",
       "496   CSF1R   CSF1R               Imatinib  Sensitivity/Response          1   \n",
       "497    EZH2   Y646H           Tazemetostat  Sensitivity/Response          7   \n",
       "498    EGFR   G598V              Erlotinib  Sensitivity/Response          7   \n",
       "499    IDH1   R132C               Olaparib  Sensitivity/Response          2   \n",
       "500   FGFR1   N546K            Pemigatinib  Sensitivity/Response          8   \n",
       "\n",
       "           start         stop reference_bases variant_bases  \\\n",
       "0     66729162.0   66729162.0               C             T   \n",
       "1     66727451.0   66727451.0               A             C   \n",
       "2     55152093.0   55152093.0               A             T   \n",
       "3     47426121.0   47426121.0               C             G   \n",
       "4      5073770.0    5073770.0               G             T   \n",
       "..           ...          ...             ...           ...   \n",
       "496  156446804.0  156470620.0             NaN           NaN   \n",
       "497  148508728.0  148508728.0               A             G   \n",
       "498   55233043.0   55233043.0               G             T   \n",
       "499  209113113.0  209113113.0               G             A   \n",
       "500   38274849.0   38274849.0               G             T   \n",
       "\n",
       "    representative_transcript  \n",
       "0           ENST00000307102.5  \n",
       "1           ENST00000307102.5  \n",
       "2           ENST00000257290.5  \n",
       "3           ENST00000377045.4  \n",
       "4           ENST00000381652.3  \n",
       "..                        ...  \n",
       "496         ENST00000348159.4  \n",
       "497         ENST00000320356.7  \n",
       "498         ENST00000275493.2  \n",
       "499         ENST00000415913.1  \n",
       "500         ENST00000341462.5  \n",
       "\n",
       "[501 rows x 10 columns]"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "singledrug_df = missense_df[missense_df['drug_interaction_type'].isnull()]\n",
    "columns = ['gene', 'variant', 'drugs', 'clinical_significance', 'chromosome', 'start', 'stop', 'reference_bases', 'variant_bases', 'representative_transcript']\n",
    "newsingledrug_df = pd.DataFrame(data=singledrug_df, columns=columns)\n",
    "newsingledrug_df.drop_duplicates(inplace=True)\n",
    "newsingledrug_df = newsingledrug_df.reset_index(drop=True)\n",
    "newsingledrug_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Resistance              250\n",
       "Sensitivity/Response    239\n",
       "Reduced Sensitivity      10\n",
       "Adverse Response          2\n",
       "Name: clinical_significance, dtype: int64"
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check significance\n",
    "newsingledrug_df['clinical_significance'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Part 4: pubchem drug mapping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "108\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "array(['Selumetinib', 'Imatinib', 'Sorafenib', 'Peginterferon Alfa-2b',\n",
       "       'Fedratinib', 'Tanespimycin', 'Temozolomide', 'Crizotinib',\n",
       "       'Alectinib', 'TAE684', 'Trametinib', 'Dasatinib', 'Crenolanib',\n",
       "       'Dabrafenib', 'Motesanib', 'JAK2 Inhibitor AZD1480', 'Vemurafenib',\n",
       "       'Cetuximab', 'NSC348884', 'Erlotinib', 'Gefitinib', 'Dacomitinib',\n",
       "       'Lapatinib', 'Hormone Therapy', 'Staurosporine', 'Neratinib',\n",
       "       'Sunitinib', 'PI3K-alpha Inhibitor MEN1611', 'Sirolimus',\n",
       "       'Doxorubicin', 'Ibrutinib', 'Ponatinib', 'Rociletinib', 'Olaparib',\n",
       "       'Fluorouracil', 'FOLFOX Regimen', 'Paclitaxel', 'Cisplatin',\n",
       "       'Epidermal Growth Factor Receptor Tyrosine Kinase Inhibitor',\n",
       "       'Uprosertib', 'Capivasertib', 'GSK321', 'Idelalisib', 'Vismodegib',\n",
       "       'Pictilisib', 'Abemaciclib', 'Palbociclib', 'Lorlatinib',\n",
       "       'Afatinib', 'Akt Inhibitor MK2206', 'Amrubicin', 'Osimertinib',\n",
       "       'AGI-5198', 'Futuximab/Modotuximab Mixture', 'Panitumumab',\n",
       "       'Tretinoin', 'Patidegib', 'Cabozantinib', 'Everolimus',\n",
       "       'Cobimetinib', 'Regorafenib', 'Foretinib', 'Ceritinib',\n",
       "       'Brigatinib', 'PD173074', 'AZD3463', 'Dactolisib', 'Alpelisib',\n",
       "       'Spliceostatin A', 'Trastuzumab', 'MEK Inhibitor TAK-733',\n",
       "       'Taselisib', 'PLX4720', 'MTOR Kinase Inhibitor AZD8055',\n",
       "       'Apitolisib', 'Pazopanib', 'Midostaurin', 'Aromatase Inhibitor',\n",
       "       'Methotrexate', 'Savolitinib', 'Entrectinib',\n",
       "       'Therapeutic Tumor Infiltrating Lymphocytes', 'Ruxolitinib',\n",
       "       'Melphalan', 'PI103', 'PI-103', 'Oxaliplatin', 'Irinotecan',\n",
       "       'Canertinib', 'Nilotinib', 'EAP Protocol', 'Ivosidenib', 'BPTES',\n",
       "       'Rucaparib', 'Tandutinib', 'MDM2 Inhibitor AMGMDS3', 'Axitinib',\n",
       "       'Bosutinib', 'Imatinib Mesylate', 'U0126',\n",
       "       'Akt/ERK Inhibitor ONC201', 'BRAF Inhibitor', 'PI3Ka/Di',\n",
       "       'Arsenic Trioxide', 'Venetoclax', 'Gilteritinib', 'Tazemetostat',\n",
       "       'Pemigatinib'], dtype=object)"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "drug_unique_list = newsingledrug_df['drugs'].unique()\n",
    "print(len(drug_unique_list))\n",
    "drug_unique_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 108/108 [02:20<00:00,  1.31s/it]\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['Peginterferon Alfa-2b',\n",
       " 'JAK2 Inhibitor AZD1480',\n",
       " 'Cetuximab',\n",
       " 'Hormone Therapy',\n",
       " 'PI3K-alpha Inhibitor MEN1611',\n",
       " 'Epidermal Growth Factor Receptor Tyrosine Kinase Inhibitor',\n",
       " 'Akt Inhibitor MK2206',\n",
       " 'Futuximab/Modotuximab Mixture',\n",
       " 'Panitumumab',\n",
       " 'Trastuzumab',\n",
       " 'MTOR Kinase Inhibitor AZD8055',\n",
       " 'Aromatase Inhibitor',\n",
       " 'Therapeutic Tumor Infiltrating Lymphocytes',\n",
       " 'EAP Protocol',\n",
       " 'MDM2 Inhibitor AMGMDS3',\n",
       " 'Akt/ERK Inhibitor ONC201',\n",
       " 'PI3Ka/Di']"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pubchempy as pcp\n",
    "from tqdm import tqdm\n",
    "\n",
    "# test\n",
    "#results = pcp.get_compounds('Selumetinib','name') # [Compound(10127622)]\n",
    "\n",
    "error_list = []\n",
    "for i in tqdm(range(len(drug_unique_list))):\n",
    "    if(len(pcp.get_compounds(drug_unique_list[i],'name')) == 0):\n",
    "        error_list.append(drug_unique_list[i])\n",
    "error_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_3240\\1316244949.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  fixdrug_df['drugs'] = fixdrug_df['drugs'].str.replace('JAK2 Inhibitor AZD1480', 'AZD1480')\n",
      "C:\\Users\\Administrator\\AppData\\Local\\Temp\\ipykernel_3240\\1316244949.py:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  fixdrug_df['drugs'] = fixdrug_df['drugs'].str.replace('Cetuximab', 'Erbitux')\n",
      "100%|██████████| 98/98 [02:07<00:00,  1.31s/it]\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[]"
      ]
     },
     "execution_count": 134,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# fix drugname for pubchem\n",
    "fixdrug_df = newsingledrug_df[~newsingledrug_df['drugs'].str.contains('Peginterferon Alfa-2b')]\n",
    "#fix_druglist = ['AZD1480' if i == 'JAK2 Inhibitor AZD1480' else i for i in drug_unique_list]\n",
    "fixdrug_df['drugs'] = fixdrug_df['drugs'].str.replace('JAK2 Inhibitor AZD1480', 'AZD1480')\n",
    "fixdrug_df['drugs'] = fixdrug_df['drugs'].str.replace('Cetuximab', 'Erbitux')\n",
    "fixdrug_df = fixdrug_df[~fixdrug_df['drugs'].str.contains('Hormone Therapy')]\n",
    "fixdrug_df['drugs'] = fixdrug_df['drugs'].str.replace('PI3K-alpha Inhibitor MEN1611', 'MEN1611')\n",
    "fixdrug_df = fixdrug_df[~fixdrug_df['drugs'].str.contains('Epidermal Growth Factor Receptor Tyrosine Kinase Inhibitor')]\n",
    "fixdrug_df['drugs'] = fixdrug_df['drugs'].str.replace('Akt Inhibitor MK2206', 'MK2206')\n",
    "fixdrug_df['drugs'] = fixdrug_df['drugs'].str.replace('Trastuzumab', 'BCP32858')\n",
    "fixdrug_df['drugs'] = fixdrug_df['drugs'].str.replace('MTOR Kinase Inhibitor AZD8055', 'AZD8055')\n",
    "fixdrug_df = fixdrug_df[~fixdrug_df['drugs'].str.contains('Aromatase Inhibitor')]\n",
    "fixdrug_df = fixdrug_df[~fixdrug_df['drugs'].str.contains('Therapeutic Tumor Infiltrating Lymphocytes')]\n",
    "fixdrug_df = fixdrug_df[~fixdrug_df['drugs'].str.contains('EAP Protocol')]\n",
    "fixdrug_df['drugs'] = fixdrug_df['drugs'].str.replace('MDM2 Inhibitor AMGMDS3', 'AMGMDS3')\n",
    "fixdrug_df['drugs'] = fixdrug_df['drugs'].str.replace('Akt/ERK Inhibitor ONC201', 'ONC201')\n",
    "fixdrug_df['drugs'] = fixdrug_df['drugs'].str.replace('Akt/ERK Inhibitor ONC201', 'ONC201')\n",
    "fixdrug_df = fixdrug_df[~fixdrug_df['drugs'].str.contains('PI3Ka/Di')]\n",
    "fixdrug_df = fixdrug_df[~fixdrug_df['drugs'].str.contains('Futuximab/Modotuximab Mixture')]\n",
    "fixdrug_df = fixdrug_df[~fixdrug_df['drugs'].str.contains('Panitumumab')]\n",
    "fixdrug_df = fixdrug_df[~fixdrug_df['drugs'].str.contains('Erbitux')]\n",
    "\n",
    "\n",
    "new_drug_unique_list = fixdrug_df['drugs'].unique()\n",
    "new_error_list = []\n",
    "for i in tqdm(range(len(new_drug_unique_list))):\n",
    "    if(len(pcp.get_compounds(new_drug_unique_list[i],'name')) == 0):\n",
    "        new_error_list.append(new_drug_unique_list[i])\n",
    "new_error_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 98/98 [02:06<00:00,  1.29s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "            drugname                                              smile  \\\n",
      "0        Selumetinib  CN1C=NC2=C1C=C(C(=C2F)NC3=C(C=C(C=C3)Br)Cl)C(=...   \n",
      "1           Imatinib  CC1=C(C=C(C=C1)NC(=O)C2=CC=C(C=C2)CN3CCN(CC3)C...   \n",
      "2          Sorafenib  CNC(=O)C1=NC=CC(=C1)OC2=CC=C(C=C2)NC(=O)NC3=CC...   \n",
      "3         Fedratinib  CC1=CN=C(N=C1NC2=CC(=CC=C2)S(=O)(=O)NC(C)(C)C)...   \n",
      "4       Tanespimycin  C[C@H]1C[C@@H]([C@@H]([C@H](/C=C(/[C@@H]([C@H]...   \n",
      "..               ...                                                ...   \n",
      "93  Arsenic Trioxide                    [O-2].[O-2].[O-2].[As+3].[As+3]   \n",
      "94        Venetoclax  CC1(CCC(=C(C1)C2=CC=C(C=C2)Cl)CN3CCN(CC3)C4=CC...   \n",
      "95      Gilteritinib  CCC1=C(N=C(C(=N1)C(=O)N)NC2=CC(=C(C=C2)N3CCC(C...   \n",
      "96      Tazemetostat  CCN(C1CCOCC1)C2=CC(=CC(=C2C)C(=O)NCC3=C(C=C(NC...   \n",
      "97       Pemigatinib  CCN1C2=C3C=C(NC3=NC=C2CN(C1=O)C4=C(C(=CC(=C4F)...   \n",
      "\n",
      "   molecular_weight molecular_formula  \\\n",
      "0             457.7   C17H15BrClFN4O3   \n",
      "1             493.6         C29H31N7O   \n",
      "2             464.8    C21H16ClF3N4O3   \n",
      "3             524.7       C27H36N6O3S   \n",
      "4             585.7        C31H43N3O8   \n",
      "..              ...               ...   \n",
      "93          197.841             As2O3   \n",
      "94            868.4     C45H50ClN7O7S   \n",
      "95            552.7        C29H44N8O3   \n",
      "96            572.7        C34H44N4O4   \n",
      "97            487.5      C24H27F2N5O4   \n",
      "\n",
      "                                                 atom  \n",
      "0   [Atom(1, Br), Atom(2, Cl), Atom(3, F), Atom(4,...  \n",
      "1   [Atom(1, O), Atom(2, N), Atom(3, N), Atom(4, N...  \n",
      "2   [Atom(1, Cl), Atom(2, F), Atom(3, F), Atom(4, ...  \n",
      "3   [Atom(1, S), Atom(2, O), Atom(3, O), Atom(4, O...  \n",
      "4   [Atom(1, O), Atom(2, O), Atom(3, O), Atom(4, O...  \n",
      "..                                                ...  \n",
      "93  [Atom(1, As), Atom(2, As), Atom(3, O), Atom(4,...  \n",
      "94  [Atom(1, Cl), Atom(2, S), Atom(3, O), Atom(4, ...  \n",
      "95  [Atom(1, O), Atom(2, O), Atom(3, O), Atom(4, N...  \n",
      "96  [Atom(1, O), Atom(2, O), Atom(3, O), Atom(4, O...  \n",
      "97  [Atom(1, F), Atom(2, F), Atom(3, O), Atom(4, O...  \n",
      "\n",
      "[98 rows x 5 columns]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "pd.options.display.max_rows = 20\n",
    "\n",
    "# generate drug table: smile\n",
    "drug_table = pd.DataFrame(columns=['drugname', 'smile', 'molecular_weight', 'molecular_formula', 'atom'])\n",
    "for i in tqdm(range(len(new_drug_unique_list))):\n",
    "    compound = pcp.get_compounds(new_drug_unique_list[i],'name')[0]\n",
    "    drugname = new_drug_unique_list[i]\n",
    "    try:\n",
    "        smile = compound.isomeric_smiles\n",
    "    except AttributeError:\n",
    "        smile = np.nan\n",
    "    try:\n",
    "        molecular_weight = compound.molecular_weight\n",
    "    except AttributeError:\n",
    "        molecular_weight = np.nan   \n",
    "    try:\n",
    "        molecular_formula = compound.molecular_formula\n",
    "    except AttributeError:\n",
    "        molecular_formula = np.nan\n",
    "    try: \n",
    "        atom = compound.atoms\n",
    "    except AttributeError:\n",
    "        atom = np.nan    \n",
    "    drug_table = drug_table.append([{'drugname':drugname, 'smile':smile, 'molecular_weight':molecular_weight, 'molecular_formula':molecular_formula, \n",
    "                                    'atom':atom}], ignore_index=True)\n",
    "print(drug_table)\n",
    "drug_table.to_csv('../datasets/middlefile/civic_drug_table.csv', index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>drugname</th>\n",
       "      <th>smile</th>\n",
       "      <th>molecular_weight</th>\n",
       "      <th>molecular_formula</th>\n",
       "      <th>atom</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Selumetinib</td>\n",
       "      <td>CN1C=NC2=C1C=C(C(=C2F)NC3=C(C=C(C=C3)Br)Cl)C(=...</td>\n",
       "      <td>457.7</td>\n",
       "      <td>C17H15BrClFN4O3</td>\n",
       "      <td>[Atom(1, Br), Atom(2, Cl), Atom(3, F), Atom(4,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Imatinib</td>\n",
       "      <td>CC1=C(C=C(C=C1)NC(=O)C2=CC=C(C=C2)CN3CCN(CC3)C...</td>\n",
       "      <td>493.6</td>\n",
       "      <td>C29H31N7O</td>\n",
       "      <td>[Atom(1, O), Atom(2, N), Atom(3, N), Atom(4, N...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Sorafenib</td>\n",
       "      <td>CNC(=O)C1=NC=CC(=C1)OC2=CC=C(C=C2)NC(=O)NC3=CC...</td>\n",
       "      <td>464.8</td>\n",
       "      <td>C21H16ClF3N4O3</td>\n",
       "      <td>[Atom(1, Cl), Atom(2, F), Atom(3, F), Atom(4, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Fedratinib</td>\n",
       "      <td>CC1=CN=C(N=C1NC2=CC(=CC=C2)S(=O)(=O)NC(C)(C)C)...</td>\n",
       "      <td>524.7</td>\n",
       "      <td>C27H36N6O3S</td>\n",
       "      <td>[Atom(1, S), Atom(2, O), Atom(3, O), Atom(4, O...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Tanespimycin</td>\n",
       "      <td>C[C@H]1C[C@@H]([C@@H]([C@H](/C=C(/[C@@H]([C@H]...</td>\n",
       "      <td>585.7</td>\n",
       "      <td>C31H43N3O8</td>\n",
       "      <td>[Atom(1, O), Atom(2, O), Atom(3, O), Atom(4, O...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>Arsenic Trioxide</td>\n",
       "      <td>[O-2].[O-2].[O-2].[As+3].[As+3]</td>\n",
       "      <td>197.841</td>\n",
       "      <td>As2O3</td>\n",
       "      <td>[Atom(1, As), Atom(2, As), Atom(3, O), Atom(4,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>Venetoclax</td>\n",
       "      <td>CC1(CCC(=C(C1)C2=CC=C(C=C2)Cl)CN3CCN(CC3)C4=CC...</td>\n",
       "      <td>868.4</td>\n",
       "      <td>C45H50ClN7O7S</td>\n",
       "      <td>[Atom(1, Cl), Atom(2, S), Atom(3, O), Atom(4, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>Gilteritinib</td>\n",
       "      <td>CCC1=C(N=C(C(=N1)C(=O)N)NC2=CC(=C(C=C2)N3CCC(C...</td>\n",
       "      <td>552.7</td>\n",
       "      <td>C29H44N8O3</td>\n",
       "      <td>[Atom(1, O), Atom(2, O), Atom(3, O), Atom(4, N...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>Tazemetostat</td>\n",
       "      <td>CCN(C1CCOCC1)C2=CC(=CC(=C2C)C(=O)NCC3=C(C=C(NC...</td>\n",
       "      <td>572.7</td>\n",
       "      <td>C34H44N4O4</td>\n",
       "      <td>[Atom(1, O), Atom(2, O), Atom(3, O), Atom(4, O...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>Pemigatinib</td>\n",
       "      <td>CCN1C2=C3C=C(NC3=NC=C2CN(C1=O)C4=C(C(=CC(=C4F)...</td>\n",
       "      <td>487.5</td>\n",
       "      <td>C24H27F2N5O4</td>\n",
       "      <td>[Atom(1, F), Atom(2, F), Atom(3, O), Atom(4, O...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>98 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            drugname                                              smile  \\\n",
       "0        Selumetinib  CN1C=NC2=C1C=C(C(=C2F)NC3=C(C=C(C=C3)Br)Cl)C(=...   \n",
       "1           Imatinib  CC1=C(C=C(C=C1)NC(=O)C2=CC=C(C=C2)CN3CCN(CC3)C...   \n",
       "2          Sorafenib  CNC(=O)C1=NC=CC(=C1)OC2=CC=C(C=C2)NC(=O)NC3=CC...   \n",
       "3         Fedratinib  CC1=CN=C(N=C1NC2=CC(=CC=C2)S(=O)(=O)NC(C)(C)C)...   \n",
       "4       Tanespimycin  C[C@H]1C[C@@H]([C@@H]([C@H](/C=C(/[C@@H]([C@H]...   \n",
       "..               ...                                                ...   \n",
       "93  Arsenic Trioxide                    [O-2].[O-2].[O-2].[As+3].[As+3]   \n",
       "94        Venetoclax  CC1(CCC(=C(C1)C2=CC=C(C=C2)Cl)CN3CCN(CC3)C4=CC...   \n",
       "95      Gilteritinib  CCC1=C(N=C(C(=N1)C(=O)N)NC2=CC(=C(C=C2)N3CCC(C...   \n",
       "96      Tazemetostat  CCN(C1CCOCC1)C2=CC(=CC(=C2C)C(=O)NCC3=C(C=C(NC...   \n",
       "97       Pemigatinib  CCN1C2=C3C=C(NC3=NC=C2CN(C1=O)C4=C(C(=CC(=C4F)...   \n",
       "\n",
       "   molecular_weight molecular_formula  \\\n",
       "0             457.7   C17H15BrClFN4O3   \n",
       "1             493.6         C29H31N7O   \n",
       "2             464.8    C21H16ClF3N4O3   \n",
       "3             524.7       C27H36N6O3S   \n",
       "4             585.7        C31H43N3O8   \n",
       "..              ...               ...   \n",
       "93          197.841             As2O3   \n",
       "94            868.4     C45H50ClN7O7S   \n",
       "95            552.7        C29H44N8O3   \n",
       "96            572.7        C34H44N4O4   \n",
       "97            487.5      C24H27F2N5O4   \n",
       "\n",
       "                                                 atom  \n",
       "0   [Atom(1, Br), Atom(2, Cl), Atom(3, F), Atom(4,...  \n",
       "1   [Atom(1, O), Atom(2, N), Atom(3, N), Atom(4, N...  \n",
       "2   [Atom(1, Cl), Atom(2, F), Atom(3, F), Atom(4, ...  \n",
       "3   [Atom(1, S), Atom(2, O), Atom(3, O), Atom(4, O...  \n",
       "4   [Atom(1, O), Atom(2, O), Atom(3, O), Atom(4, O...  \n",
       "..                                                ...  \n",
       "93  [Atom(1, As), Atom(2, As), Atom(3, O), Atom(4,...  \n",
       "94  [Atom(1, Cl), Atom(2, S), Atom(3, O), Atom(4, ...  \n",
       "95  [Atom(1, O), Atom(2, O), Atom(3, O), Atom(4, N...  \n",
       "96  [Atom(1, O), Atom(2, O), Atom(3, O), Atom(4, O...  \n",
       "97  [Atom(1, F), Atom(2, F), Atom(3, O), Atom(4, O...  \n",
       "\n",
       "[98 rows x 5 columns]"
      ]
     },
     "execution_count": 147,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "drug_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>variant</th>\n",
       "      <th>drugs</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>reference_bases</th>\n",
       "      <th>variant_bases</th>\n",
       "      <th>representative_transcript</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>P124S</td>\n",
       "      <td>Selumetinib</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>15</td>\n",
       "      <td>66729162.0</td>\n",
       "      <td>66729162.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000307102.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>Q56P</td>\n",
       "      <td>Selumetinib</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>15</td>\n",
       "      <td>66727451.0</td>\n",
       "      <td>66727451.0</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>ENST00000307102.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PDGFRA</td>\n",
       "      <td>D842V</td>\n",
       "      <td>Imatinib</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>4</td>\n",
       "      <td>55152093.0</td>\n",
       "      <td>55152093.0</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000257290.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ARAF</td>\n",
       "      <td>S214C</td>\n",
       "      <td>Sorafenib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>X</td>\n",
       "      <td>47426121.0</td>\n",
       "      <td>47426121.0</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>ENST00000377045.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>JAK2</td>\n",
       "      <td>V617F</td>\n",
       "      <td>Fedratinib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>9</td>\n",
       "      <td>5073770.0</td>\n",
       "      <td>5073770.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000381652.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>496</th>\n",
       "      <td>CSF1R</td>\n",
       "      <td>CSF1R</td>\n",
       "      <td>Imatinib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>1</td>\n",
       "      <td>156446804.0</td>\n",
       "      <td>156470620.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENST00000348159.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>497</th>\n",
       "      <td>EZH2</td>\n",
       "      <td>Y646H</td>\n",
       "      <td>Tazemetostat</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>7</td>\n",
       "      <td>148508728.0</td>\n",
       "      <td>148508728.0</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>ENST00000320356.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>498</th>\n",
       "      <td>EGFR</td>\n",
       "      <td>G598V</td>\n",
       "      <td>Erlotinib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>7</td>\n",
       "      <td>55233043.0</td>\n",
       "      <td>55233043.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000275493.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499</th>\n",
       "      <td>IDH1</td>\n",
       "      <td>R132C</td>\n",
       "      <td>Olaparib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>2</td>\n",
       "      <td>209113113.0</td>\n",
       "      <td>209113113.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ENST00000415913.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>500</th>\n",
       "      <td>FGFR1</td>\n",
       "      <td>N546K</td>\n",
       "      <td>Pemigatinib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>8</td>\n",
       "      <td>38274849.0</td>\n",
       "      <td>38274849.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000341462.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>455 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       gene variant         drugs clinical_significance chromosome  \\\n",
       "0    MAP2K1   P124S   Selumetinib            Resistance         15   \n",
       "1    MAP2K1    Q56P   Selumetinib            Resistance         15   \n",
       "2    PDGFRA   D842V      Imatinib            Resistance          4   \n",
       "3      ARAF   S214C     Sorafenib  Sensitivity/Response          X   \n",
       "5      JAK2   V617F    Fedratinib  Sensitivity/Response          9   \n",
       "..      ...     ...           ...                   ...        ...   \n",
       "496   CSF1R   CSF1R      Imatinib  Sensitivity/Response          1   \n",
       "497    EZH2   Y646H  Tazemetostat  Sensitivity/Response          7   \n",
       "498    EGFR   G598V     Erlotinib  Sensitivity/Response          7   \n",
       "499    IDH1   R132C      Olaparib  Sensitivity/Response          2   \n",
       "500   FGFR1   N546K   Pemigatinib  Sensitivity/Response          8   \n",
       "\n",
       "           start         stop reference_bases variant_bases  \\\n",
       "0     66729162.0   66729162.0               C             T   \n",
       "1     66727451.0   66727451.0               A             C   \n",
       "2     55152093.0   55152093.0               A             T   \n",
       "3     47426121.0   47426121.0               C             G   \n",
       "5      5073770.0    5073770.0               G             T   \n",
       "..           ...          ...             ...           ...   \n",
       "496  156446804.0  156470620.0             NaN           NaN   \n",
       "497  148508728.0  148508728.0               A             G   \n",
       "498   55233043.0   55233043.0               G             T   \n",
       "499  209113113.0  209113113.0               G             A   \n",
       "500   38274849.0   38274849.0               G             T   \n",
       "\n",
       "    representative_transcript  \n",
       "0           ENST00000307102.5  \n",
       "1           ENST00000307102.5  \n",
       "2           ENST00000257290.5  \n",
       "3           ENST00000377045.4  \n",
       "5           ENST00000381652.3  \n",
       "..                        ...  \n",
       "496         ENST00000348159.4  \n",
       "497         ENST00000320356.7  \n",
       "498         ENST00000275493.2  \n",
       "499         ENST00000415913.1  \n",
       "500         ENST00000341462.5  \n",
       "\n",
       "[455 rows x 10 columns]"
      ]
     },
     "execution_count": 146,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fixdrug_df.to_csv('../datasets/middlefile/civic_evidence_table.csv', index=None)\n",
    "fixdrug_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Sensitivity/Response    225\n",
       "Resistance              218\n",
       "Reduced Sensitivity      10\n",
       "Adverse Response          2\n",
       "Name: clinical_significance, dtype: int64"
      ]
     },
     "execution_count": 149,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fixdrug_df['clinical_significance'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Part 5: gene variant mapping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ABL1      108\n",
       "EGFR       50\n",
       "ALK        36\n",
       "PIK3CA     30\n",
       "KIT        26\n",
       "         ... \n",
       "GSTP1       1\n",
       "MTHFR       1\n",
       "NPM1        1\n",
       "JAK2        1\n",
       "EZH2        1\n",
       "Name: gene, Length: 57, dtype: int64"
      ]
     },
     "execution_count": 150,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fixdrug_df['gene'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['MAP2K1', 'PDGFRA', 'ARAF', 'JAK2', 'NRAS', 'ALK', 'KIT', 'BRAF',\n",
       "       'RET', 'NPM1', 'ABL1', 'EGFR', 'ERBB2', 'DDR2', 'PIK3CA', 'TP53',\n",
       "       'BTK', 'AKT1', 'ATM', 'MTHFR', 'GSTP1', 'ABCB1', 'ERCC2', 'IDH1',\n",
       "       'SMO', 'MAPK1', 'KRAS', 'NQO1', 'MYD88', 'PML', 'ROS1', 'MTOR',\n",
       "       'KDR', 'FGFR1', 'FLT3', 'FGFR2', 'CDK4', 'SF3B1', 'GNAQ', 'FGFR3',\n",
       "       'STK11', 'MET', 'ERBB3', 'SLCO1B1', 'NTRK3', 'JAK1', 'BRCA2',\n",
       "       'NTRK1', 'HRAS', 'AKT3', 'BRCA1', 'PREX2', 'CSF3R', 'H3-3A',\n",
       "       'BCL2', 'CSF1R', 'EZH2'], dtype=object)"
      ]
     },
     "execution_count": 151,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_list = fixdrug_df['gene'].unique()\n",
    "gene_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Q02750 MPKKKPTPIQLNPAPDGSAVNGTSSAETNLEALQKKLEELELDEQQRKRLEAFLTQKQKVGELKDDDFEKISELGAGNGGVVFKVSHKPSGLVMARKLIHLEIKPAIRNQIIRELQVLHECNSPYIVGFYGAFYSDGEISICMEHMDGGSLDQVLKKAGRIPEQILGKVSIAVIKGLTYLREKHKIMHRDVKPSNILVNSRGEIKLCDFGVSGQLIDSMANSFVGTRSYMSPERLQGTHYSVQSDIWSMGLSLVEMAVGRYPIPPPDAKELELMFGCQVEGDAAETPPRPRTPGRPLSSYGMDSRPPMAIFELLDYIVNEPPPKLPSGVFSLEFQDFVNKCLIKNPAERADLKQLMVHAFIKRSDAEEVDFAGWLCSTIGLNQPSTPTHAAGV\n"
     ]
    }
   ],
   "source": [
    "import requests\n",
    "import re\n",
    "\n",
    "# test\n",
    "url = 'https://rest.uniprot.org/uniprotkb/stream?compressed=false&format=fasta&query=(reviewed:true)%20AND%20(organism_id:9606)%20AND%20MAP2K1'\n",
    "all_fastas = requests.get(url).text\n",
    "fasta_list = re.split(r'\\n(?=>)', all_fastas)\n",
    "entry = [fasta for fasta in fasta_list if 'MAP2K1' in fasta][0]\n",
    "uniprotac = entry.split('|')[1]\n",
    "fasta = entry.split(' ')[-1].replace('\\n','')[4:]\n",
    "print(uniprotac, fasta)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>variant</th>\n",
       "      <th>drugs</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>reference_bases</th>\n",
       "      <th>variant_bases</th>\n",
       "      <th>representative_transcript</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>P124S</td>\n",
       "      <td>Selumetinib</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>15</td>\n",
       "      <td>66729162.0</td>\n",
       "      <td>66729162.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000307102.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>Q56P</td>\n",
       "      <td>Selumetinib</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>15</td>\n",
       "      <td>66727451.0</td>\n",
       "      <td>66727451.0</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>ENST00000307102.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PDGFRA</td>\n",
       "      <td>D842V</td>\n",
       "      <td>Imatinib</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>4</td>\n",
       "      <td>55152093.0</td>\n",
       "      <td>55152093.0</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000257290.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ARAF</td>\n",
       "      <td>S214C</td>\n",
       "      <td>Sorafenib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>X</td>\n",
       "      <td>47426121.0</td>\n",
       "      <td>47426121.0</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>ENST00000377045.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JAK2</td>\n",
       "      <td>V617F</td>\n",
       "      <td>Fedratinib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>9</td>\n",
       "      <td>5073770.0</td>\n",
       "      <td>5073770.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000381652.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>445</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>H396P</td>\n",
       "      <td>Dasatinib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>9</td>\n",
       "      <td>133750356.0</td>\n",
       "      <td>133750356.0</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>ENST00000318560.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>446</th>\n",
       "      <td>EZH2</td>\n",
       "      <td>Y646H</td>\n",
       "      <td>Tazemetostat</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>7</td>\n",
       "      <td>148508728.0</td>\n",
       "      <td>148508728.0</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>ENST00000320356.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>447</th>\n",
       "      <td>EGFR</td>\n",
       "      <td>G598V</td>\n",
       "      <td>Erlotinib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>7</td>\n",
       "      <td>55233043.0</td>\n",
       "      <td>55233043.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000275493.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>448</th>\n",
       "      <td>IDH1</td>\n",
       "      <td>R132C</td>\n",
       "      <td>Olaparib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>2</td>\n",
       "      <td>209113113.0</td>\n",
       "      <td>209113113.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ENST00000415913.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>449</th>\n",
       "      <td>FGFR1</td>\n",
       "      <td>N546K</td>\n",
       "      <td>Pemigatinib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>8</td>\n",
       "      <td>38274849.0</td>\n",
       "      <td>38274849.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000341462.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>450 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       gene variant         drugs clinical_significance chromosome  \\\n",
       "0    MAP2K1   P124S   Selumetinib            Resistance         15   \n",
       "1    MAP2K1    Q56P   Selumetinib            Resistance         15   \n",
       "2    PDGFRA   D842V      Imatinib            Resistance          4   \n",
       "3      ARAF   S214C     Sorafenib  Sensitivity/Response          X   \n",
       "4      JAK2   V617F    Fedratinib  Sensitivity/Response          9   \n",
       "..      ...     ...           ...                   ...        ...   \n",
       "445    ABL1   H396P     Dasatinib  Sensitivity/Response          9   \n",
       "446    EZH2   Y646H  Tazemetostat  Sensitivity/Response          7   \n",
       "447    EGFR   G598V     Erlotinib  Sensitivity/Response          7   \n",
       "448    IDH1   R132C      Olaparib  Sensitivity/Response          2   \n",
       "449   FGFR1   N546K   Pemigatinib  Sensitivity/Response          8   \n",
       "\n",
       "           start         stop reference_bases variant_bases  \\\n",
       "0     66729162.0   66729162.0               C             T   \n",
       "1     66727451.0   66727451.0               A             C   \n",
       "2     55152093.0   55152093.0               A             T   \n",
       "3     47426121.0   47426121.0               C             G   \n",
       "4      5073770.0    5073770.0               G             T   \n",
       "..           ...          ...             ...           ...   \n",
       "445  133750356.0  133750356.0               A             C   \n",
       "446  148508728.0  148508728.0               A             G   \n",
       "447   55233043.0   55233043.0               G             T   \n",
       "448  209113113.0  209113113.0               G             A   \n",
       "449   38274849.0   38274849.0               G             T   \n",
       "\n",
       "    representative_transcript  \n",
       "0           ENST00000307102.5  \n",
       "1           ENST00000307102.5  \n",
       "2           ENST00000257290.5  \n",
       "3           ENST00000377045.4  \n",
       "4           ENST00000381652.3  \n",
       "..                        ...  \n",
       "445         ENST00000318560.5  \n",
       "446         ENST00000320356.7  \n",
       "447         ENST00000275493.2  \n",
       "448         ENST00000415913.1  \n",
       "449         ENST00000341462.5  \n",
       "\n",
       "[450 rows x 10 columns]"
      ]
     },
     "execution_count": 174,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# fix error H3-3A: for drug table and evidence table\n",
    "error_gene = 'H3-3A'\n",
    "error_drug = 'ONC201'\n",
    "fixdrug_df = fixdrug_df[~fixdrug_df['gene'].str.contains(error_gene)]\n",
    "fixdrug_df = fixdrug_df.dropna(subset=['reference_bases','variant_bases']).reset_index(drop=True)\n",
    "fixdrug_df.to_csv('../datasets/middlefile/civic_evidence_table.csv', index=None)\n",
    "fixdrug_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>drugname</th>\n",
       "      <th>smile</th>\n",
       "      <th>molecular_weight</th>\n",
       "      <th>molecular_formula</th>\n",
       "      <th>atom</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Selumetinib</td>\n",
       "      <td>CN1C=NC2=C1C=C(C(=C2F)NC3=C(C=C(C=C3)Br)Cl)C(=...</td>\n",
       "      <td>457.7</td>\n",
       "      <td>C17H15BrClFN4O3</td>\n",
       "      <td>[Atom(1, Br), Atom(2, Cl), Atom(3, F), Atom(4,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Imatinib</td>\n",
       "      <td>CC1=C(C=C(C=C1)NC(=O)C2=CC=C(C=C2)CN3CCN(CC3)C...</td>\n",
       "      <td>493.6</td>\n",
       "      <td>C29H31N7O</td>\n",
       "      <td>[Atom(1, O), Atom(2, N), Atom(3, N), Atom(4, N...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Sorafenib</td>\n",
       "      <td>CNC(=O)C1=NC=CC(=C1)OC2=CC=C(C=C2)NC(=O)NC3=CC...</td>\n",
       "      <td>464.8</td>\n",
       "      <td>C21H16ClF3N4O3</td>\n",
       "      <td>[Atom(1, Cl), Atom(2, F), Atom(3, F), Atom(4, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Fedratinib</td>\n",
       "      <td>CC1=CN=C(N=C1NC2=CC(=CC=C2)S(=O)(=O)NC(C)(C)C)...</td>\n",
       "      <td>524.7</td>\n",
       "      <td>C27H36N6O3S</td>\n",
       "      <td>[Atom(1, S), Atom(2, O), Atom(3, O), Atom(4, O...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Tanespimycin</td>\n",
       "      <td>C[C@H]1C[C@@H]([C@@H]([C@H](/C=C(/[C@@H]([C@H]...</td>\n",
       "      <td>585.7</td>\n",
       "      <td>C31H43N3O8</td>\n",
       "      <td>[Atom(1, O), Atom(2, O), Atom(3, O), Atom(4, O...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>Arsenic Trioxide</td>\n",
       "      <td>[O-2].[O-2].[O-2].[As+3].[As+3]</td>\n",
       "      <td>197.841</td>\n",
       "      <td>As2O3</td>\n",
       "      <td>[Atom(1, As), Atom(2, As), Atom(3, O), Atom(4,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>Venetoclax</td>\n",
       "      <td>CC1(CCC(=C(C1)C2=CC=C(C=C2)Cl)CN3CCN(CC3)C4=CC...</td>\n",
       "      <td>868.4</td>\n",
       "      <td>C45H50ClN7O7S</td>\n",
       "      <td>[Atom(1, Cl), Atom(2, S), Atom(3, O), Atom(4, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>Gilteritinib</td>\n",
       "      <td>CCC1=C(N=C(C(=N1)C(=O)N)NC2=CC(=C(C=C2)N3CCC(C...</td>\n",
       "      <td>552.7</td>\n",
       "      <td>C29H44N8O3</td>\n",
       "      <td>[Atom(1, O), Atom(2, O), Atom(3, O), Atom(4, N...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>Tazemetostat</td>\n",
       "      <td>CCN(C1CCOCC1)C2=CC(=CC(=C2C)C(=O)NCC3=C(C=C(NC...</td>\n",
       "      <td>572.7</td>\n",
       "      <td>C34H44N4O4</td>\n",
       "      <td>[Atom(1, O), Atom(2, O), Atom(3, O), Atom(4, O...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>Pemigatinib</td>\n",
       "      <td>CCN1C2=C3C=C(NC3=NC=C2CN(C1=O)C4=C(C(=CC(=C4F)...</td>\n",
       "      <td>487.5</td>\n",
       "      <td>C24H27F2N5O4</td>\n",
       "      <td>[Atom(1, F), Atom(2, F), Atom(3, O), Atom(4, O...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>97 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            drugname                                              smile  \\\n",
       "0        Selumetinib  CN1C=NC2=C1C=C(C(=C2F)NC3=C(C=C(C=C3)Br)Cl)C(=...   \n",
       "1           Imatinib  CC1=C(C=C(C=C1)NC(=O)C2=CC=C(C=C2)CN3CCN(CC3)C...   \n",
       "2          Sorafenib  CNC(=O)C1=NC=CC(=C1)OC2=CC=C(C=C2)NC(=O)NC3=CC...   \n",
       "3         Fedratinib  CC1=CN=C(N=C1NC2=CC(=CC=C2)S(=O)(=O)NC(C)(C)C)...   \n",
       "4       Tanespimycin  C[C@H]1C[C@@H]([C@@H]([C@H](/C=C(/[C@@H]([C@H]...   \n",
       "..               ...                                                ...   \n",
       "93  Arsenic Trioxide                    [O-2].[O-2].[O-2].[As+3].[As+3]   \n",
       "94        Venetoclax  CC1(CCC(=C(C1)C2=CC=C(C=C2)Cl)CN3CCN(CC3)C4=CC...   \n",
       "95      Gilteritinib  CCC1=C(N=C(C(=N1)C(=O)N)NC2=CC(=C(C=C2)N3CCC(C...   \n",
       "96      Tazemetostat  CCN(C1CCOCC1)C2=CC(=CC(=C2C)C(=O)NCC3=C(C=C(NC...   \n",
       "97       Pemigatinib  CCN1C2=C3C=C(NC3=NC=C2CN(C1=O)C4=C(C(=CC(=C4F)...   \n",
       "\n",
       "   molecular_weight molecular_formula  \\\n",
       "0             457.7   C17H15BrClFN4O3   \n",
       "1             493.6         C29H31N7O   \n",
       "2             464.8    C21H16ClF3N4O3   \n",
       "3             524.7       C27H36N6O3S   \n",
       "4             585.7        C31H43N3O8   \n",
       "..              ...               ...   \n",
       "93          197.841             As2O3   \n",
       "94            868.4     C45H50ClN7O7S   \n",
       "95            552.7        C29H44N8O3   \n",
       "96            572.7        C34H44N4O4   \n",
       "97            487.5      C24H27F2N5O4   \n",
       "\n",
       "                                                 atom  \n",
       "0   [Atom(1, Br), Atom(2, Cl), Atom(3, F), Atom(4,...  \n",
       "1   [Atom(1, O), Atom(2, N), Atom(3, N), Atom(4, N...  \n",
       "2   [Atom(1, Cl), Atom(2, F), Atom(3, F), Atom(4, ...  \n",
       "3   [Atom(1, S), Atom(2, O), Atom(3, O), Atom(4, O...  \n",
       "4   [Atom(1, O), Atom(2, O), Atom(3, O), Atom(4, O...  \n",
       "..                                                ...  \n",
       "93  [Atom(1, As), Atom(2, As), Atom(3, O), Atom(4,...  \n",
       "94  [Atom(1, Cl), Atom(2, S), Atom(3, O), Atom(4, ...  \n",
       "95  [Atom(1, O), Atom(2, O), Atom(3, O), Atom(4, N...  \n",
       "96  [Atom(1, O), Atom(2, O), Atom(3, O), Atom(4, O...  \n",
       "97  [Atom(1, F), Atom(2, F), Atom(3, O), Atom(4, O...  \n",
       "\n",
       "[97 rows x 5 columns]"
      ]
     },
     "execution_count": 169,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "drug_table = drug_table[~drug_table['drugname'].str.contains(error_drug)]\n",
    "drug_table.to_csv('../datasets/middlefile/civic_drug_table.csv', index=None)\n",
    "drug_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 53/53 [01:55<00:00,  2.18s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      gene uniprotac                                              fasta\n",
      "0   MAP2K1    Q02750  MPKKKPTPIQLNPAPDGSAVNGTSSAETNLEALQKKLEELELDEQQ...\n",
      "1   PDGFRA    P16234  MGTSHPAFLVLGCLLTGLSLILCQLSLPSILPNENEKVVQLNSSFS...\n",
      "2     ARAF    P10398  MEPPRGPPANGAEPSRAVGTVKVYLPNKQRTVVTVRDGMSVYDSLD...\n",
      "3     JAK2    O60674  MGMACLTMTEMEGTSTSSIYQNGDISGNANSMKQIDPVLQVYLYHS...\n",
      "4     NRAS    P01111  MTEYKLVVVGAGGVGKSALTIQLIQNHFVDEYDPTIEDSYRKQVVI...\n",
      "..     ...       ...                                                ...\n",
      "48   BRCA1    P38398  MDLSALRVEEVQNVINAMQKILECPICLELIKEPVSTKCDHIFCKF...\n",
      "49   PREX2    Q70Z35  MSEDSRGDSRAESAKDLEKQLRLRVCVLSELQKTERDYVGTLEFLV...\n",
      "50   CSF3R    Q99062  MARLGNCSLTWAALIILLLPGSLEECGHISVSAPIVHLGDPITASC...\n",
      "51    BCL2    O43521  MAKQPSDVSSECDREGRQLQPAERPPQLRPGAPTSLQTEPQGNPEG...\n",
      "52    EZH2    Q15910  MGQTGKKSEKGPVCWRKRVKSEYMRLRQLKRFRRADEVKSMFSSNR...\n",
      "\n",
      "[53 rows x 3 columns]\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "gene_list = fixdrug_df['gene'].unique()\n",
    "gene_table = pd.DataFrame(columns=['gene', 'uniprotac', 'fasta'])\n",
    "for i in tqdm(range(len(gene_list))):\n",
    "    url = 'https://rest.uniprot.org/uniprotkb/stream?compressed=false&format=fasta&query=(reviewed:true)%20AND%20(organism_id:9606)%20AND%20' + gene_list[i]\n",
    "    all_fastas = requests.get(url).text\n",
    "    fasta_list = re.split(r'\\n(?=>)', all_fastas)\n",
    "    entry = [fasta for fasta in fasta_list if gene_list[i] in fasta][0] # fix: use 'GN='+gene_list[i] instead of gene_list[i]\n",
    "    gene = gene_list[i]\n",
    "    uniprotac = entry.split('|')[1]\n",
    "    fasta = entry.split(' ')[-1].replace('\\n','')[4:]\n",
    "    gene_table = gene_table.append([{'gene':gene, 'uniprotac':uniprotac, 'fasta':fasta}], ignore_index=True)\n",
    "print(gene_table)\n",
    "gene_table.to_csv('../datasets/middlefile/civic_gene_table.csv', index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>variant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>P124S</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>Q56P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PDGFRA</td>\n",
       "      <td>D842V</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ARAF</td>\n",
       "      <td>S214C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JAK2</td>\n",
       "      <td>V617F</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>445</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>H396P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>446</th>\n",
       "      <td>EZH2</td>\n",
       "      <td>Y646H</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>447</th>\n",
       "      <td>EGFR</td>\n",
       "      <td>G598V</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>448</th>\n",
       "      <td>IDH1</td>\n",
       "      <td>R132C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>449</th>\n",
       "      <td>FGFR1</td>\n",
       "      <td>N546K</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>450 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       gene variant\n",
       "0    MAP2K1   P124S\n",
       "1    MAP2K1    Q56P\n",
       "2    PDGFRA   D842V\n",
       "3      ARAF   S214C\n",
       "4      JAK2   V617F\n",
       "..      ...     ...\n",
       "445    ABL1   H396P\n",
       "446    EZH2   Y646H\n",
       "447    EGFR   G598V\n",
       "448    IDH1   R132C\n",
       "449   FGFR1   N546K\n",
       "\n",
       "[450 rows x 2 columns]"
      ]
     },
     "execution_count": 176,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check variant with fasta\n",
    "check_variant_df = fixdrug_df[['gene','variant']]\n",
    "check_variant_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "metadata": {},
   "outputs": [],
   "source": [
    "protein_dict = {'C':0, 'D':1, 'S':2, 'Q':3, 'K':4,\n",
    "        'I':5, 'P':6, 'T':7, 'F':8, 'N':9,\n",
    "        'G':10, 'H':11, 'L':12, 'R':13, 'W':14,\n",
    "        'A':15, 'V':16, 'E':17, 'Y':18, 'M':19}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "pos_before MYD88 L265P\n",
      "pos_before FGFR2 N550K\n",
      "pos_after BRCA1 W1815X\n",
      "pos_before ABL1 R351W\n",
      "pos_before ABL1 R351W\n",
      "pos_before EZH2 Y646H\n"
     ]
    }
   ],
   "source": [
    "gene_table_manu = pd.read_csv('../datasets/middlefile/civic_gene_table_fixed.csv')\n",
    "for i in range(len(check_variant_df)):\n",
    "    gene = check_variant_df['gene'][i]\n",
    "    #print(gene)\n",
    "    variant = check_variant_df['variant'][i]\n",
    "    fasta = gene_table_manu[gene_table_manu['gene'] == gene]['fasta'].values[0]\n",
    "    #print(fasta)\n",
    "    pos_before = variant[0]\n",
    "    pos = int(variant[1:-1])\n",
    "    #print(pos)\n",
    "    pos_after = variant[-1]\n",
    "    \n",
    "    # check pos_after\n",
    "    if(pos_after not in protein_dict.keys()):\n",
    "        print('pos_after', gene, variant)\n",
    "    # check pos_before\n",
    "    if(fasta[pos-1] != pos_before):\n",
    "        print('pos_before', gene, variant)\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 202,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "pos_before MYD88 L265P\n",
      "pos_before FGFR2 N550K\n",
      "pos_after BRCA1 W1815X\n",
      "pos_before ABL1 R351W\n",
      "pos_before ABL1 R351W\n",
      "pos_before EZH2 Y646H\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>variant</th>\n",
       "      <th>drugs</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>reference_bases</th>\n",
       "      <th>variant_bases</th>\n",
       "      <th>representative_transcript</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>P124S</td>\n",
       "      <td>Selumetinib</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>15</td>\n",
       "      <td>66729162.0</td>\n",
       "      <td>66729162.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000307102.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>Q56P</td>\n",
       "      <td>Selumetinib</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>15</td>\n",
       "      <td>66727451.0</td>\n",
       "      <td>66727451.0</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>ENST00000307102.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PDGFRA</td>\n",
       "      <td>D842V</td>\n",
       "      <td>Imatinib</td>\n",
       "      <td>Resistance</td>\n",
       "      <td>4</td>\n",
       "      <td>55152093.0</td>\n",
       "      <td>55152093.0</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000257290.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ARAF</td>\n",
       "      <td>S214C</td>\n",
       "      <td>Sorafenib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>X</td>\n",
       "      <td>47426121.0</td>\n",
       "      <td>47426121.0</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>ENST00000377045.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JAK2</td>\n",
       "      <td>V617F</td>\n",
       "      <td>Fedratinib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>9</td>\n",
       "      <td>5073770.0</td>\n",
       "      <td>5073770.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000381652.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>444</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>G250E</td>\n",
       "      <td>Dasatinib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>9</td>\n",
       "      <td>133738349.0</td>\n",
       "      <td>133738349.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ENST00000318560.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>445</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>H396P</td>\n",
       "      <td>Dasatinib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>9</td>\n",
       "      <td>133750356.0</td>\n",
       "      <td>133750356.0</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>ENST00000318560.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>447</th>\n",
       "      <td>EGFR</td>\n",
       "      <td>G598V</td>\n",
       "      <td>Erlotinib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>7</td>\n",
       "      <td>55233043.0</td>\n",
       "      <td>55233043.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000275493.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>448</th>\n",
       "      <td>IDH1</td>\n",
       "      <td>R132C</td>\n",
       "      <td>Olaparib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>2</td>\n",
       "      <td>209113113.0</td>\n",
       "      <td>209113113.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>ENST00000415913.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>449</th>\n",
       "      <td>FGFR1</td>\n",
       "      <td>N546K</td>\n",
       "      <td>Pemigatinib</td>\n",
       "      <td>Sensitivity/Response</td>\n",
       "      <td>8</td>\n",
       "      <td>38274849.0</td>\n",
       "      <td>38274849.0</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>ENST00000341462.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>444 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       gene variant        drugs clinical_significance chromosome  \\\n",
       "0    MAP2K1   P124S  Selumetinib            Resistance         15   \n",
       "1    MAP2K1    Q56P  Selumetinib            Resistance         15   \n",
       "2    PDGFRA   D842V     Imatinib            Resistance          4   \n",
       "3      ARAF   S214C    Sorafenib  Sensitivity/Response          X   \n",
       "4      JAK2   V617F   Fedratinib  Sensitivity/Response          9   \n",
       "..      ...     ...          ...                   ...        ...   \n",
       "444    ABL1   G250E    Dasatinib  Sensitivity/Response          9   \n",
       "445    ABL1   H396P    Dasatinib  Sensitivity/Response          9   \n",
       "447    EGFR   G598V    Erlotinib  Sensitivity/Response          7   \n",
       "448    IDH1   R132C     Olaparib  Sensitivity/Response          2   \n",
       "449   FGFR1   N546K  Pemigatinib  Sensitivity/Response          8   \n",
       "\n",
       "           start         stop reference_bases variant_bases  \\\n",
       "0     66729162.0   66729162.0               C             T   \n",
       "1     66727451.0   66727451.0               A             C   \n",
       "2     55152093.0   55152093.0               A             T   \n",
       "3     47426121.0   47426121.0               C             G   \n",
       "4      5073770.0    5073770.0               G             T   \n",
       "..           ...          ...             ...           ...   \n",
       "444  133738349.0  133738349.0               G             A   \n",
       "445  133750356.0  133750356.0               A             C   \n",
       "447   55233043.0   55233043.0               G             T   \n",
       "448  209113113.0  209113113.0               G             A   \n",
       "449   38274849.0   38274849.0               G             T   \n",
       "\n",
       "    representative_transcript  \n",
       "0           ENST00000307102.5  \n",
       "1           ENST00000307102.5  \n",
       "2           ENST00000257290.5  \n",
       "3           ENST00000377045.4  \n",
       "4           ENST00000381652.3  \n",
       "..                        ...  \n",
       "444         ENST00000318560.5  \n",
       "445         ENST00000318560.5  \n",
       "447         ENST00000275493.2  \n",
       "448         ENST00000415913.1  \n",
       "449         ENST00000341462.5  \n",
       "\n",
       "[444 rows x 10 columns]"
      ]
     },
     "execution_count": 202,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# remove error evidence\n",
    "fix_evidence_df = fixdrug_df.copy()\n",
    "for i in range(len(check_variant_df)):\n",
    "    gene = check_variant_df['gene'][i]\n",
    "    variant = check_variant_df['variant'][i]\n",
    "    fasta = gene_table_manu[gene_table_manu['gene'] == gene]['fasta'].values[0]\n",
    "    pos_before = variant[0]\n",
    "    pos = int(variant[1:-1])\n",
    "    pos_after = variant[-1]\n",
    "    \n",
    "    # check pos_after\n",
    "    if(pos_after not in protein_dict.keys()):\n",
    "        print('pos_after', gene, variant)\n",
    "        fix_evidence_df = fix_evidence_df[~((fix_evidence_df['gene'] == gene) & (fix_evidence_df['variant'] == variant))]\n",
    "    # check pos_before\n",
    "    if(fasta[pos-1] != pos_before):\n",
    "        print('pos_before', gene, variant)\n",
    "        fix_evidence_df = fix_evidence_df[~((fix_evidence_df['gene'] == gene) & (fix_evidence_df['variant'] == variant))]\n",
    "fix_evidence_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "metadata": {},
   "outputs": [],
   "source": [
    "fix_evidence_df.to_csv('../datasets/middlefile/civic_evidence_table.csv', index=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 204,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Resistance              217\n",
       "Sensitivity/Response    215\n",
       "Reduced Sensitivity      10\n",
       "Adverse Response          2\n",
       "Name: clinical_significance, dtype: int64"
      ]
     },
     "execution_count": 204,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fix_evidence_df['clinical_significance'].value_counts()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.9.12 64-bit",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "6fff98fc3b3d81bd655c2cc48858186e4d9e2db7b515bf1c3221888f12a62f87"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
